---
title: "101-data"
output: html_document
date: "2024-10-12"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```

wget -np -r --cut-dirs 3 http://slimsdata.genomecenter.ucdavis.edu/Data/izi9cfibu/Un_DTSA984/Project_AFSP_LONGFIN_RAD/

Two plates of data, putting in data/raw

Have demultiplexed.   

`/home/maccamp/longfin-histories/data/raw/plate-1`    
`/home/maccamp/longfin-histories/data/raw/plate-2`    

```{r}
m1<-read_csv("meta/samples-10312024.csv") %>% mutate(Well=paste0(Row,Column)) %>% select(-Row, -Column)
barcodes<-read_csv("meta/expected-barcodes.csv")
m1<-m1 %>% left_join(barcodes) 
```

Samples are of the form:
PLATE01_RA_GGAAACATCGTGCAGG.fastq

```{r}
p1<-m1 %>% filter(Plate==1) %>% mutate(File1=paste0("PLATE0",Plate,"_RA_GG",Barcode,"TGCAGG.fastq")) %>%
  mutate(File2=paste0("PLATE0",Plate,"_RB_GG",Barcode,"TGCAGG.fastq")) %>%
  mutate(File3=paste0(SampleID,"_R1.fastq")) %>%
  mutate(File4=paste0(SampleID,"_R2.fastq")) %>%
  mutate(Command1 = paste0("mv data/raw/plate-1/", File1, " data/renamed/",File3)) %>%
  mutate(Command2 = paste0("mv data/raw/plate-1/", File2, " data/renamed/",File4)) 
p2<-m1 %>% filter(Plate==2) %>% mutate(File1=paste0("PLATE0",Plate,"_RA_GG",Barcode,"TGCAGG.fastq")) %>%
  mutate(File2=paste0("PLATE0",Plate,"_RB_GG",Barcode,"TGCAGG.fastq")) %>%
  mutate(File3=paste0(SampleID,"_R1.fastq")) %>%
  mutate(File4=paste0(SampleID,"_R2.fastq")) %>%
  mutate(Command1 = paste0("mv data/raw/plate-2/", File1, " data/renamed/",File3)) %>%
  mutate(Command2 = paste0("mv data/raw/plate-2/", File2, " data/renamed/",File4)) 

c1<-p1%>%select(Command1) %>% rename(Command=Command1)
c2<-p1%>%select(Command2) %>% rename(Command=Command2)
c3<-p2%>%select(Command1) %>% rename(Command=Command1)
c4<-p2%>%select(Command2) %>% rename(Command=Command2)
bind_rows(c1,c2,c3,c4) %>% write_tsv("101.1-rename-commands.txt", col_names = FALSE)
```


For comparison:
/home/skieran/lfs/raw/mac_ototyped$ 

ls -al /home/skieran/lfs/raw/mac_ototyped | awk '{ if ($5 >= 40) print }' | grep R1 | wc -l
26

```{r}
fish<-read_csv("Longfin smelt meta and tissues/all_fish_clust_ids.csv") 
fish$SampleID<-gsub("_0","",fish$fishid)
fish$SampleID<-gsub("_","",fish$SampleID)
```

```{r}
m2<-m1 %>% left_join(fish) 
m2 %>% group_by(clust_name_100) %>% summarize(Count=n())
```

I think what would be a good thing to do is sequence all of clusters 1, 2, 3 and a randomized subset of cluster 4 (n ~ 25) to represent that.  In that way we can model the transition timing (clusters 1,2,3) and the brackish water residency (discrete, clusters 1,2,3 vs 4).


# Do align

 GCF_021917145.1_fHypTra1_genomic.fna in data/dsm-align
 `bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/GCF021917145.1/GCF_021917145.1_fHypTra1_genomic.fna`
 
Also trying Shannon's individuals in data/alt-demultiplexed

Presumably, it'll be fine. Running against `bash ../../101.2-do-align.sh to-align.txt /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq.gz`    

## Check metrics 
```{r}
df<-read_csv("meta/alt-demulti.txt", col_names = c("SampleID","Aligned","Dedup","Cov"))
ddf<-m2 %>% left_join(df) %>% mutate(Path=paste0("data/alt-demultiplexed/",SampleID,".sort.flt.bam"))
```


```{r}
ggplot(ddf) +
  geom_histogram(aes(x=Dedup)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("Count")
```

```{r}
m100<-ddf %>% filter(Dedup>1e5) 
m100 %>% group_by(clust_name_100) %>% summarize(Count=n())
m100 %>% select(Path) %>% write_tsv(col_names = FALSE, file="bamlists/49.bamlist")

m300<-ddf %>% filter(Dedup > 3e5)
m300 %>% select(Path) %>% write_tsv(col_names = FALSE, file="bamlists/34.bamlist")
```


Lets quickly generate a covariance matrix.
```{sh,, eval=FALSE}
srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/49.bamlist -ref /home/maccamp/genomes/GCF021917145.1/GCF_021917145.1_fHypTra1_genomic.fna \
-minInd 37 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/101/m100-ibs-75 >outputs/101/m100-ibs-75.out 2> outputs/101/m100-ibs-75.err &
```
	-> Number of sites retained after filtering: 37493 

```{r}
m <- as.matrix(read.table("outputs/101/m100-ibs-75.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(m100)

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, shape=clust_name_100, fill=Dedup), cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  scale_shape_manual(values=c(23,24,21,22)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_c(option="viridis") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16)) #+
#  facet_wrap(.~clust_name_100)

pc12
```

Scree plot

```{r}
vardf<-var %>% as_tibble() %>% rename(`Percent Variance`=value) %>% head() %>% mutate(PC=1:n())

scree<-ggplot(vardf, aes(x=PC, y=`Percent Variance`)) +
  geom_line() +
  geom_point() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks = seq(1:6)) +
  theme(axis.text = element_text(size=10)) +
  theme(axis.title = element_text(size=14)) +
  theme(plot.title = element_text(hjust=0, size=16))


scree
```


## Filter to HS/LS

```{r}
m36<-m100 %>% filter(clust_name_100 %in% c("HS","LS")) %>% 
  mutate(ybin=ifelse(clust_name_100=="HS",0,1))
write_csv(m36, file="meta/m36.csv")
```


Call some snps to see about PCA/LD

```{sh, eval=FALSE}
srun -t 12:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 31 -bam bamlists/34.bamlist -ref /home/maccamp/genomes/GCF021917145.1/GCF_021917145.1_fHypTra1_genomic.fna \
-rf outputs/101/chrom-regions.txt \
-out outputs/101/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/101/snps-wgs.out 2> outputs/101/snps-wgs.err &

#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz
```
