---
title: "103-lfs-gwas"
output: html_document
date: "2024-12-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(RColorBrewer)
library(snpR)
library(vcfR)
library(adegenet)
library(ggrepel)
```

let's get alt-demultiplexed fish aligned against lfs

data/alt-demulti has the properly sourced files.

Running alignment against the basic genome assembly from pacBio of lfs

symlinking fastqs in alt-demulti to alt-longfin and aligning against pacbio assembly without hic

bash ../../101.2-do-align.sh to-align.txt /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq.gz

## Check metrics 
```{r}
fish<-read_csv("Longfin smelt meta and tissues/all_fish_clust_ids.csv") 
fish$SampleID<-gsub("_0","",fish$fishid)
fish$SampleID<-gsub("_","",fish$SampleID)
```



```{r}
df<-read_csv("meta/alt-longfin.txt", col_names = c("SampleID","Aligned","Dedup","Cov"))
ddf<- left_join(df,fish, by=c("SampleID"="SampleID")) %>% mutate(Path=paste0("data/alt-longfin/",SampleID,".sort.flt.bam"))
```



```{r}
ggplot(ddf) +
  geom_histogram(aes(x=Dedup)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("Count")
```

```{r}
m100<-ddf %>% filter(Dedup>1e5) 
m100 %>% group_by(clust_name_100) %>% summarize(Count=n())
m100 %>% select(Path) %>% write_tsv(col_names = FALSE, file="bamlists/53.bamlist")
```

53 samples   

Lets quickly generate a covariance matrix.
```{sh,, eval=FALSE}
srun -p bigmemm -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12 -bam bamlists/53.bamlist -ref  /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq \
-minInd 37 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/103/m100-ibs-75 >outputs/103/m100-ibs-75.out 2> outputs/103/m100-ibs-75.err &
```
	-> Number of sites retained after filtering:  46829 (from 37,493)!!! 

```{r}
m <- as.matrix(read.table("outputs/103/m100-ibs-75.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(m100)

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, shape=clust_name_100, fill=Dedup), cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  scale_shape_manual(values=c(23,24,21,22)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_c(option="viridis") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16)) #+
#  facet_wrap(.~clust_name_100)

pc12
```

Ack, looks inversion-y

Scree plot

```{r}
vardf<-var %>% as_tibble() %>% rename(`Percent Variance`=value) %>% head() %>% mutate(PC=1:n())

scree<-ggplot(vardf, aes(x=PC, y=`Percent Variance`)) +
  geom_line() +
  geom_point() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks = seq(1:6)) +
  theme(axis.text = element_text(size=10)) +
  theme(axis.title = element_text(size=14)) +
  theme(plot.title = element_text(hjust=0, size=16))


scree
```


## Filter to HS/LS

```{r}
m37 <-m100 %>% filter(clust_name_100 %in% c("HS","LS")) %>% 
  mutate(ybin=ifelse(clust_name_100=="HS",0,1))
write_csv(m37, file="meta/m37.csv")

m37 %>% select(Path) %>% write_tsv(col_names=FALSE, file="bamlists/37.bamlist")
m37 %>% select(ybin) %>% write_tsv(col_names=FALSE, file="bamlists/37.pheno")

```

# Now for basic GWAS

```{sh, eval=FALSE}
 srun -p bigmemm -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=8 \
 $HOME/angsd/angsd -P 8 -doAsso 1 -doPost 1 -yBin bamlists/37.pheno \
 -GL 1 -nThreads 8 -minInd 18 \
 -minMapQ 10 -minQ 20 -minMaf 0.05 \
 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -out outputs/103/do-asso \
  -bam bamlists/37.bamlist > outputs/103/do-asso.out 2> outputs/103/do-asso.err &
```


Previously with lfs  outputs/101/chrom-regions.txt  
	-> Number of sites retained after filtering: 62429 
Now
	-> Number of sites retained after filtering: 87232 

```{r}
data<-read_tsv("outputs/103/do-asso.lrt0.gz")
```

```{r}
df <- data %>% mutate(log10p = -log10(dchisq(LRT, df = 1))) %>%
  filter(log10p>=0 & log10p != "Inf") %>%
  mutate(p = dchisq(LRT, df=1)) %>%
  mutate(bonfer = p.adjust(p, method = "bonferroni")) %>% 
  mutate(BHfdr = p.adjust(p, method = "BH")) %>%
  group_by(Chromosome) %>%
  mutate(adjpchrom = p.adjust(p, method = "bonferroni")) %>%
  mutate(BHfdrchrom = p.adjust(p, method = "BH"))

df %>% arrange(-LRT) %>% head(n=20)
```
```{r}
df20<-df %>% arrange(-LRT) %>% head(n=10)

dfs<-df %>%filter(Chromosome %in% df20$Chromosome)
```

```{r}
nb.cols <- length(unique(dfs$Chromosome))
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)
#randomize
mycolors<-sample(mycolors)
```

```{r}
ggplot(dfs) +
  geom_point(aes(x=Position, y=log10p, color=Chromosome), alpha=0.75, cex=0.5) +
 # geom_hline(yintercept = -log10(0.05/nrow(dddf)), col="black", linetype=2, alpha=0.5) +
  geom_hline(yintercept= 4, col="black", linetype=1, alpha=0.5) +
  theme_bw() +
#  scale_x_continuous(breaks=chroms$Position, labels=chroms$Chromosome) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=8)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("-log10(p)\n") +
  xlab("\nChromosome") +
  ggtitle("HS vs LS") +
  theme(plot.title = element_text(hjust=0.5) ) +
  theme(plot.subtitle = element_text(hjust=0.5)) +
  facet_wrap(.~Chromosome, ncol = 5, scales="free_x") +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
```

# Now to consider what to test in GWAS.

Growth rates?


# Higher coverage samples and the pc patterns

```{r}
m250<-ddf %>% filter(Dedup>4e5) 
m250 %>% group_by(clust_name_100) %>% summarize(Count=n())
m250 %>% select(Path) %>% write_tsv(col_names = FALSE, file="bamlists/m250.bamlist")
m250 %>% write_csv(col_names = TRUE, file="meta/m250.csv")

```

37 inds    

May expect this:    

```{r}
sub<-ggplot(covs %>% filter(Dedup > 4e5)) +
  geom_point(aes(x=V1, y=V2, shape=clust_name_100, fill=Dedup), cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  scale_shape_manual(values=c(23,24,21,22)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_c(option="viridis") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16))

sub
```

redoing with .9 inds doIbs
```{sh, eval=FALSE}
srun -p bigmemm -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12 -bam bamlists/m250.bamlist -ref  /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq \
-minInd 33 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/103/m250-ibs-90 >outputs/103/m250-ibs-90.out 2> outputs/103/m250-ibs-90.err &
```

Call some snps to see about PCA/LD

```{sh, eval=FALSE}
srun -t 12:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 33 -bam bamlists/m250.bamlist -ref /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq \
 -out outputs/103/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/103/snps-wgs.out 2> outputs/103/snps-wgs.err &

#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz
```

Total genotyping rate is 0.937394.
16778 variants and 37 people pass filters and QC.

Prune etc. locall
   -n, --nsites-per-win N          keep at most N sites in the -w window, removing sites with small AF first

bcftools +fill-tags plink.vcf.gz | bcftools +prune -l 0.3 -w10000bp -n 1 > plink-pruned.vcf


```{r}
snps<-import.snpR.data(genotypes = "outputs/103/plink-pruned.vcf", 
      sample.meta = m250 %>% select(SampleID, Aligned, Dedup, year, WY, region, clust_name_100))
```

```{r}
p <- plot_clusters(snps, facets = c("Dedup.clust_name_100"), viridis.option = "H")
p$plot$pca
```


```{r}
vcf<-read.vcfR(file="outputs/103/plink-pruned.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(m250$clust_name_100)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(m250)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=as.factor(year), shape=clust_name_100), alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  scale_shape_manual(values=c(21,24,23,22)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
 # theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=SampleID))
pc12

ggsave("outputs/103/lfs-lfs-align-prunedMAF-pca.jpeg")
```


```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1)  
loadings$Position<-gsub("ptg.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)

tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .01) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("ptg.*\\.","",Allele)) 
tops$Chrom<-gsub("_.*$","",tops$Allele)
tops %>% group_by(Chrom) %>% summarize(Count=n()) %>% arrange(-Count)

```

What about admixture analyses?

```{r}
plot_structure(snps, facet = "year", k=2)
ggsave("outputs/103/admixture-years.jpeg")
```

Interesting....
forage fishes
HS2253_AGTCAA-CAAGACTA	Night Smelt
HS2253_AGTCAA-CTCAATGA	Night Smelt
HS2253_AGTCAA-GGTGCGAA	Night Smelt
plate-3-ACACTGAC	Night Smelt
plate-3-AGTCACTA	Night Smelt
plate-3-CCGACAAC	Night Smelt
```{sh, eval=FALSE}
#plate1 alt-1
cp HS2253_AGTCAA_R*_*CAAGACTA*.fastq ~/longfin-histories/data/night/
cp HS2253_AGTCAA_R*_*CTCAATGA*.fastq ~/longfin-histories/data/night/
cp HS2253_AGTCAA_R*_*GGTGCGAA*.fastq ~/longfin-histories/data/night/
#plate3 alt-3
cp HS2253_AGTCAA_R*ACACTGAC*.fastq ~/longfin-histories/data/night/
cp HS2253_AGTCAA_R*AGTCACTA*.fastq ~/longfin-histories/data/night/
cp HS2253_AGTCAA_R*CCGACAAC*.fastq ~/longfin-histories/data/night/
```

gzip,  then align!
bash ../../101.2-do-align.sh to-align.txt /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq.gz`

renaming Night01-06

```{sh, eval=FALSE}
srun -t 12:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 35 -bam bamlists/test-night.bamlist -ref /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq \
 -out outputs/103/snps-wgs-night  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/103/snps-night-wgs.out 2> outputs/103/snps-wgs-night.err &

#create vcf
plink --tped snps-wgs-night.tped --tfam snps-wgs-night.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out night
bgzip night.vcf 
tabix night.vcf.gz
```
