---
title: "104-lower-coverage"
output: html_document
date: "2025-02-04"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

Setting up low coverage analysis


```{r}
library(tidyverse)
library(RColorBrewer)
library(snpR)
library(vcfR)
library(adegenet)
library(ggrepel)
```

let's get alt-demultiplexed fish aligned against lfs

data/alt-demulti has the properly sourced files.

Running alignment against the basic genome assembly from pacBio of lfs

symlinking fastqs in alt-demulti to alt-longfin and aligning against pacbio assembly without hic

bash ../../101.2-do-align.sh to-align.txt /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq.gz

## Check metrics 
```{r}
fish<-read_csv("Longfin smelt meta and tissues/all_fish_clust_ids.csv") 
fish$SampleID<-gsub("_0","",fish$fishid)
fish$SampleID<-gsub("_","",fish$SampleID)
```



```{r}
df<-read_csv("meta/alt-longfin.txt", col_names = c("SampleID","Aligned","Dedup","Cov"))
ddf<- left_join(df,fish, by=c("SampleID"="SampleID")) %>% mutate(Path=paste0("data/alt-longfin/",SampleID,".sort.flt.bam"))
```



```{r}
ggplot(ddf) +
  geom_histogram(aes(x=Dedup)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("Count")
```

```{r}
m100<-ddf %>% filter(Dedup>1e5) 
m100 %>% group_by(clust_name_100) %>% summarize(Count=n())
m100 %>% select(Path) %>% write_tsv(col_names = FALSE, file="bamlists/53.bamlist")
```


53 samples   

Lets quickly generate a covariance matrix.
```{sh,, eval=FALSE}
srun -p bigmemm -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12 -bam bamlists/53.bamlist -ref  /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq \
-minInd 37 -minMapQ 10 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/103/m100-ibs-75 >outputs/103/m100-ibs-75.out 2> outputs/103/m100-ibs-75.err &
```
	-> Number of sites retained after filtering:  46829 (from 37,493)!!! 

```{r}
m <- as.matrix(read.table("outputs/103/m100-ibs-75.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(m100)

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, shape=clust_name_100, fill=Dedup), cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  scale_shape_manual(values=c(23,24,21,22)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_c(option="viridis") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16)) #+
#  facet_wrap(.~clust_name_100)

pc12
```

Ack, looks inversion-y

Scree plot

```{r}
vardf<-var %>% as_tibble() %>% rename(`Percent Variance`=value) %>% head() %>% mutate(PC=1:n())

scree<-ggplot(vardf, aes(x=PC, y=`Percent Variance`)) +
  geom_line() +
  geom_point() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks = seq(1:6)) +
  theme(axis.text = element_text(size=10)) +
  theme(axis.title = element_text(size=14)) +
  theme(plot.title = element_text(hjust=0, size=16))


scree
```


## Called snp version

90% data presence 
```{sh, eval=FALSE}
srun -t 8:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=8 $HOME/angsd/angsd -nthreads 9 \
-minInd 48 -bam bamlists/53.bamlist -ref /home/maccamp/longfin-histories/genomes/denovo-genome/lfs_draft_assembly.fastq \
 -out outputs/104/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/104/snps-wgs.out 2> outputs/104/snps-wgs.err &

#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz

#use bcftools to filter a bit?
bcftools +fill-tags plink.vcf.gz | bcftools +prune -l 0.3 -w10000bp -n 1 > plink-pruned.vcf

```



```{r}
snps<-import.snpR.data(genotypes = "outputs/104/plink-pruned.vcf", 
      sample.meta = m100 %>% select(SampleID, Aligned, Dedup, year, WY, region, clust_name_100))
```

```{r}
p <- plot_clusters(snps, facets = c("Dedup.clust_name_100"), viridis.option = "H")
p$plot$pca
```