---
title: "202-kmer-gwas"
output: html_document
date: "2025-07-02"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE)
```

```{r}
library(tidyverse)
```

Need to set up a conda environment with:

python 2.7
R
R packages; MASS, Mvnpermute, matrixcalc

conda create -n kmers python=2.7
conda activate kmers
conda install R

mkdir kmers
cd kmers
wget https://github.com/voichek/kmersGWAS/releases/download/v0.3-beta/v0_3_beta.zip
unzip

Make kmers?

in outputs/201

_1_ need fastqs as a text file, 'LFS540.txt'

/home/maccamp/longfin-histories/data/alt-demultiplexed/LFS540_R1.fastq.gz
/home/maccamp/longfin-histories/data/alt-demultiplexed/LFS540_R2.fastq.gz

_2_ run kmc.  

```{sh, eval=FALSE}
/home/maccamp/kmers/external_programs/kmc_v3 -t2 -k31 -ci2 @LFS540.txt LFS540_kmc_canon ./ 1> kmc_canon.1 2> kmc_canon.2
 
/home/maccamp/kmers/external_programs/kmc_v3 -t2 -k31 -ci0 -b @LFS540.txt LFS540_kmc_all ./ 1> kmc_all.1 2> kmc_all.2

/home/maccamp/kmers/bin/kmers_add_strand_information -c LFS540_kmc_canon -n LFS540_kmc_all -k 31 -o LFS540-kmers_with_strand

## clean up
rm *.kmc*
```

Produces LFS540-kmers_with_strand

Should make each individual a directory?

Can restrict to the set of fish that we run GWAS on?


Starting with 37 individuals

```{r}
m37<-read_csv("meta/m37.csv")
m37 %>% select(SampleID) %>% mutate(Commands=paste0("/home/maccamp/kmers/external_programs/kmc_v3 -t2 -k31 -ci2 @", SampleID,".txt ",SampleID,"_kmc_canon ./ 1> kmc.canon.1 2>kmc_canon.2; ", 
                                                    "/home/maccamp/kmers/external_programs/kmc_v3 -t2 -k31 -ci0 -b @", SampleID,".txt ",SampleID,"_kmc_all ./ 1> kmc_all.1 2>kmc_all.2; ",
                                                    "/home/maccamp/kmers/bin/kmers_add_strand_information -c ",SampleID,"_kmc_canon -n ",SampleID,"_kmc_all -k 31 -o ", SampleID,"-kmers_with_strand;",
                                                    " rm *.kmc* ;")) %>%
  select(Commands) %>% write_tsv("202.1-commands.txt", col_names = FALSE)
```

need to write a bunch of files with text

LFS540.txt
/home/maccamp/longfin-histories/data/alt-demultiplexed/LFS540_R1.fastq.gz
/home/maccamp/longfin-histories/data/alt-demultiplexed/LFS540_R2.fastq.gz

```{sh, eval=FALSE}
cut -f 1 -d',' ../../meta/m37.csv  | while read sample; do echo $sample; echo /home/maccamp/longfin-histories/data/alt-demultiplexed/${sample}_R1.fastq.gz >> $sample.txt; echo /home/maccamp/longfin-histories/data/alt-demultiplexed/${sample}_R2.fastq.gz >> $sample.txt; done;
```

Now to run commands, in outputs/202/

Seems to run out of memory with 32GB, moving to bigmemh

```{sh, eval=FALSE}
ln -s ../../202.1-commands.txt .
srun -p bigmemh -t 14:00:00 --mem=128GB --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 bash 202.1-commands.txt
```


Looks to have worked...

need a file with path to samples in it

(kmers) maccamp@farm:~/longfin-histories/outputs/202$ ls | grep with_strand > strands
(kmers) maccamp@farm:~/longfin-histories/outputs/202$ ls | grep LFS | grep txt | perl -pe 's/.txt//g' > sampleids 
(kmers) maccamp@farm:~/longfin-histories/outputs/202$ paste strands sampleids > kmers_list_paths.txt 

~/kmers/bin/list_kmers_found_in_multiple_samples  -l kmers_list_paths.txt -k 31 --mac 5 -p 0.2 -o kmers_to_use

this runs for a while...

~/kmers/bin/build_kmers_table -l kmers_list_paths.txt -k 31 -a kmers_to_use -o kmers_table

Runs to completion!

Run k-mers-based GWAS with permutation based-threshold

Format of phenotype file: the phenotype file should be with two columns separated by tabs (\t),
with “accession_id” and “phenotype_value” in the header. The first column lists the name of the
individuals, as in the kmers_table, and the second, phenotype_value, the phenotypic values in
numerical form.

```{r}
m37<-read_csv("meta/m37.csv") %>% mutate(Pheno=ybin+1)
m37 %>% select(SampleID, Pheno) %>% rename("accession_id"="SampleID", "phenotype_value"="Pheno") %>% write_tsv("meta/37.plink.pheno")
```
(kmers) maccamp@farm:~/longfin-histories/outputs/202$ ln -s ~/longfin-histories/meta/37.plink.pheno ./phenotypes.pheno

Looking for kinship table so calculating
~/kmers/bin/emma_kinship_kmers -t kmers_table -k 31 --maf 0.05 > kmers_table.kinship

python ~/kmers/kmers_gwas.py  --pheno phenotypes.pheno --kmers_table kmers_table -l 31 -p 8 --outdir output_dir


Convert to plink k-mers table conversion to PLINK binary format

~/kmers/bin/kmers_table_to_bed -t kmers_table -k 31 -p phenotypes.pheno --maf 0.05 --mac 5 -b 10000000 -o output_fil