---
title: "203-previous-data"
output: html_document
date: "2025-07-04"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message = FALSE)
```

```{r}
library(tidyverse)
library(snpR)
library(adegenet)
library(vcfR)
library(ggrepel)
```
What about the Saglam et al. data?

/home/maccamp/data/forage-fishes/longfin-previous

Putting in data/saglam with symlinks

to-align.txt

```{sh, eval=FALSE}
bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/longfin-chrom/lfs_final_scaff_assembly.fasta
```


get stats...


```{r}
sf<-read_csv("meta/saglam.csv", col_names = c("SampleID","Aligned","Dedup","Cov")) %>% mutate(Path=paste0("data/saglam/",SampleID,".sort.flt.bam"))
sf$Pop<-gsub("_\\d+","",sf$SampleID)
sf
```

```{r}
ggplot(sf) +
  geom_histogram(aes(x=Dedup)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("Count")
```


```{r}
median(sf$Dedup)
```

Can probably downsample some seqs.

In the meantime, pulling some fish from SF Bay. I recall some of these weren't longfin, but which ones?

```{r}
bay<-sf %>% filter(Pop %in% c("SFB", "PETA", "ALVS", "SUIB", "CHPI")) %>% filter(Dedup > 4e5)
bay %>% group_by(Pop) %>% summarize(Count=n(), Median=median(Dedup))
bay %>% write_csv("meta/bay-43.csv")
```

43 samples

```{r}
bay %>% select(Path) %>% write_tsv("bamlists/sf43.bamlist", col_names = FALSE)
bay %>% select(SampleID) %>% write_tsv("bamlists/sf43.names", col_names = FALSE)
```

Quickly check one chrom, lcl|scaffold_1
redoing without
-r 'lcl|scaffold_1' 
```{sh, eval=FALSE}
srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/sf43.bamlist -ref /home/maccamp/genomes/longfin-chrom/lfs_final_scaff_assembly.fasta \
-minInd 33 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/203/sf43-ibs-75 >outputs/203/sf43-ibs-75.out 2> outputs/203/sf43-ibs-75.err &

```
	-> Number of sites retained after filtering: 4583 chrom01
	-> 	-> Number of sites retained after filtering: 109343 genome-wide



	
```{r}
m <- as.matrix(read.table("outputs/203/sf43-ibs-75.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(bay)

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, shape=Pop, fill=Dedup), cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  scale_shape_manual(values=c(23,24,21,22)) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_c(option="viridis") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16)) #+
#  facet_wrap(.~clust_name_100)

pc12
```

Scree plot

```{r}
vardf<-var %>% as_tibble() %>% rename(`Percent Variance`=value) %>% head() %>% mutate(PC=1:n())

scree<-ggplot(vardf, aes(x=PC, y=`Percent Variance`)) +
  geom_line() +
  geom_point() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks = seq(1:6)) +
  theme(axis.text = element_text(size=10)) +
  theme(axis.title = element_text(size=14)) +
  theme(plot.title = element_text(hjust=0, size=16))


scree
```

Interesting, I suppose.

Call snps
```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 38 -bam bamlists/sf43.bamlist -ref /home/maccamp/genomes/longfin-chrom/lfs_final_scaff_assembly.fasta \
-out outputs/203/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/203/snps.out 2> outputs/203/snps.err &

#Number of snps

#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/sf43.names -o plink-renamed.vcf plink.vcf
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
```

Let's see if they show up in the pruned snps.


```{r}
snps<-import.snpR.data(genotypes = "outputs/203/plink-pruned.vcf", 
      sample.meta = bay)
```

```{r}
p <- plot_clusters(snps, facets = c("Pop"), viridis.option = "H")
p$plot$pca
```

```{r, eval=FALSE, echo=FALSE, message=FALSE}
str<-plot_structure(snps, k = c(1,2,3,4))
str$K_plot
```

```{r, eval=FALSE, echo=FALSE, message=FALSE}
str$plot
```


```{r}
vcf<-read.vcfR(file="outputs/203/plink-pruned.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(bay$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(bay)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=SampleID))
pc12

ggsave("outputs/200/lfs-pruned-pca.pdf")
```

```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .01) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)


```

## LD

```{sh, eval=FALSE}
cat outputs/200/long-seqs.tsv  | while read line; do bcftools view -Ov -r $line outputs/203/filtered.vcf.gz > outputs/203/vcfs/$line.vcf; done;

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;
```

```{r, eval=FALSE}
files<-list.files("outputs/203/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/203/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.2)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/203/chrom-ld/",chrom,".pdf"))
}

lapply(files, plotLd)
```


Scaffold 28 is interesting

```{r}
lc<-read.delim("outputs/203/vcfs/lcl|scaffold_28.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.2)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_28 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())
  
ggsave("outputs/203/scaffold28-ld-plot.pdf")
```

Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/203/vcfs",pattern = "*.vcf", full.names = TRUE)

plotPCA<-function(file) {
  meta<-bay
  chrom<-gsub("outputs/203/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$Pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=SampleID))
pc12

ggsave(paste0("outputs/203/chrom-pcs/",chrom,".pdf"))
}

lapply(files, plotPCA)
```


## Pull loadings from chrom28
```{r}
vcf<-read.vcfR(file="outputs/203/vcfs/lcl|scaffold_28.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(bay$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(bay)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=SampleID))
pc12
```

```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .01) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```

lcl|scaffold_28	2160110	lcl|scaffold_28_2160110	G	A	.	.	PR;F_MISSING=0;NS=43;AN=86;AF=0.290698;MAF=0.290698;AC=25;AC_Het=23;AC_Hom=2;AC_Hemi=0;HWE=0.0737253;ExcHet=0.0590054	GT	0/0	0/0	0/0	0/1	0/0	0/0	0/1	0/1	0/1	0/1	0/1 1/1	0/1	0/1	0/1	0/0	0/0	0/1	0/1	0/1	0/0	0/1	0/0	0/1	0/0	0/1	0/1	0/1	0/1	0/10/0	0/0	0/1	0/0	0/1	0/0	0/1	0/0	0/0	0/1	0/0	0/0	0/0

lcl|scaffold_28	1917056	lcl|scaffold_28_1917056	C	A	.	.	PR;F_MISSING=0;NS=43;AN=86;AF=0.27907;MAF=0.27907;AC=24;AC_Het=24;AC_Hom=0;AC_Hemi=0;HWE=0.0183276;ExcHet=0.0108235	GT	0/0	0/0	0/0	0/1	0/0	0/0	0/1	0/1	0/1	0/1	0/1	0/1 0/1	0/1	0/1	0/0	0/0	0/1	0/1	0/1	0/0	0/1	0/0	0/1	0/0	0/1	0/1	0/1	0/1	0/1	0/0 0/0	0/1	0/0	0/1	0/0	0/1	0/0	0/0	0/1	0/0	0/0	0/0

## What about other pops?

Ahw, crap. SFBY

```{r}
sf %>% filter(!Pop %in% c("SFB", "PETA", "ALVS", "SUIB", "CHPI")) %>% filter(Dedup > 4e5) %>% group_by(Pop) %>%
  summarize(Count=n())

```


```{r}
yak<-sf %>% filter(Pop %in% c("YBAK")) %>% filter(Dedup > 4e5) 
yak %>% write_csv("meta/yak.csv")

bay2<-sf %>% filter(Pop %in% c("SFB","SFBY", "PETA", "ALVS", "SUIB", "CHPI")) %>% filter(Dedup > 4e5) 
bay2 %>% write_csv("meta/bay2.csv")
```

Can redo bay2 in 204
