---
title: "205-ybak"
output: html_document
date: "2025-07-05"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message = FALSE)
```

YBAK


```{r}
library(tidyverse)
library(snpR)
library(adegenet)
library(vcfR)
library(ggrepel)
```

```{r}
yak<-read_csv("meta/yak.csv")
```

```{r}
yak %>% select(Path) %>% write_tsv("bamlists/yak.bamlist", col_names = FALSE)
yak %>% select(SampleID) %>% write_tsv("bamlists/yak.names", col_names = FALSE)
```

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 18 -bam bamlists/yak.bamlist -ref /home/maccamp/genomes/longfin-chrom/lfs_final_scaff_assembly.fasta \
-out outputs/205/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/205/snps.out 2> outputs/205/snps.err &

#Number of snps

#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/yak.names -o plink-renamed.vcf plink.vcf
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
```
