---
title: "207-delta-smelt"
output: html_document
date: "2025-07-09"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(snpR)
library(adegenet)
library(vcfR)
library(ggrepel)
```


Need to convert .sorted.bam to fastqs


Using the test80.bamlist in dsm-omics/test80.bamlist, also test80.pheno:
1 /home/maccamp/dsm-omics/data/SOMM516/121922_R1.sort.flt.bam
1 /home/maccamp/dsm-omics/data/SOMM516/401432_R1.sort.flt.bam
1 /home/maccamp/dsm-omics/data/SOMM516/402282_R1.sort.flt.bam
1 /home/maccamp/dsm-omics/data/SOMM516/500742_R1.sort.flt.bam
0 /home/maccamp/dsm-omics/data/SOMM516/130081_R1.sort.flt.bam etc.
0 Sex is based on the sample name.

(base) maccamp@farm:~/dsm-omics/data/fastq$ cat ../../bamlists/test80.bamlist | perl -pe 's/.flt//g'  | while read line; do ln -s $line ./ ; done;
mv fastq to /home/maccamp/longfin-histories/data/fastq

Need to convert to fastqs

```{sh, eval=FALSE}
 ls | grep .bam | perl -pe 's/_R1.sort.bam//g'  > samples.txt
 # this is going to take forever!! Hahahah
 cat samples.txt | while read line; do samtools fastq "$line"_R1.sort.bam -n -1 $line-R1.fastq.gz -2 $line-R2.fastq.gz -0 /dev/null -s /dev/null ; done; # doesn't work because the files were sorted. Sigh.

```


Retrieving from barbera, and that is still extant!
```{sh, eval=FALSE}
~/data/delta-smelt-omics$ scp -r maccampbell@barbera.genomecenter.ucdavis.edu:/share/schreierlab/smelt-wgs/demulti* ./ 
```

Need some meta, several of these fish are sexed based on prefixes. Previously I used an 80 bamlist. 
Symlink to data/dsm subdir those with sex info
e.g. 111392

104 samples


bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa


 have some aligned, lets take a look:
 
```{r}
dm<-read_csv("meta/test-dsm.csv", col_names = c("Sample","Aligned","Dedup","Coverage"))
dm$Sex<-gsub("^\\d\\d\\d\\d\\d","",dm$Sample)
dm$Sex<-as.numeric(dm$Sex)
dm<-dm %>% mutate(Path=paste0("data/dsm/",Sample,".sort.flt.bam"))
```
 
 
```{r}
ggplot(dm) +
  geom_histogram(aes(x=Coverage))
```


```{r}
dm8<-dm %>% filter(Coverage>8) %>% mutate(Pheno=Sex-1)
dm8 %>% group_by(Sex) %>% summarize(Count=n())
```

#Run GWAS across chroms

```{r}
dm8 %>% select(Path) %>% write_tsv("bamlists/dm8.bamlist", col_names = FALSE)
dm8 %>% select(Pheno) %>% write_tsv("bamlists/dm8.pheno", col_names = FALSE)
```

dsm-omics/1300-do-asso.sh is really basic
```{sh, eval=FALSE}
bash $HOME/dsm-omics/301-do-asso.sh $HOME/longfin-histories/bamlists/dm8.bamlist  $HOME/genomes/longfin-polished/1mbseqs.txt $HOME/longfin-histories/bamlists/dm8.pheno
```

Get long seqs
maccamp@farm:~/genomes/longfin-polished$ cat lfs_run3_scaffolds_final.fa.fai | awk '$2 > 1000000 {print;}' | cut -f 1 > 1mbseqs.txt


```