---
title: "207-delta-smelt"
output: html_document
date: "2025-07-09"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(snpR)
library(adegenet)
library(vcfR)
library(RColorBrewer)
library(ggplotify)
library(ggrepel)
```


Need to convert .sorted.bam to fastqs


Using the test80.bamlist in dsm-omics/test80.bamlist, also test80.pheno:
1 /home/maccamp/dsm-omics/data/SOMM516/121922_R1.sort.flt.bam
1 /home/maccamp/dsm-omics/data/SOMM516/401432_R1.sort.flt.bam
1 /home/maccamp/dsm-omics/data/SOMM516/402282_R1.sort.flt.bam
1 /home/maccamp/dsm-omics/data/SOMM516/500742_R1.sort.flt.bam
0 /home/maccamp/dsm-omics/data/SOMM516/130081_R1.sort.flt.bam etc.
0 Sex is based on the sample name.

(base) maccamp@farm:~/dsm-omics/data/fastq$ cat ../../bamlists/test80.bamlist | perl -pe 's/.flt//g'  | while read line; do ln -s $line ./ ; done;
mv fastq to /home/maccamp/longfin-histories/data/fastq

Need to convert to fastqs

```{sh, eval=FALSE}
 ls | grep .bam | perl -pe 's/_R1.sort.bam//g'  > samples.txt
 # this is going to take forever!! Hahahah
 cat samples.txt | while read line; do samtools fastq "$line"_R1.sort.bam -n -1 $line-R1.fastq.gz -2 $line-R2.fastq.gz -0 /dev/null -s /dev/null ; done; # doesn't work because the files were sorted. Sigh.

```


Retrieving from barbera, and that is still extant!
```{sh, eval=FALSE}
~/data/delta-smelt-omics$ scp -r maccampbell@barbera.genomecenter.ucdavis.edu:/share/schreierlab/smelt-wgs/demulti* ./ 
```

Need some meta, several of these fish are sexed based on prefixes. Previously I used an 80 bamlist. 
Symlink to data/dsm subdir those with sex info
e.g. 111392

104 samples


bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa


 have some aligned, lets take a look:
 
```{r}
dm<-read_csv("meta/test-dsm.csv", col_names = c("Sample","Aligned","Dedup","Coverage"))
dm$Sex<-gsub("^\\d\\d\\d\\d\\d","",dm$Sample)
dm$Sex<-as.numeric(dm$Sex)
dm<-dm %>% mutate(Path=paste0("/home/maccamp/longfin-histories/data/dsm/",Sample,".sort.flt.bam"))
```
 
 
```{r}
ggplot(dm) +
  geom_histogram(aes(x=Coverage))
```


```{r}
dm8<-dm %>% filter(Coverage>8) %>% mutate(Pheno=Sex-1)
dm8 %>% group_by(Sex) %>% summarize(Count=n())
```

#Run GWAS across chroms

```{r}
dm8 %>% select(Path) %>% write_tsv("bamlists/dm8.bamlist", col_names = FALSE)
dm8 %>% select(Pheno) %>% write_tsv("bamlists/dm8.pheno", col_names = FALSE)
dm8 %>% select(Sample) %>% write_tsv("bamlists/dm8.names", col_names = FALSE)
dm8 %>% arrange(Sex) %>% select(Sample) %>% write_tsv("bamlists/dm8-sort.names", col_names = FALSE)

```

dsm-omics/1300-do-asso.sh is really basic
editing to deal with lcl| in file names, now in 207.1-do-asso.sh

```{sh, eval=FALSE}
bash $HOME/longfin-histories/207.1-do-asso.sh $HOME/longfin-histories/bamlists/dm8.bamlist  $HOME/genomes/longfin-polished/1mbseqs.txt $HOME/longfin-histories/bamlists/dm8.pheno
```

Get long seqs
maccamp@farm:~/genomes/longfin-polished$ cat lfs_run3_scaffolds_final.fa.fai | awk '$2 > 1000000 {print;}' | cut -f 1 > 1mbseqs.txt


Plot

```{r}
files = list.files(path="outputs/207/", pattern="*.lrt0.gz", full.names = TRUE)
list = lapply(files, read_tsv)
data<-bind_rows(list)
```
```{r}
df <- data %>% mutate(log10p = -log10(dchisq(LRT, df = 1))) %>%
  filter(log10p>=0 & log10p != "Inf") %>%
  mutate(p = dchisq(LRT, df=1)) %>%
  mutate(bonfer = p.adjust(p, method = "bonferroni")) %>% 
  mutate(BHfdr = p.adjust(p, method = "BH")) %>%
  group_by(Chromosome) %>%
  mutate(adjpchrom = p.adjust(p, method = "bonferroni")) %>%
  mutate(BHfdrchrom = p.adjust(p, method = "BH")) %>% ungroup() %>%
  mutate(Index=1:n())

df %>% arrange(-LRT) %>% head(n=20)
```
```{r}
nb.cols <- length(unique(df$Chromosome))
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)
#randomize
mycolors<-sample(mycolors)
```

```{r}
ggplot(df %>% filter(log10p>1)) +
  geom_point(aes(x=Position, y=log10p, color=Chromosome), alpha=0.75, cex=0.5) +
 # geom_hline(yintercept = -log10(0.05/nrow(dddf)), col="black", linetype=2, alpha=0.5) +
#  geom_hline(yintercept= 8, col="black", linetype=1, alpha=0.5) +
  theme_bw() +
#  scale_x_continuous(breaks=chroms$Position, labels=chroms$Chromosome) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=8)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("-log10(p)\n") +
  xlab("\nChromosome") +
  ggtitle("DSM Sex GWAS with LFS Genome") +
  theme(plot.title = element_text(hjust=0.5) ) +
  theme(plot.subtitle = element_text(hjust=0.5)) +
  facet_wrap(.~Chromosome, ncol=5, scales="free_x") +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())

ggsave("outputs/207/dsm-lfs-gwas.jpeg", width=18, height=12)
```

```{r}
ggplot(df %>% filter(log10p>1)) +
  geom_point(aes(x=Index, y=log10p, color=Chromosome), alpha=0.75, cex=0.5) +
 # geom_hline(yintercept = -log10(0.05/nrow(dddf)), col="black", linetype=2, alpha=0.5) +
#  geom_hline(yintercept= 8, col="black", linetype=1, alpha=0.5) +
  theme_bw() +
#  scale_x_continuous(breaks=chroms$Position, labels=chroms$Chromosome) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=8)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("-log10(p)\n") +
  xlab("\nChromosome") +
  ggtitle("DSM Sex GWAS with LFS Genome") +
  theme(plot.title = element_text(hjust=0.5) ) +
  theme(plot.subtitle = element_text(hjust=0.5)) +
 # facet_wrap(.~Chromosome, ncol=5, scales="free_x") +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
ggsave("outputs/207/dsm-lfs-manhattan.jpeg", width=12, height=6)

```

### Call snps across lg13

Region:
```{r}
mins<-df %>% filter(Chromosome=="lcl|scaffold_13") %>% arrange(-LRT) %>% head(n=60) 
max(mins$Position)
min(mins$Position)
mins %>% arrange(LRT)
mins %>% select(Chromosome, Position) %>% arrange(Position) %>% write_tsv(file="meta/topgwas-sites.txt", col_names = FALSE)
```

We would expect the Minor Frequency to about .25 if 50:50, we have 31:23 so:

X chrom 0.79
Y chrom 0.21

Let's call the top20
54 inds 90% ~49

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 49 -bam bamlists/dm8.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-r 'lcl|scaffold_13' -out outputs/207/top-scaffold13-snps -sites outputs/207/topgwas-sites.txt \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/207/snps.out 2> outputs/207/snps.err &
```

```{sh,eval=FALSE}
angsd sites index outputs/207/topgwas-sites.txt
```

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 49 -bam bamlists/dm8.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-r 'lcl|scaffold_13:966304-1681282' -out outputs/207/top-scaffold13-snps-region \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/207/snps-region.out 2> outputs/207/snps-region.err &
```

Make vcf

```{sh, eval=FALSE}
#create vcf
plink --tped top-scaffold13-snps-region.tped --tfam top-scaffold13-snps-region.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/dm8.names -o plink-renamed.vcf plink.vcf

#filter a bit
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.15' | bcftools view -i 'INFO/MAF < 0.4'> filtered.vcf
bcftools view -S ../../bamlists/dm8-sort.names filtered.vcf > sorted-by-names.vcf        
cut -f 2 ../../meta/topgwas-sites.txt| while read line; do grep $line sorted-by-names.vcf; done;
```

Make PCA
```{r}
#vcf<-read.vcfR(file="outputs/207/topsites.vcf")
vcf<-read.vcfR(file="outputs/207/filtered.vcf")

genind<-vcfR2genind(vcf)
genind@pop<-as.factor(dm8$Sex)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(dm8)
df$Sex<-as.factor(df$Sex)
eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Sex), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=Sample), max.overlaps = Inf)
pc12

ggsave("outputs/207/dsm-PCA.jpeg")
```

```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .01) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```
(base) ➜  207 git:(main) ✗ bcftools view -S ../../bamlists/dm8-sort.names filtered.vcf > sorted-by-names.vcf

```{r}

ddf<-read_tsv("outputs/207/top-scaffold13-snps-region.geno.gz", col_names = c("FALSE")) %>% select(-X57)
#replace with most commmon entry?
#replace with NA for now
ddf[ddf==-1]<-NA
ddf2<-ddf %>% filter(X2 %in% head(mins$Position, 15))
m<-ddf2 %>% select(-`FALSE`, -X2) %>% as.matrix()

annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Sex<-dm8$Sex
annot$Sex<-gsub("1","Female",annot$Sex)
annot$Sex<-gsub("2","Male",annot$Sex)
```

```{r}

as.ggplot(pheatmap(m, labels_col = dm8$Sample, cluster_rows=TRUE, annotation_col = annot, width=12, height=8, cluster_cols = TRUE, legend=TRUE, labels_row = head(mins$Position, 15),
         fontsize_col = 6)) + ggtitle("Delta Smelt Variants on Longfin scaffold_13\n") +
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/207/dsm-sex-heatmap-subset.pdf", width=12, height=6)
```
## Model2

```{r}
files = list.files(path="outputs/207/model2/", pattern="*.lrt0.gz", full.names = TRUE)
list = lapply(files, read_tsv)
data<-bind_rows(list)
df <- data %>% mutate(log10p = -log10(dchisq(LRT, df = 1))) %>%
  filter(log10p>=0 & log10p != "Inf") %>%
  mutate(p = dchisq(LRT, df=1)) %>%
  mutate(bonfer = p.adjust(p, method = "bonferroni")) %>% 
  mutate(BHfdr = p.adjust(p, method = "BH")) %>%
  group_by(Chromosome) %>%
  mutate(adjpchrom = p.adjust(p, method = "bonferroni")) %>%
  mutate(BHfdrchrom = p.adjust(p, method = "BH")) %>% ungroup() %>%
  mutate(Index=1:n())

df %>% arrange(-LRT) %>% head(n=20)
```

```{r}
ggplot(df %>% filter(log10p>1)) +
  geom_point(aes(x=Position, y=log10p, color=Chromosome), alpha=0.75, cex=0.5) +
 # geom_hline(yintercept = -log10(0.05/nrow(dddf)), col="black", linetype=2, alpha=0.5) +
#  geom_hline(yintercept= 8, col="black", linetype=1, alpha=0.5) +
  theme_bw() +
#  scale_x_continuous(breaks=chroms$Position, labels=chroms$Chromosome) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=8)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("-log10(p)\n") +
  xlab("\nChromosome") +
  ggtitle("DSM Sex GWAS with LFS Genome") +
  theme(plot.title = element_text(hjust=0.5) ) +
  theme(plot.subtitle = element_text(hjust=0.5)) +
  facet_wrap(.~Chromosome, ncol=5, scales="free_x") +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())

ggsave("outputs/207/dsm-lfs-gwas-model2.jpeg", width=18, height=12)
```

```{r}
ggplot(df %>% filter(log10p>1)) +
  geom_point(aes(x=Index, y=log10p, color=Chromosome), alpha=0.75, cex=0.5) +
 # geom_hline(yintercept = -log10(0.05/nrow(dddf)), col="black", linetype=2, alpha=0.5) +
#  geom_hline(yintercept= 8, col="black", linetype=1, alpha=0.5) +
  theme_bw() +
#  scale_x_continuous(breaks=chroms$Position, labels=chroms$Chromosome) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=8)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("-log10(p)\n") +
  xlab("\nChromosome") +
  ggtitle("DSM Sex GWAS with LFS Genome") +
  theme(plot.title = element_text(hjust=0.5) ) +
  theme(plot.subtitle = element_text(hjust=0.5)) +
 # facet_wrap(.~Chromosome, ncol=5, scales="free_x") +
  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
ggsave("outputs/207/dsm-lfs-manhattan-model2.jpeg", width=12, height=6)

```

