---
title: "208-eulachon"
output: html_document
date: "2025-07-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

https://www.ncbi.nlm.nih.gov/sra?linkname=bioproject_sra_all&from_uid=274282

Some sequencing is single ended some is paired end

Design: Restriction site associated (RAD) sequencing was done on a sample of 494 individual Pacific Eulachon. Sbf1 was used as the restriction enzyme. This library was paired end sequenced in order to create scaffolds of interesting RAD loci. The individual samples contained in this library are the same as those in DFO_Tpa_L0124.fastq. 28.4G


Run	# of Spots	# of Bases	Size	Published
SRR1786609	156,074,772	15.8G	10.5Gb	2016-02-03
SRR1786610	213,923,577	21.6G	14.4Gb	2016-02-03
SRR1786611	219,066,721	22.1G	14.8Gb	2016-02-03
SRR1786612	189,237,566	19.1G	12.1Gb	2016-02-03
SRR1786613	171,647,080	17.3G	10.8Gb	2016-02-03
SRR1786614	168,511,175	17G	10.6Gb	2016-02-03
SRR1786615	187,543,719	18.9G	12.1Gb	2016-02-03
SRR1786617	201,444,434	20.3G	13.4Gb	2016-02-03
SRR1786618	194,039,214	19.6G	12.6Gb	2016-02-03

wget https://sra-downloadb.be-md.ncbi.nlm.nih.gov/sos5/sra-pub-zq-11/SRR001/786/SRR1786609/SRR1786609.sralite.1 
Will download and split

module load sratoolkit

fastq-dump --outdir ./ --gzip --skip-technical --read-filter pass --clip SRR1786609.sralite.1   

Check barcodes and compare to ~/scripts
BarcodeList.txt
BarcodeListRev.txt
BarcodeSplit.pl

Potential barcodes in BarcodeList.txt
AAACGG,AACGTT,AACTGA

At the start of reads:
ATTCCGTGCAGG
ATTCCG

TAATTC
TAATTCTGCAGG

These look to be in our BarcodeList.txt file...
trying to run like so in ~/longfin-histories/data/eul     

```{sh, eval=FALSE}
./BarcodeSplit.pl SRR1786609.sralite.1_pass.fastq.gz BarcodeList.txt 
```

Make barcodes:
grep "^\w{6}TGCAGG"

gunzip -c SRR1786609.sralite.1_pass.fastq.gz | grep "^\w\w\w\w\w\wTGCAGG" | cut -c1-12 | sort | uniq -c > bs.txt

fastq-multx should work with inline barcodes
fastq-multx -B barcodes.tsv SRR1786609.sralite.1_pass.fastq.gz -o %.fq.gz

Sample1 TGCCCGTGCAGG
Sample2 AACGTTTGCAGG

DO NOT put a newline at the end of this

dummy trial
```{sh}
~/fastq-multx/fastq-multx -B barcodes.tsv SRR1786609.sralite.1_pass.fastq.gz -o %.fq.gz
```

Does seem to work, and clips out the barcodes,
Sample 1 TGCCCGTGCAGG

@SRR1786609.sralite.1.1519132 BZH9VV1:53:D1FAPACXX:1:1102:13311:70218 length=101
CAAACCTCCTGCGCTGGTACTCATTCATATCTATCACCTGGAGAGAGACAGGGTCATATCTATCACCTGGAGAGAGACAGGGTCATATC
+SRR1786609.sralite.1.1519132 BZH9VV1:53:D1FAPACXX:1:1102:13311:70218 length=101
?????????????????????????????????????????????????????????????????????????????????????????


@SRR1786609.sralite.1.1519132 BZH9VV1:53:D1FAPACXX:1:1102:13311:70218 length=101
TGCCCGTGCAGGCAAACCTCCTGCGCTGGTACTCATTCATATCTATCACCTGGAGAGAGACAGGGTCATATCTATCACCTGGAGAGAGACAGGGTCATATC
+SRR1786609.sralite.1.1519132 BZH9VV1:53:D1FAPACXX:1:1102:13311:70218 length=101
?????????????????????????????????????????????????????????????????????????????????????????????????????

What is the encoding here?

Getting the top hits:
 451216 ACTGGCTGCAGG
 462997 CACCTCTGCAGG
 469269 GGGGCGTGCAGG
 480977 ACTCTTTGCAGG
 486111 CAAAAATGCAGG
 493721 TACACATGCAGG
 497850 CAGGCATGCAGG
 516012 CGCAGATGCAGG
 526687 TACGGGTGCAGG
 534373 AACTGATGCAGG
 535241 CAATCGTGCAGG
 537453 GAACCATGCAGG
 541918 AGCGCATGCAGG
 548433 GACGACTGCAGG
 555380 ATTCCGTGCAGG
 558056 AGGTGTTGCAGG
 560981 CGGTCCTGCAGG
 567597 AGGGTCTGCAGG
 567599 GATCGTTGCAGG
 572025 CGCGTGTGCAGG
 572229 GGTACATGCAGG
 589308 AGTAGGTGCAGG
 591980 AACGTTTGCAGG
 595056 AGCCATTGCAGG
 596006 ATCAAATGCAGG
 598240 AATATCTGCAGG
 605265 CGTGATTGCAGG
 606219 CCCGGTTGCAGG
 609201 GGGCGCTGCAGG
 611698 ACAAGATGCAGG
 614229 CCATTTTGCAGG
 617672 GAAATGTGCAGG
 617829 TCCTGCTGCAGG
 626683 GCCGTATGCAGG
 635629 CGAGGCTGCAGG
 639756 TCGATTTGCAGG
 645367 GGCAAGTGCAGG
 660971 CCCTAATGCAGG
 661373 GCAGATTGCAGG
 663281 TGAACCTGCAGG
 663489 CTGCGATGCAGG
 665646 TAATTCTGCAGG
 671786 TTCTAGTGCAGG
 673093 CATACTTGCAGG
 676030 GAGAGATGCAGG
 677416 AAGCTATGCAGG
 685562 GACTCTTGCAGG
 688247 CGTCTATGCAGG
 690491 ACATACTGCAGG
 698996 TGACAATGCAGG
 700049 ATAGTATGCAGG
 702150 GTCATCTGCAGG
 703280 GTATCCTGCAGG
 710724 TGGGGATGCAGG
 715628 AGTTAATGCAGG
 722666 TAGTATTGCAGG
 723033 TTCCGTTGCAGG
 727590 CTCGCCTGCAGG
 740233 TCTTCTTGCAGG
 741536 TATCACTGCAGG
 760282 GCATGGTGCAGG
 762408 ACAGCGTGCAGG
 769168 AATGAGTGCAGG
 769391 GTAAGTTGCAGG
 770368 TCAAAGTGCAGG
 791617 TGCCCGTGCAGG
 802357 TAAGCTTGCAGG
 804467 GGACTTTGCAGG
 828896 GGTTTGTGCAGG
 838381 TTATGATGCAGG
 838793 CTACAGTGCAGG
 855944 AAGACGTGCAGG
 934744 TGCTTATGCAGG
 
 Created 206 barcodes file
 
 206.barcodes, now we can try this thing
 
```{sh}
srun -t 8:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 ~/fastq-multx/fastq-multx -B 206.barcodes SRR1786609.sralite.1_pass.fastq.gz -o %.fq.gz
```

About 6 GB of unamatched FQ

Expected 934744 reads and had 877895 reads.

../../208-do-align-single.sh samples.list /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa

```{r}
eul<-read_tsv("206.barcodes", col_names = c("SampleID","Barcode")) %>% mutate(Path=paste0("data/eul/",SampleID,".sort.bam")) %>% mutate(Pop="Unknown")
eul %>% select(Path) %>% write_tsv("bamlists/eul73.bamlist",col_names = FALSE)
eul %>% select(SampleID) %>% write_tsv("bamlists/eul73.names",col_names = FALSE)
```

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 66 -bam bamlists/eul73.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-out outputs/208/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/208/snps.out 2> outputs/208/snps.err &
```


## Copy from here and change as necessary


```{sh, eval=FALSE}
#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/eu73.names -o plink-renamed.vcf plink.vcf
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
```


Basic PCA

```{r}
snps<-import.snpR.data(genotypes = "outputs/208/plink-pruned.vcf", 
      sample.meta = yak)
```

```{r}
p <- plot_clusters(snps, facets = c("Pop"), viridis.option = "H")
p$plot$pca
```




## LD

```{sh, eval=FALSE}
#long seqs are > 884174 bp
cat outputs/206/long-seqs.tsv  | while read line; do bcftools view -Ov -r $line outputs/208/filtered.vcf.gz > outputs/208/vcfs/$line.vcf; done;

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;

```

```{r, eval=FALSE}
files<-list.files("outputs/208/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/208/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.5)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/208/chrom-ld/",chrom,".pdf"))
}

lapply(files, plotLd)
```


Scaffold 26 is interesting

```{r}
lc<-read.delim("outputs/208/vcfs/lcl|scaffold_26.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.2)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank()) 
  
ggsave("outputs/208/scaffold26-ld-plot.pdf")
```

````{r}
ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlim(8.25e6, max(lc$BP_A))
```

```{sh, eval=FALSE}
bcftools view -Ov -r 'lcl|scaffold_26:8250000-' outputs/208/filtered.vcf.gz > outputs/208/lcl26-sub.vcf
```


Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/208/vcfs",pattern = "*.vcf", full.names = TRUE)

plotPCA<-function(file) {
  meta<-yak
  chrom<-gsub("outputs/208/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$Pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=SampleID))
pc12

ggsave(paste0("outputs/208/chrom-pcs/",chrom,".pdf"))
}

lapply(files, plotPCA)
```


## Check loadings
```{r}
vcf<-read.vcfR(file="outputs/208/lcl26-sub.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(yak$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(yak)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=SampleID))
pc12
```

```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .01) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```