---
title: "209-eul-tpac2"
output: html_document
date: "2025-07-11"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(tidyverse)
library(snpR)
library(adegenet)
library(vcfR)
library(ggrepel)
```


Using the Tpac2 reference to look for interesting things. 

/home/maccamp/genomes/tpac2/GCA_023658055.1_Tpac_2.0_genomic.fna

data in data/eul-tpac2
chromlist is /home/maccamp/genomes/tpac2/chroms.txt

running align

../../208-do-align-single.sh samples.list /home/maccamp/genomes/tpac2/GCA_023658055.1_Tpac_2.0_genomic.fna

Use stats to create meta...

Have about 95% alignment success (over 91% or so with LFS)

```{r}
eul<-read_csv("outputs/209/eul.csv", col_names = c("SampleID","Aligned")) %>% mutate(Path=paste0("data/eul-tpac2/",SampleID,".sort.bam")) %>% mutate(Pop="Unknown")
eul %>% select(Path) %>% write_tsv("bamlists/eul73-tpac2.bamlist",col_names = FALSE)
eul %>% select(SampleID) %>% write_tsv("bamlists/eul73-tpac2.names",col_names = FALSE)
```

```{r}
ggplot(eul) +
  geom_histogram(aes(x=Aligned)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("Count")
```

Note to self to include a region file here for larger chroms.

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 66 -bam bamlists/eul73-tpac2.bamlist -ref /home/maccamp/genomes/tpac2/GCA_023658055.1_Tpac_2.0_genomic.fna \
-rf /home/maccamp/genomes/tpac2/chroms.txt -out outputs/209/snps-wgs  \
-minMaf 0.05 -minMapQ 10 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/209/snps.out 2> outputs/209/snps.err &
```

	-> Number of sites retained after filtering: 17944 

```{sh, eval=FALSE}
#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/eul73.names -o plink-renamed.vcf plink.vcf
#changing missing threshold from .10 to .20
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
```

Basic PCA

```{r}
snps<-import.snpR.data(genotypes = "outputs/209/plink-pruned.vcf", 
      sample.meta = eul)
```


```{r}
p <- plot_clusters(snps, facets = c("Pop"), viridis.option = "H")
p$plot$pca
```

Another way:

```{r}
meta<-eul
vcf<-read.vcfR(file="outputs/209/plink-pruned.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(meta$Pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=SampleID))
pc12
```



## LD

```{sh, eval=FALSE}
cat outputs/209/chroms.txt  | while read line; do bcftools view -Ov -r $line outputs/209/filtered.vcf.gz > outputs/209/vcfs/$line.vcf; done;

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;

```

```{r, eval=FALSE}
files<-list.files("outputs/209/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/209/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.2)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/209/chrom-ld/",chrom,".pdf"))
}

lapply(files, plotLd)
```



```{r}
lc<-read.delim("outputs/209/vcfs/CM042872.1.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.5)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("CM042872 LD")+
  theme_bw() +
  theme(panel.grid = element_blank()) 
  
```

````{r}
ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("CM042872.1 LD")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlim(9e6, max(lc$BP_A))
```

```{sh, eval=FALSE}
bcftools view -Ov -r 'CM042872.1:9000000-' outputs/209/filtered.vcf.gz > outputs/209/CM042872.1-sub.vcf
```


Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/209/vcfs",pattern = "*.vcf", full.names = TRUE)

plotPCA<-function(file) {
  meta<-eul
  chrom<-gsub("outputs/209/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$Pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
 # geom_text_repel(aes(x=Axis1, y=Axis2,label=SampleID))
pc12

ggsave(paste0("outputs/209/chrom-pcs/",chrom,".pdf"))
}

lapply(files, plotPCA)
```


## Check loadings
```{r}
vcf<-read.vcfR(file="outputs/209/CM042872.1-sub.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(eul$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(eul)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
#  geom_text_repel(aes(x=Axis1, y=Axis2,label=SampleID))
pc12
```

```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .1) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```


Local PCA

Need bcf files

```{sh, eval=FALSE}

cat outputs/209/chroms.txt  | while read line; do bcftools view -Ob -r $line outputs/209/filtered.vcf.gz > outputs/209/bcf/$line.bcf; done;

for f in outputs/209/bcf/*.bcf; do bcftools index $f; done;
```

need a file like so:

```{r}
samples<-eul %>% select(SampleID) %>% rename(ID=SampleID)

population<-eul %>% select(Pop) %>% rename(population=Pop)

table<-cbind(samples, population)
write.table(table, "outputs/209/bcf/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```

Run lostruct in `/Users/mac/github/local_pca/templated`
`./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/209/bcf -t snp -s 10 -m 4 -I /Users/mac/github/longfin-histories/outputs/209/bcf/sample_info.tsv -j 209`    
`cp lostruct_results/type_snp_size_10_weights_none_jobid_209/mds_coords.csv ~/github/longfin-histories/outputs/209 `

`./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/209/bcf -t snp -s 15 -m 4 -I /Users/mac/github/longfin-histories/outputs/209/bcf/sample_info.tsv -j 210`    
`cp lostruct_results/type_snp_size_15_weights_none_jobid_210/mds_coords.csv ~/github/longfin-histories/outputs/209/mds_coords_15.csv `

```{r}
mds<-read_csv("outputs/209/mds_coords.csv") # 10 snp windows
mds<-read_csv("outputs/209/mds_coords_15.csv")
#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```

     
     
     
```{r}
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```
     
Plotting outliers by chrom.

### MDS1

```{r}
p1<-MDS1 %>% mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
```


```{r}
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```

### MDS2
```{r}
p2<-MDS2 %>% mutate(Index=1:n())

out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
```
```{r}
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```

### MDS3     

```{r}
p3<-MDS3 %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
```
```{r}
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)

```

```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


#Hmm.. I should be able to include a bunch of rectangles to offset chroms
ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```

Locations:

```{r}
MDS1 %>% arrange(-Value1)
MDS2 %>% arrange(-Value2)

```