---
title: "212-subtraction"
output: html_document
date: "2025-10-13"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
```


```{r}
fish<-read_tsv("outputs/210/m80-ordered.bamlist", col_names = c("Path"))
fish$Sample<-gsub("/home/maccamp/dsm-omics/data/SOMM51\\d/","",fish$Path)
fish$Sample<-gsub("_R1.sort.flt.bam","",fish$Sample)
fish$Sex<-gsub("^\\d\\d\\d\\d\\d","",fish$Sample)
fish$Sex<-as.numeric(fish$Sex)
fish<-fish %>% relocate(Sample, Sex, Path)
fish %>% select(Sample) %>% write_tsv("bamlists/test80.names", col_names = FALSE)
fish %>% select(Path) %>% write_tsv("bamlists/test80.bamlist", col_names = FALSE)
fish
```


Let's look at unaligned and male specific sequence

annot$Sex<-gsub("1","Female",annot$Sex)
annot$Sex<-gsub("2","Male",annot$Sex)

https://pmc.ncbi.nlm.nih.gov/articles/PMC11481317/

_1_ Identify the highest coverage 10 females and 10 males (and omit ones Mac finds suspicious)
_2_ Extract unaligned sequences from males/females
_3_ De novo asssemble unaligned male data, SPADES (?)
_4_ Realign unaligned male/female data to de denovo assembly
_5_ Separate male specific contigs as candidate sex markers
_6_ Use other sequences to verify male specificity (not from the original 10 highest coverage)

samtools flagstat 121922_R1.sort.bam 87140918 + 0 mapped (92.27% : N/A)
samtools flagstat 401432_R1.sort.bam 80013837 + 0 mapped (93.39% : N/A)

samtools flagstat 110611_R1.sort.bam 56403988 + 0 mapped (93.22% : N/A)
samtools flagstat 110381_R1.sort.bam 79366019 + 0 mapped (92.58% : N/A)


New curated genome in 
(base) maccamp@farm:~/genomes/hypomesus-curated$

Checking busco 
conda activate busco

srun -p high -t 6:00:00 --mem=32GB --nodes=1 --cpus-per-task=12 busco -i fHypTra1.pri.cur.20251017.fasta -o busco -m genome -l actinopterygii_odb12 --cpu 12 --skip_bbtools

 -------------------------------------------------------------------------------------------
    |Results from dataset actinopterygii_odb12                                                 |
    -------------------------------------------------------------------------------------------
    |C:96.4%[S:95.5%,D:0.9%],F:1.2%,M:2.4%,n:7207,E:5.2%                                       |
    |6949    Complete BUSCOs (C)    (of which 359 contain internal stop codons)                |
    |6886    Complete and single-copy BUSCOs (S)                                               |
    |63    Complete and duplicated BUSCOs (D)                                                  |
    |88    Fragmented BUSCOs (F)                                                               |
    |170    Missing BUSCOs (M)                                                                 |
    |7207    Total BUSCO groups searched                                                       |
    -------------------------------------------------------------------------------------------
20


We can link fish data in ./data/sub
```{r}
fish$Link<-gsub("1.sort.flt.bam","",fish$Path) 
fish %>% select(Link) %>% write_tsv("outputs/212/link.tsv")
#linking in all files
```

Set up names to work with 101.2-do-align.sh

bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta


We'll compare our alignment success rate to these ones

samtools flagstat 121922.sort.bam 91682741 + 0 mapped (97.01% : N/A)
samtools flagstat 401432.sort.bam 83569992 + 0 mapped (97.45% : N/A)
samtools flagstat 110611.sort.bam 58969314 + 0 mapped (97.41% : N/A)
samtools flagstat 110381.sort.bam 83330826 + 0 mapped (97.15% : N/A)

Then we can rerun our sex gwas
'first two digits is a generation number, the three in the middle is a cross number, and the last number is for male/ female (1 or 2)."
```{r}
m80<-fish %>% select(-Path, -Link) %>% mutate(Pheno=Sex-1) %>% mutate(SexChar=ifelse(Sex==2, "Female","Male")) %>% mutate(Path=paste0("/home/maccamp/longfin-histories/data/sub/",Sample,".sort.flt.bam"))
m80 %>% select(Pheno) %>% write_tsv("bamlists/80.pheno", col_names = FALSE)
m80 %>% select(Path) %>% write_tsv("bamlists/80.bamlist", col_names = FALSE)

#bind on data
dat<- read_csv("outputs/212/curated.dat", col_names = c("Sample","Sort","Dedup","Cov"))
dat$Sample<-as.character(dat$Sample)
nm<-left_join(m80,dat)
topc<-nm %>% group_by(SexChar) %>% top_n(10, Cov)

```

no -rf -rf meta/lates-lgs.txt
```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12 \
 -doAsso 1 -yBin $HOME/longfin-histories/bamlists/80.pheno -GL 1 -minInd 76 -minMaf 0.10 \
 -minMapQ 20 -minQ 20 \
 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -out outputs/212/asso-95 \
 -bam $HOME/longfin-histories/bamlists/80.bamlist  > outputs/212/asso.out 2> outputs/212/asso.err &
```

```{r}
files = list.files(path="outputs/212/", pattern="*.lrt0.gz", full.names = TRUE)
list = lapply(files, read_tsv)
data<-bind_rows(list)
```

```{r}
df <- data %>% mutate(log10p = -log10(dchisq(LRT, df = 1))) %>%
  filter(log10p>=0 & log10p != "Inf") %>%
  mutate(p = dchisq(LRT, df=1)) %>%
  mutate(bonfer = p.adjust(p, method = "bonferroni")) %>% 
  mutate(BHfdr = p.adjust(p, method = "BH")) %>%
  group_by(Chromosome) %>%
  mutate(adjpchrom = p.adjust(p, method = "bonferroni")) %>%
  mutate(BHfdrchrom = p.adjust(p, method = "BH")) %>% ungroup() %>%
  mutate(Index=1:n())

df %>% arrange(-LRT) %>% head(n=20)
```



_1_ Identify the highest coverage 10 females and 10 males (and omit ones Mac finds suspicious)
These may be ones that don't track with markers id'd by gwas
_2_ Extract unaligned sequences from males/females

(can try chrom by chrom pc)

samtools view -b -f 4 file.bam > unmapped.bam
samtools view -b -f 8 file.bam > unmapped.bam # both read1 and read2 unmapped
using the sort.bam files

```{r}
topc %>% ungroup() %>% mutate(Command=paste0("samtools view -b -f 12 /home/maccamp/longfin-histories/data/sub/",Sample,".sort.bam > ",
                              "/home/maccamp/longfin-histories/data/unmapped/",Sample,".bam; samtools collate -u -O /home/maccamp/longfin-histories/data/unmapped/",Sample,".bam | samtools fastq -1 /home/maccamp/longfin-histories/data/unmapped/",Sample,"-r1.fastq -2 /home/maccamp/longfin-histories/data/unmapped/",Sample,"-r2.fastq")) %>%
  select(Command) %>% write_tsv("212.1-unmapped-commands.tsv", col_names = FALSE)
```

module load parallel
srun -p high –time 4:00:00 –nodes=1 –ntasks-per-node=1 –cpus-per-task=11 parallel -j 11 < 212.1-unmapped-commands.tsv



samtools flagstat 110381.bam (with -f 12)
1889938 + 0 in total 
1889938 + 0 paired in sequencing
944969 + 0 read1
944969 + 0 read2

samtools collate -u -O 110381.bam | samtools fastq -1 110381-r1.fastq -2 110381-r2.fastq

check for read through?
flash 110381-r1.fastq 110381-r2.fastq -o 110381-flash --max-overlap 120  --min-overlap 20

_3_ De novo asssemble unaligned male data, SPADES

erro   1:16:02.922  3668M / 5153M ERROR   General                 (hammer_tools.cpp          : 194)   Pair of read files "/home/maccamp/longfin-histories>

when back and redid with -f 12
[M::bam2fq_mainloop] discarded 0 singletons
[M::bam2fq_mainloop] processed 1889938 reads

Can cat together several individual fastq files then run something like this:

conda activate spades
```{r, eval=FALSE}
#trialing indonesian sample
ind %>% filter(Lineage=="uwisara") %>% mutate(Command=paste0("srun -p bigmemm -t 168:00:00 --mem=520GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/uwisara/",Run,"-R1.fastq.gz -2 data/uwisara/",Run,"-R2.fastq.gz -o data/denovo/",Run," > data/denovo/",Run,".out 2> data/denovo/",Run,".err &")) %>% select(Command) %>%
  write_tsv("1301.2-spades.txt")
```


```{sh, eval=FALSE}
srun -p bigmemm -t 48:00:00 --mem=256GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/unmapped/110381-r1.fastq -2 data/unmapped/110381-r2.fastq -o data/unmapped/110381 > data/unmapped/110381.out 2> data/unmapped/110381.err & 

#Spades with unpaired
srun -p bigmemm -t 48:00:00 --mem=256GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/unmapped/110381-flash.notCombined_1.fastq -2 data/unmapped/110381-flash.notCombined_2.fastq --merged data/unmapped/110381-flash.extendedFrags.fastq -o data/unmapped/110381-flash > data/unmapped/110381-flash.out 2> data/unmapped/110381-flash.err & 
````


Pretty similar assemblies
Comparing spades flashed/unflashed scaffolds
(base) maccamp@farm:~/longfin-histories/data/unmapped/110381$ assembledSeqStats.pl scaffolds.fasta 
Number of Contigs	29419
Total Bases	22325036
No. of Contigs >=1000	6685
Max Contig Length	15845

(base) maccamp@farm:~/longfin-histories/data/unmapped/110381-flash$ assembledSeqStats.pl scaffolds.fasta 
Number of Contigs	22714
Total Bases	20086919
No. of Contigs >=1000	6510
Max Contig Length	15845

For multiple fish, should I specify different libraries, or cat together reads?


I like the plan of assembly of 2 x 5 different males and 2 x 5 females and checking for concordance.


A

mosdepth, installing with bioconda

conda create -n mosdepth
mamba install mosdepth