---
title: "212-subtraction"
output: html_document
date: "2025-10-13"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(pheatmap)
library(ggplotify)
```


```{r}
fish<-read_tsv("outputs/210/m80-ordered.bamlist", col_names = c("Path"))
fish$Sample<-gsub("/home/maccamp/dsm-omics/data/SOMM51\\d/","",fish$Path)
fish$Sample<-gsub("_R1.sort.flt.bam","",fish$Sample)
fish$Sex<-gsub("^\\d\\d\\d\\d\\d","",fish$Sample)
fish$Sex<-as.numeric(fish$Sex)
fish<-fish %>% relocate(Sample, Sex, Path)
fish %>% select(Sample) %>% write_tsv("bamlists/test80.names", col_names = FALSE)
fish %>% select(Path) %>% write_tsv("bamlists/test80.bamlist", col_names = FALSE)
fish
```


Let's look at unaligned and male specific sequence

annot$Sex<-gsub("1","Female",annot$Sex)
annot$Sex<-gsub("2","Male",annot$Sex)

https://pmc.ncbi.nlm.nih.gov/articles/PMC11481317/

_1_ Identify the highest coverage 10 females and 10 males (and omit ones Mac finds suspicious)
_2_ Extract unaligned sequences from males/females
_3_ De novo asssemble unaligned male data, SPADES (?)
_4_ Realign unaligned male/female data to de denovo assembly
_5_ Separate male specific contigs as candidate sex markers
_6_ Use other sequences to verify male specificity (not from the original 10 highest coverage)

samtools flagstat 121922_R1.sort.bam 87140918 + 0 mapped (92.27% : N/A)
samtools flagstat 401432_R1.sort.bam 80013837 + 0 mapped (93.39% : N/A)

samtools flagstat 110611_R1.sort.bam 56403988 + 0 mapped (93.22% : N/A)
samtools flagstat 110381_R1.sort.bam 79366019 + 0 mapped (92.58% : N/A)


New curated genome in 
(base) maccamp@farm:~/genomes/hypomesus-curated$

Checking busco 
conda activate busco

srun -p high -t 6:00:00 --mem=32GB --nodes=1 --cpus-per-task=12 busco -i fHypTra1.pri.cur.20251017.fasta -o busco -m genome -l actinopterygii_odb12 --cpu 12 --skip_bbtools

 -------------------------------------------------------------------------------------------
    |Results from dataset actinopterygii_odb12                                                 |
    -------------------------------------------------------------------------------------------
    |C:96.4%[S:95.5%,D:0.9%],F:1.2%,M:2.4%,n:7207,E:5.2%                                       |
    |6949    Complete BUSCOs (C)    (of which 359 contain internal stop codons)                |
    |6886    Complete and single-copy BUSCOs (S)                                               |
    |63    Complete and duplicated BUSCOs (D)                                                  |
    |88    Fragmented BUSCOs (F)                                                               |
    |170    Missing BUSCOs (M)                                                                 |
    |7207    Total BUSCO groups searched                                                       |
    -------------------------------------------------------------------------------------------
20


We can link fish data in ./data/sub
```{r}
fish$Link<-gsub("1.sort.flt.bam","",fish$Path) 
fish %>% select(Link) %>% write_tsv("outputs/212/link.tsv")
#linking in all files
```

Set up names to work with 101.2-do-align.sh

bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta


We'll compare our alignment success rate to these ones

samtools flagstat 121922.sort.bam 91682741 + 0 mapped (97.01% : N/A)
samtools flagstat 401432.sort.bam 83569992 + 0 mapped (97.45% : N/A)
samtools flagstat 110611.sort.bam 58969314 + 0 mapped (97.41% : N/A)
samtools flagstat 110381.sort.bam 83330826 + 0 mapped (97.15% : N/A)

Then we can rerun our sex gwas
'first two digits is a generation number, the three in the middle is a cross number, and the last number is for male/ female (1 or 2)."
```{r}
m80<-fish %>% select(-Path, -Link) %>% mutate(Pheno=Sex-1) %>% mutate(SexChar=ifelse(Sex==2, "Male","Female")) %>% mutate(Path=paste0("/home/maccamp/longfin-histories/data/sub/",Sample,".sort.flt.bam"))
m80 %>% select(Pheno) %>% write_tsv("bamlists/80.pheno", col_names = FALSE)
m80 %>% select(Path) %>% write_tsv("bamlists/80.bamlist", col_names = FALSE)
m80 %>% select(Sample) %>% write_tsv("bamlists/80.names", col_names=FALSE)
m80 %>% filter(SexChar=="Female") %>% select(Sample) %>% write_tsv("bamlists/80-females.names", col_names=FALSE)
m80 %>% filter(SexChar=="Male") %>% select(Sample) %>% write_tsv("bamlists/80-males.names", col_names=FALSE)


#bind on data
dat<- read_csv("outputs/212/curated.dat", col_names = c("Sample","Sort","Dedup","Cov"))
dat$Sample<-as.character(dat$Sample)
nm<-left_join(m80,dat)
topc<-nm %>% group_by(SexChar) %>% top_n(10, Cov)

```

no -rf -rf meta/lates-lgs.txt
```{sh, eval=FALSE}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12 \
 -doAsso 1 -yBin $HOME/longfin-histories/bamlists/80.pheno -GL 1 -minInd 76 -minMaf 0.10 \
 -minMapQ 20 -minQ 20 \
 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -out outputs/212/asso-95 \
 -bam $HOME/longfin-histories/bamlists/80.bamlist  > outputs/212/asso.out 2> outputs/212/asso.err &
 
 srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12 \
 -doAsso 1 -yBin $HOME/longfin-histories/bamlists/80.pheno -GL 1 -minInd 60 -minMaf 0.10 \
 -minMapQ 20 -minQ 20 \
 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -out outputs/212/asso-75 \
 -bam $HOME/longfin-histories/bamlists/80.bamlist  > outputs/212/asso-75.out 2> outputs/212/asso-75.err &
```

```{r}
files = list.files(path="outputs/212/", pattern="*.lrt0.gz", full.names = TRUE)
list = lapply(files, read_tsv)
data<-bind_rows(list)
```

```{r}
df <- data %>% mutate(log10p = -log10(dchisq(LRT, df = 1))) %>%
  filter(log10p>=0 & log10p != "Inf") %>%
  mutate(p = dchisq(LRT, df=1)) %>%
  mutate(bonfer = p.adjust(p, method = "bonferroni")) %>% 
  mutate(BHfdr = p.adjust(p, method = "BH")) %>%
  group_by(Chromosome) %>%
  mutate(adjpchrom = p.adjust(p, method = "bonferroni")) %>%
  mutate(BHfdrchrom = p.adjust(p, method = "BH")) %>% ungroup() %>%
  mutate(Index=1:n())

df %>% arrange(-LRT) %>% head(n=20)
```

75%
	-> Number of sites retained after filtering: 1242834 
95%
	-> Number of sites retained after filtering: 1136385 


```{r}
ggplot(df %>% filter(Chromosome %in% c("SUPER_1","SUPER_10","SUPER_11"))) +
  geom_point(aes(x=Position, y=log10p)) +
  facet_wrap(.~Chromosome, scales="free_x") +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("-log10 (p-value)\n") +
  xlab("\n Position") 
ggsave("outputs/212/sex-gwas-curated-genome.jpeg")
```

```{r}
sub<-df %>% arrange(-LRT) %>% filter(Chromosome=="SUPER_10") %>% head(n=80) %>% arrange(Position)
ggplot(sub) +
  geom_point(aes(x=Position, y=log10p))
sub %>% select(Chromosome, Position) %>% write_tsv("meta/212-sites.tsv", col_names = FALSE)

sub2<-df %>% arrange(-LRT) %>% filter(Chromosome=="SUPER_10") %>% head(n=250) %>% arrange(Position)
ggplot(sub2) +
  geom_point(aes(x=Position, y=log10p))
sub2 %>% select(Chromosome, Position) %>% write_tsv("meta/212-sites2.tsv", col_names = FALSE)

```

```{sh, eval=FALSE}
srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 72 -bam $HOME/longfin-histories/bamlists/80.bamlist -ref /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta  \
-r SUPER_10:16000000- -sites outputs/212/212-sites.tsv -out outputs/212/snps \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/212/snps.out 2> outputs/212/snps.err &

#57 sites called

srun -t 2:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 60 -bam $HOME/longfin-histories/bamlists/80.bamlist -ref /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta  \
-r SUPER_10:16000000- -sites outputs/212/212-sites.tsv -out outputs/212/snps-relaxed \
-minMaf 0.05 -minMapQ 10 -minQ 10 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-5 -doGlf	2 \
-doGeno 2 -doPost 1 -postCutoff 0.80 -doPlink 2  > outputs/212/snps-relaxed.out 2> outputs/212/snps-relaxed.err &

#	-> Number of sites retained after filtering: 66  (not including top3 sites)
#	-> Number of sites retained after filtering: 75  (when -postCutoff 80)

#sites2 
srun -t 2:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 72 -bam $HOME/longfin-histories/bamlists/80.bamlist -ref /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta  \
-r SUPER_10:14000000- -sites outputs/212/212-sites2.tsv -out outputs/212/snps2 \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/212/snps2.out 2> outputs/212/snps2.err &

```

Top 3 sites
SUPER_10   16495263
SUPER_10   17774598
SUPER_10   16577582
-doGlf	2

```{r}
meta<-fish
ddf<-read_tsv("outputs/212/snps.geno.gz", col_names = c("FALSE")) %>% select(-X83)
#replace with most commmon entry?
#replace with NA for now
ddf[ddf==-1]<-NA
ddf2<-ddf
#ddf2<-ddf %>% filter(X2 %in% head(sub$Position,30))
m<-ddf2 %>% select(-`FALSE`, -X2) %>% as.matrix()

annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Sex<-meta$Sex
annot$Sex<-gsub("1","Female",annot$Sex)
annot$Sex<-gsub("2","Male",annot$Sex)
```

```{r}

as.ggplot(pheatmap(m, labels_col = meta$Sample, cluster_rows=FALSE, annotation_col = annot, width=12, height=8, cluster_cols = FALSE, legend=TRUE,
                   labels_row = ddf$X2,
         fontsize_col = 6)) + ggtitle("Delta Smelt Variants on Delta Smelt SUPER_10") +
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/212/dsm-sex-dsm-reference-heatmap-subset.pdf", width=12, height=8)
```

Can I calculater an AF difference of
0.0 - 0.5 = 0.5?


plink --tped snps2.tped --tfam snps2.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink
bgzip plink.vcf 
tabix plink.vcf.gz
bcftools reheader --samples meta/544.names outputs/108/plink.vcf.gz | bcftools +fill-tags -- -t MAF


#renaming vcf with sample names, filtering again on MAF, and then pruning, selecting random SNP

module load bcftools/1.13

#Basically, pruning before was like this: bcftools +prune -l 0.20 -w 10000



Could split into separate VCFs then calculater freqs with bcftools and remerge (but watch for changing reference allele)
_1_ Identify the highest coverage 10 females and 10 males (and omit ones Mac finds suspicious)
These may be ones that don't track with markers id'd by gwas

Which ones are these?

Unlikely females

901071
701931
902671

Unlikely males
121922
401432
402282
500742
602392
300662
302002
400282
402592
501802
600902
600872
600772
602992
701932
701782
110382

Is there low coverage and a bias towards homozygosity here??





_2_ Extract unaligned sequences from males/females

(can try chrom by chrom pc)

samtools view -b -f 4 file.bam > unmapped.bam
samtools view -b -f 8 file.bam > unmapped.bam # both read1 and read2 unmapped
using the sort.bam files

```{r}
topc %>% ungroup() %>% mutate(Command=paste0("samtools view -b -f 12 /home/maccamp/longfin-histories/data/sub/",Sample,".sort.bam > ",
                              "/home/maccamp/longfin-histories/data/unmapped/",Sample,".bam; samtools collate -u -O /home/maccamp/longfin-histories/data/unmapped/",Sample,".bam | samtools fastq -1 /home/maccamp/longfin-histories/data/unmapped/",Sample,"-r1.fastq -2 /home/maccamp/longfin-histories/data/unmapped/",Sample,"-r2.fastq")) %>%
  select(Command) %>% write_tsv("212.1-unmapped-commands.tsv", col_names = FALSE)

topc %>% ungroup() %>% mutate(Command=paste0("flash data/unmapped/",Sample,"-r1.fastq  data/unmapped/",Sample,"-r2.fastq -o data/unmapped/",Sample,"-flash --max-overlap 120 --min-overlap 20"))  %>%
  select(Command) %>% write_tsv("212.2-flash.tsv", col_names = FALSE)

                              
```


Let's identify from topc some candidate males.
-> not 121922, 402282, 300662

Candidate males
600232, 901522, 901232, 120092

less good candidate
601512, 700232

Let's identify from topc some candidate females
-not 701931

Candidate females
400281, 401261, 500131, 701331, 700901, 801931, 802601

less good candidate
402281

module load parallel
srun -p high --time 4:00:00 --nodes=1 --ntasks-per-node=1 --cpus-per-task=11 parallel -j 11 < 212.1-unmapped-commands.tsv
(this command isn't working for whatever reason)


samtools flagstat 110381.bam (with -f 12)
1889938 + 0 in total 
1889938 + 0 paired in sequencing
944969 + 0 read1
944969 + 0 read2

samtools collate -u -O 110381.bam | samtools fastq -1 110381-r1.fastq -2 110381-r2.fastq

check for read through?


Let's repeat for all our fish, and use these candidate ones
600232, 901522, 901232, 120092
400281, 401261, 500131, 701331



```{sh, eval=FALSE}
#on farm in ~/longfin-histories
bash 212.2-flash.tsv
```

_3_ De novo asssemble unaligned male data, SPADES

erro   1:16:02.922  3668M / 5153M ERROR   General                 (hammer_tools.cpp          : 194)   Pair of read files "/home/maccamp/longfin-histories>

when back and redid with -f 12
[M::bam2fq_mainloop] discarded 0 singletons
[M::bam2fq_mainloop] processed 1889938 reads

Can cat together several individual fastq files 
600232, 901522, 901232, 120092


then run something like this:
```{sh, eval=FALSE}
srun -p bigmemm -t 48:00:00 --mem=256GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/unmapped/110381-r1.fastq -2 data/unmapped/110381-r2.fastq -o data/unmapped/110381 > data/unmapped/110381.out 2> data/unmapped/110381.err & 

#Spades with unpaired
srun -p bigmemm -t 48:00:00 --mem=256GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/unmapped/110381-flash.notCombined_1.fastq -2 data/unmapped/110381-flash.notCombined_2.fastq --merged data/unmapped/110381-flash.extendedFrags.fastq -o data/unmapped/110381-flash > data/unmapped/110381-flash.out 2> data/unmapped/110381-flash.err & 
````


combine, then assemble our candidate males

```{sh, eval=FALSE}
cat 600232-flash.notCombined_1.fastq 901522-flash.notCombined_1.fastq 901232-flash.notCombined_1.fastq 120092-flash.notCombined_1.fastq > male-r1.fastq
cat 600232-flash.notCombined_2.fastq 901522-flash.notCombined_2.fastq 901232-flash.notCombined_2.fastq 120092-flash.notCombined_2.fastq > male-r2.fastq
cat 600232-flash.extendedFrags.fastq 901522-flash.extendedFrags.fastq 901232-flash.extendedFrags.fastq 120092-flash.extendedFrags.fastq > male-extended.fastq
```

```{sh, eval=FALSE}
#Spades with unpaired
conda activate spades
srun -p bigmemm -t 48:00:00 --mem=256GB --nodes=1 --cpus-per-task=6 spades.py --phred-offset 33 -1 data/unmapped/male-r1.fastq -2 data/unmapped/male-r2.fastq --merged data/unmapped/male-extended.fastq -o data/unmapped/male-flash > data/unmapped/male-flash.out 2> data/unmapped/male-flash.err & 
````


Pretty similar assemblies
Comparing spades flashed/unflashed scaffolds
(base) maccamp@farm:~/longfin-histories/data/unmapped/110381$ assembledSeqStats.pl scaffolds.fasta 
Number of Contigs	29419
Total Bases	22325036
No. of Contigs >=1000	6685
Max Contig Length	15845

(base) maccamp@farm:~/longfin-histories/data/unmapped/110381-flash$ assembledSeqStats.pl scaffolds.fasta 
Number of Contigs	22714
Total Bases	20086919
No. of Contigs >=1000	6510
Max Contig Length	15845

For multiple fish, should I specify different libraries, or cat together reads?


I like the plan of assembly of 2 x 5 different males and 2 x 5 females and checking for concordance.

I assembled four candidate males using flashed reads and spades as one assembly.



aligning to/110381-flash scaffolds in data/110381-flash, indexing with bwa...
aligning to/110381 scaffolds in data/110381, indexing with bwa...

bash ../../101.2-do-align.sh to-align.txt /home/maccamp/longfin-histories/data/110381-flash/scaffolds.fasta
bash ../../101.2-do-align.sh to-align.txt /home/maccamp/longfin-histories/data/110381/scaffolds.fasta


making a directory to do this again
male-flash/scaffolds.fasta, linking over all our fish



bash ../../101.2-do-align.sh to-align.txt /home/maccamp/longfin-histories/data/male-flash/scaffolds.fasta



mosdepth, installing with bioconda

now to test across aligned files

--quantize 0:1:4:100:200: # ... arbitary number of quantize bins.

indicates bins of: 0:1, 1:4, 4:100, 100:200, 200:infinity where the upper endpoint is non-inclusive.
0:0:1:
--quantize 0:0:1:4:100:200 


These are our candidate fish though
600232, 901522, 901232, 120092
400281, 401261, 500131, 701331

```{sh, eval=FALSE}
#males
srun -p high --time=1:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 mosdepth -n 600232-depth 600232-realign.sort.bam
srun -p high --time=1:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 mosdepth -n 901522-depth 901522-realign.sort.bam
srun -p high --time=1:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 mosdepth -n 901232-depth 901232-realign.sort.bam
srun -p high --time=1:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 mosdepth -n 120092-depth 120092-realign.sort.bam

#females
srun -p high --time=1:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 mosdepth -n 400281-depth 400281-realign.sort.bam
srun -p high --time=1:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 mosdepth -n 401261-depth 401261-realign.sort.bam
srun -p high --time=1:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 mosdepth -n 500131-depth 500131-realign.sort.bam
srun -p high --time=1:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=2 mosdepth -n 701331-depth 701331-realign.sort.bam


```


