---
title: "213-all-sexed-dsm"
output: html_document
date: "2025-11-04"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(data.table)
library(pheatmap)
library(ggplotify)
library(RColorBrewer)
```


Copying from here
https://github.com/MacCampbell/dsm-omics/blob/main/300-gwas.Rmd


```{r}
df<-read_csv("outputs/213/samples-barcodes-01042022.csv")
df
```

```{r}
df<-read_csv("outputs/213/samples-barcodes-01042022.csv") %>% filter(str_detect(`Sample name`, '^\\d+')) %>%
  separate(col = `Sample name`, into=c("X1","X2","X3","X4","X5","X6","X7"), sep="", remove=FALSE ) %>%
  mutate(Sex = ifelse(X7==2, "female","male")) %>%
  mutate(Pheno=ifelse(Sex=="female", 1, 0))
```

first two digits is a generation number, the three in the middle is a cross number, and the last number is for male/ female (1 or 2).


Bring in coverage:

```{r}
files<-list.files(path="outputs/213", pattern="*.cov", full.names=TRUE)
list<-lapply(files, read_tsv, col_names=FALSE) 
names<-lapply(files, basename)
all<-mapply(c, list, names, SIMPLIFY = FALSE)
comb<-as_tibble(rbindlist(all, fill=T)) %>% rename(Coverage=X1, CovFile=V1)

comb$SeqFile<-gsub(".cov",".bam",comb$CovFile)
comb$`Sample name`<-gsub("_R1.sort.flt.bam","",comb$SeqFile)
comb<-select(comb, -CovFile)
```


```{r}
files<-list.files(path="outputs/213", pattern="*.cov", full.names=TRUE)
list<-lapply(files, read_tsv, col_names=FALSE) 
names<-lapply(files, basename)
all<-mapply(c, list, names, SIMPLIFY = FALSE)
comb<-as_tibble(rbindlist(all, fill=T)) %>% rename(Coverage=X1, CovFile=V1)

comb$SeqFile<-gsub(".cov",".bam",comb$CovFile)
comb$`Sample name`<-gsub("_R1.sort.flt.bam","",comb$SeqFile)
comb<-select(comb, -CovFile)
```

Merge   
```{r}
df<-left_join(df,comb) %>% filter(Coverage !="NA")
nrow(df)
mean(df$Coverage)
```

```{r}
ggplot(df) +
  geom_histogram(aes(Coverage, fill=Sex)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("Count") +
  theme(axis.title = element_text(face="bold")) +
  scale_y_continuous(breaks=c(0,5,10,15,20,25))
```


Creating a tester bamlit

```{r}
tester<-df %>% filter(Coverage > mean(df$Coverage) - sd(df$Coverage)) %>%
  filter(Coverage < mean(df$Coverage) + sd(df$Coverage)) %>%
  mutate(String=paste0("/home/maccamp/dsm-omics/data/",`library number`,"/",SeqFile))
  
  
tester %>% group_by(Sex) %>% summarize(Count=n())
```

```{r}
nrow(tester)
mean(tester$Coverage)
```



Let's get fish not in tester

```{r}
df2<-df %>% filter(!`Sample name` %in% tester$`Sample name`)

ggplot(df2) +
  geom_histogram(aes(Coverage, fill=Sex)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("Count") +
  theme(axis.title = element_text(face="bold")) +
  scale_y_continuous(breaks=c(0,5,10,15,20,25))
df2 %>% filter(Coverage >= 10)
```

Can expect 16 fish with >=10x coverage

in sub2, putting these 24 seqs, ln -s from 
/home/maccamp/dsm-omics/data/ into
data/sub2/

Aligning like so:
bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta

Have sub2.csv

```{r}
sub2<-read_csv("outputs/213/sub2.csv", col_names = c("Sample","Sort","Dedup","Cov")) %>% mutate(Path=paste0("/home/maccamp/longfin-histories/data/sub2/",Sample,".sort.flt.bam"))
sub2$Sex<-gsub("^\\d\\d\\d\\d\\d","",sub2$Sample)
sub2$Sex<-as.numeric(sub2$Sex)
sub2<-sub2 %>% mutate(Pheno=Sex-1) %>% mutate(SexChar=ifelse(Sex==2, "Male","Female"))

nm2<-read_csv(file="meta/nm.csv")
nm3 <- bind_rows(nm2,sub2)
```

```{r}
ggplot(nm3 %>% filter(Cov >=10)) +
  geom_histogram(aes(Cov, fill=SexChar)) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("Count") +
  theme(axis.title = element_text(face="bold")) +
  scale_y_continuous(breaks=c(0,5,10,15,20,25))
```

```{r}
m92<-nm3 %>% filter(Cov >=10)
m92 %>% select(Pheno) %>% write_tsv("bamlists/92.pheno", col_names = FALSE)
m92 %>% select(Path) %>% write_tsv("bamlists/92.bamlist", col_names = FALSE)
m92 %>% select(Sample) %>% write_tsv("bamlists/92.names", col_names=FALSE)
m92 %>% filter(SexChar=="Female") %>% select(Sample) %>% write_tsv("bamlists/92-females.names", col_names=FALSE)
m92 %>% filter(SexChar=="Male") %>% select(Sample) %>% write_tsv("bamlists/92-males.names", col_names=FALSE)
```


```{sh,eval=FALSE}
 srun -t 72:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12 \
 -doAsso 1 -yBin $HOME/longfin-histories/bamlists/92.pheno -GL 1 -minInd 69 -minMaf 0.10 \
 -minMapQ 20 -minQ 20 \
 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -out outputs/213/asso-75 \
 -bam $HOME/longfin-histories/bamlists/92.bamlist  > outputs/213/asso-75.out 2> outputs/213/asso-75.err &
 
 srun -t 48:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 72 -bam $HOME/longfin-histories/bamlists/92.bamlist -ref /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta  \
-out outputs/213/snps-05 \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/213/snps-05.out 2> outputs/213/snps-05.err &
```

```{r}
files = list.files(path="outputs/213/", pattern="*75.lrt0.gz", full.names = TRUE)
list = lapply(files, read_tsv)
data<-bind_rows(list)
```

```{r}
df <- data %>% mutate(log10p = -log10(dchisq(LRT, df = 1))) %>%
  filter(log10p>=0 & log10p != "Inf") %>%
  mutate(p = dchisq(LRT, df=1)) %>%
  mutate(bonfer = p.adjust(p, method = "bonferroni")) %>% 
  mutate(BHfdr = p.adjust(p, method = "BH")) %>%
  group_by(Chromosome) %>%
  mutate(adjpchrom = p.adjust(p, method = "bonferroni")) %>%
  mutate(BHfdrchrom = p.adjust(p, method = "BH")) %>% ungroup() %>%
  mutate(Index=1:n())

df %>% arrange(-LRT) %>% head(n=20)
```
