---
title: "214-ld-pcas-05"
output: html_document
date: "2025-11-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message=FALSE)
```

```{r}
library(RColorBrewer)
library(vcfR)
library(adegenet)
library(tidyverse)
```

```{r}
meta<-read_csv("meta/nm.csv")
```


```{sh,eval=FALSE}
bcftools view -Ov -r SUPER_10:16000000- outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_10-region.vcf
bcftools view -Ov -r SUPER_6:0-3500000 outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_6-region.vcf
bcftools view -Ov -r SUPER_23:0-2500000 outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_23-region.vcf
bcftools view -Ov -r SUPER_24:6000000- outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_24-region.vcf
bcftools view -Ov -r SUPER_1:0-10000000 outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_1-region.vcf
```

Using the 05 ones and visualizing here to prevent confusion

```{sh, eval=FALSE}
#with thinned 10
./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/214/bcfs -t snp -s 20 -m 5 -I /Users/mac/github/longfin-histories/outputs/214/bcfs/sample_info.tsv -j 214ld
cp lostruct_results/type_snp_size_20_weights_none_jobid_214/mds_coords.csv  ~/github/longfin-histories/outputs/214/mds_coords_214ld.csv
```


```{r}
mds<-read_csv("outputs/214//mds_coords_214ld.csv") # 20 snp windows


mds$chrom<-factor(mds$chrom,levels=c("SUPER_1","SUPER_2","SUPER_3","SUPER_4","SUPER_5","SUPER_6","SUPER_7","SUPER_8","SUPER_9","SUPER_10","SUPER_11","SUPER_12","SUPER_13","SUPER_14","SUPER_15","SUPER_16","SUPER_17","SUPER_18","SUPER_19","SUPER_20","SUPER_21","SUPER_22","SUPER_23","SUPER_24","SUPER_25","SUPER_26","SUPER_27","SUPER_28"))
#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')


ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')


ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


### MDS1

```{r}
p1<-MDS1 %>% arrange(chrom,window) %>%mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```


### MDS2
```{r}
p2<-MDS2 %>% arrange(chrom,window) %>% mutate(Index=1:n())
out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```


```{r}
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```


### MDS3
```{r}
p3<-MDS3  %>% arrange(chrom,window) %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```
```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))


#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```


### MDS4
```{r}
p4<-MDS4  %>% arrange(chrom,window) %>% mutate(Index=1:n())

out4 <- boxplot.stats(p4$Value4)$out
out_ind4 <- which(p4$Value4 %in% c(out4))

out_ind4
outliers4<-p4[out_ind4,]
outliers4 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```
```{r}
#places to put labels based on index
chroms<-p4 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


ggplot(p4) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p4$Value4), ymax=max(p4$Value4)), fill=mycolors, alpha=0.25) +
  geom_point(data=p4, aes(x=Index, y=Value4, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers4, aes(x=Index, y=Value4), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS4") +
  xlab("Chromosome")
```

