---
title: "215-redo-rad"
output: html_document
date: "2025-11-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(RColorBrewer)
library(vcfR)
library(adegenet)
library(tidyverse)
```

The goal is to go to the radseq from the G3 paper and realign etc.

(base) ➜  github git clone https://github.com/MacCampbell/delta-smelt/
Cloning into 'delta-smelt'...


```{r}
lh<-read_csv("../delta-smelt/metadata/2012-all-samples.csv") %>% mutate(Pheno=ifelse(aggregated_new_classes=="MIG",0,1)) %>% mutate(SexPheno=sex-1) %>% mutate(SexChar=ifelse(sex==1, "Female","Male"))

lh
```

We want to move all the .fastqs to my longfin-histories/redo Looks like they are gzipped

(base) maccamp@farm:~/longfin-histories/data/redo$ cut -f 4 -d ',' ~/delta-smelt/metadata/2012-all-samples.csv  | while read line; do cp ~/ds-lh/data/$line*fastq ./; done;

/group/millermrgrp2/shannon/projects/DS_history/data/BMAG048/ACTGAT$ 

Files are in 
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG043/ATCACG/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG043/CGATGT/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG044/ACAGTG/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG044/GCCAAT/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG045/ACTTGA/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG045/CAGATC/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG045/GATCAG/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG046/AGTCAA/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG046/AGTTCC/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG046/CTTGTA/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG047/ATGTCA/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG047/CCGTCC/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG047/GTCCGC/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG047/GTGAAA/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG048/ACTGAT/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG048/ATTCCT/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG048/CGTACG/
/group/millermrgrp2/shannon/projects/DS_history/data/BMAG048/GTGGCC/

cat dirs  | while read line; do ln -s $line*.fastq.gz .; done;
 
 
we want a file like
Ht20-33_2012_A05_R1.fastq.gz    Ht20-33_2012_A05_R2.fastq.gz    Ht20-33_2012_A05

```{r}
lh %>% select(`Sequence File Name`) %>% mutate(Forward=paste0(`Sequence File Name`,"_R1.fastq.gz")) %>% mutate(Reverse=paste0(`Sequence File Name`,"_R2.fastq.gz")) %>% relocate(Forward, Reverse, `Sequence File Name`) %>%
  write_tsv("215.1-to-align.tsv", col_names = FALSE)
```

165 samples to align
in data/redo
```{sh, eval=FALSE}
cp ../../215.1-to-align.tsv ./to-align.txt
cut -f 1 to-align.txt  | while read line; do ls $line; done; #all files present

bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta
```

couple of jobs stuck
28559307  maccamp          med            Ht19-23_2012_G03.sh  S   8-10:58:15   2     1     N/A           cpu-9-64            
28559303  maccamp          med            Ht19-86_2012_F11.sh  S   8-10:59:52   2     1     N/A           cpu-9-63            

```{sh, eval=FALSE}
sbatch -p high -t 1-11:00:00 --mem=8G  Ht19-23_2012_G03.sh
sbatch -p high -t 1-11:00:00 --mem=8G  Ht19-86_2012_F11.sh
```

Now I need to drop individuals without sex id. Jim says:    
1 = male      
2= female     
0= can't tell, if the fish had spawned and not residual gonad tissue was available the fish is scored a 0    

```{r}
lh %>% filter(sex !=0) %>% group_by(aggregated_new_classes, sex) %>% summarize(Count=n())
reads<-lh %>% left_join(read_csv("outputs/215/stats.csv", col_names=c("Sequence File Name","Reads","Dedup","Cov")))
```

Join with coverage data, and get pheno and bamlist sorted based on some sort of coverage. For the paper, kept 122 samples previously. This approach gives us 121, so altering a bit and gitting 121. not caring

couple jobs 

we can bind stats on to lh with a 'reads' df and copy from here



```{r}
tops<-reads %>% top_frac(.75, Dedup) %>% filter(sex!=0)
#previous<-read_csv("~/github/delta-smelt/metadata/2012-samples.csv") %>% select(`Sequence File Name`)
tops<-reads %>% filter(`Sequence File Name` %in% previous$`Sequence File Name`)
samplesize<-tops %>% group_by(aggregated_new_classes ,sex) %>% summarize(`Sample Size`=n())
samplesize
```

```{r}
ggplot(tops) +
  geom_histogram(aes(Dedup)) +
  facet_wrap(.~Pheno) +
  xlim(0, max(tops$Dedup))
```

```{r}
tops %>% group_by(aggregated_new_classes, sex) %>% select(Dedup) %>% summarize_all(mean) %>% left_join(samplesize)
```



## Generate Bamlists, Phenotype and Sex Files

```{r}
bamlist<-tops %>% mutate(Path=paste0("data/redo/",`Sequence File Name`,".sort.flt.bam")) %>% select(Path)
write_tsv(bamlist, "bamlists/2012.bamlist", col_names=FALSE)

phenos<-tops %>% select(Pheno) 
write_tsv(phenos, "phenos/2012.phenos", col_names=FALSE)

sex<-tops %>% select(sex)
write_tsv(sex, "phenos/2012.cov", col_names=FALSE)

sexpheno<-tops %>% select(SexPheno)
write_tsv(sexpheno, "phenos/2012-sex.phenos", col_names=FALSE)

names<-tops %>% select(`Sequence File Name`)
write_tsv(names, "bamlists/2012.names", col_names=FALSE)

write_csv(tops, "meta/2012-samples.csv")
```

Get higher coverage fish
```{r}
m81<-reads %>% top_frac(.5, Dedup) %>% filter(sex!=0)  %>% mutate(Path=paste0("data/redo/",`Sequence File Name`,".sort.flt.bam")) 

m81 %>% select(Path) %>% write_tsv("bamlists/2012-81.bamlist", col_names=FALSE)

m81 %>% select(Pheno) %>% write_tsv("phenos/2012-81.phenos", col_names=FALSE)

m81 %>% select(sex) %>% write_tsv("phenos/2012-81.cov", col_names=FALSE)

m81 %>% select(SexPheno) %>% write_tsv("phenos/2012-sex-81.phenos", col_names=FALSE)

m81 %>% select(`Sequence File Name`) %>% write_tsv("bamlists/2012-81.names", col_names=FALSE)

write_csv(m81, "meta/2012-81-samples.csv")
```
For sanity's sake, I'll call snps

snps-all.err

```{sh, eval=FALSE}
srun -t 24:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 109 -bam $HOME/longfin-histories/bamlists/2012.bamlist -ref /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta  \
-out outputs/215/snps-all \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/215/snps-all.out 2> outputs/215/snps-all.err &
#	-> Number of sites retained after filtering: 6978 


srun -t 24:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 73 -bam $HOME/longfin-histories/bamlists/2012-81.bamlist -ref /home/maccamp/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta  \
-out outputs/215/snps-all-81 \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/215/snps-all-81.out 2> outputs/215/snps-all-81.err &
# 	-> Number of sites retained after filtering: 10317 



# convert to vcf etc

plink --tped snps-all.tped --tfam snps-all.tfam  --out plink --recode --allow-extra-chr --noweb
plink --ped plink.ped --map plink.map --recode vcf --allow-extra-chr -out plink-05
bgzip plink-05.vcf 
tabix plink-05.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/2012.names -o outputs/215/renamed-05.vcf.gz outputs/215/plink-05.vcf.gz

bcftools +fill-tags outputs/215/renamed-05.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.05' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/215/pruned-05.vcf

bcftools +fill-tags outputs/215/renamed-05.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.05' > outputs/215/plink-05-filtered.vcf

bgzip outputs/215/plink-05-filtered.vcf
tabix outputs/215/plink-05-filtered.vcf.gz

# convert to vcf etc 81

plink --tped snps-all-81.tped --tfam snps-all-81.tfam  --out plink-81 --recode --allow-extra-chr --noweb
plink --ped plink-81.ped --map plink-81.map --recode vcf --allow-extra-chr -out plink-81-05
bgzip plink-81-05.vcf 
tabix plink-81-05.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/2012-81.names -o outputs/215/renamed-81-05.vcf.gz outputs/215/plink-81-05.vcf.gz

bcftools +fill-tags outputs/215/renamed-81-05.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.05' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/215/pruned-81-05.vcf

bcftools +fill-tags outputs/215/renamed-81-05.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.05' > outputs/215/plink-81-05-filtered.vcf

bgzip outputs/215/plink-81-05-filtered.vcf
tabix outputs/215/plink-81-05-filtered.vcf.gz


```


Run LD and PCAS

```{sh, eval=FALSE}
cat meta/super-chroms.txt  | while read line; do bcftools view -Ov -r $line outputs/215/plink-05-filtered.vcf.gz  > outputs/215/vcfs/$line.vcf; done;

bcftools view -Ov -r SUPER_10:16000000- outputs/215/plink-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_10-region.vcf
bcftools view -Ov -r SUPER_6:0-3500000 outputs/215/plink-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_6-region.vcf
bcftools view -Ov -r SUPER_23:0-2500000 outputs/215/plink-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_23-region.vcf
bcftools view -Ov -r SUPER_24:6000000- outputs/215/plink-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_24-region.vcf
bcftools view -Ov -r SUPER_1:0-10000000 outputs/215/plink-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_1-region.vcf

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;

cat meta/super-chroms.txt  | while read line; do bcftools view -Ov -r $line outputs/215/plink-81-05-filtered.vcf.gz  > outputs/215/vcfs/$line.vcf; done;

bcftools view -Ov -r SUPER_10:16000000- outputs/215/plink-81-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_10-region.vcf
bcftools view -Ov -r SUPER_6:0-3500000 outputs/215/plink-81-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_6-region.vcf
bcftools view -Ov -r SUPER_23:0-2500000 outputs/215/plink-81-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_23-region.vcf
bcftools view -Ov -r SUPER_24:6000000- outputs/215/plink-81-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_24-region.vcf
bcftools view -Ov -r SUPER_1:0-10000000 outputs/215/plink-81-05-filtered.vcf.gz     > outputs/215/vcfs/SUPER_1-region.vcf

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;

```


```{r, eval=FALSE}
files<-list.files("outputs/215/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/215/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% dplyr::arrange(R2) %>%  filter(R2 >0.1)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
#ggsave(paste0("outputs/215/chrom-ld/",chrom,"-ld.pdf"))
ggsave(paste0("outputs/215/chrom-ld/",chrom,"-ld.jpeg"))

}

lapply(files, plotLd)
```


Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/215/vcfs",pattern = "*.vcf", full.names = TRUE)
meta<-m81
meta$pop<-meta$SexChar
plotPCA<-function(file) {
  meta<-meta
  chrom<-gsub("outputs/215/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="viridis") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
 # geom_text_repel(aes(x=Axis1, y=Axis2,label=id))
pc12

ggsave(paste0("outputs/215/chrom-pcs/",chrom,"-pcs.pdf"))
}

lapply(files, plotPCA)
```


## Figure 2

doAsso     
With and without sex as cov at 75% thresh for individuals
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12 
no -rf $HOME/genomes/hypomesus-20210204/large-contigs.txt 
```{sh, eval=FALSE}
srun -p bigmemm -t 24:00:00 --mem=32G --nodes=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  -bam $HOME/longfin-histories/bamlists/2012.bamlist \
-yBin $HOME/longfin-histories/phenos/2012.phenos -cov $HOME/longfin-histories/phenos/2012.cov -minMapQ 20 -minQ 20 -minInd 91 -doAsso 2 \
-doPost 1 -GL 1 -minCount 2 -out $HOME/longfin-histories/outputs/215/assoc05-75-2 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-ref $HOME/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta \
> outputs/215/assoc05-75-2.out 2> outputs/215/assoc05-75-2.err &

srun -p bigmemm -t 24:00:00 --mem=32G --nodes=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  -bam $HOME/longfin-histories/bamlists/2012.bamlist \
-yBin $HOME/longfin-histories/phenos/2012.phenos -minMapQ 20 -minQ 20 -minInd 91 -doAsso 2 -doPost 1 -GL 1 -minCount 2 \
-out $HOME/longfin-histories/outputs/215/assoc05-75-2-nocov -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-ref $HOME/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta \
> outputs/215/assoc05-75-2-nocov.out 2> outputs/215/assoc05-75-2-nocov.err &

srun -p bigmemm -t 24:00:00 --mem=32G --nodes=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12  -bam $HOME/longfin-histories/bamlists/2012.bamlist \
-yBin $HOME/longfin-histories/phenos/2012.phenos -minMapQ 20 -minQ 20 -minInd 91 -doAsso 1 -GL 1 -minCount 2 \
-out $HOME/longfin-histories/outputs/215/assoc05-75-2-nocov -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-ref $HOME/genomes/hypomesus-curated/fHypTra1.pri.cur.20251017.fasta \
> outputs/215/assoc05-75-1-nocov.out 2> outputs/215/assoc05-75-1-nocov.err &

#sex gwas

srun -t 24:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12 \
 -doAsso 1 -yBin $HOME/longfin-histories/phenos/2012-sex.phenos -GL 1 -minInd 91 -minMaf 0.10 \
 -minMapQ 20 -minQ 20 \
 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -out outputs/215/asso-75 \
 -bam $HOME/longfin-histories/bamlists/2012.bamlist  > outputs/215/asso.out 2> outputs/215/asso.err &
#	-> Number of sites retained after filtering: 17773 


srun -t 24:00:00 -p bigmemm --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12 \
 -doAsso 1 -yBin $HOME/longfin-histories/phenos/2012-sex.phenos -GL 1 -minInd 61 -minMaf 0.10 \
 -minMapQ 20 -minQ 20 \
 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -out outputs/215/asso-50 \
 -bam $HOME/longfin-histories/bamlists/2012.bamlist  > outputs/215/asso-50.out 2> outputs/215/asso-50.err &
 
```

Check on these:

```{r}
df<-read_tsv(file="outputs/215/assoc05-75-2-nocov.lrt0.gz") %>% mutate(log10p = -log10(dchisq(LRT, df = 1))) %>%
  filter(log10p>=0 & log10p != "Inf") %>%
  mutate(p = dchisq(LRT, df=1)) %>%
  mutate(bonfer = p.adjust(p, method = "bonferroni")) %>% 
  mutate(BHfdr = p.adjust(p, method = "BH")) %>%
  group_by(Chromosome) %>%
  mutate(adjpchrom = p.adjust(p, method = "bonferroni")) %>%
  mutate(BHfdrchrom = p.adjust(p, method = "BH"))

df %>% arrange(-LRT) %>% head(n=20)
```

Calculate some significance threshold
```{r}
variants<-nrow(df)
#Number of expected variants
num<-50
p.T = num/variants 
prior.odds = p.T/(1-p.T) 
pwr = 1 #upper bound for power --> upper bound for alpha 
post.odds = 0.95/(1-0.95) 
alpha = prior.odds*pwr/post.odds 
paste(signif(alpha,3)) 
-log10(alpha)
df %>% dplyr::filter(log10p > -log10(alpha)) 
write_csv(df, file="outputs/215/association-results.csv")
```

```{r}

data<-df  %>% filter(str_detect(Chromosome, "SUPER_"))
data<-data  %>% filter(!str_detect(Chromosome, "unloc"))

dd <- data %>% ungroup %>% mutate(Index=1:n())
dd$Chromosome<-as.factor(dd$Chromosome)

chroms<-dd %>% group_by(Chromosome) %>% mutate(Start=min(Index), Stop=max(Index)) %>% 
  select(Chromosome,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

outliers <- dd %>% filter(log10p >= -log10(alpha))
#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)
#randomize
mycolors<-sample(mycolors)

ggplot(dd) +
  geom_point(data=dd, aes(x=Index, y=log10p, color=Chromosome), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=log10p, color=Chromosome), cex=2, alpha=0.9) +
  geom_point(data=outliers, aes(x=Index, y=log10p), pch=21, cex=2, alpha=0.9) +

 # geom_hline(yintercept = -log10(0.05/nrow(dddf)), col="black", linetype=2, alpha=0.5) +
  geom_hline(yintercept= -log10(alpha), col="black", linetype=1, alpha=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$Chromosome) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=8)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  scale_fill_manual(breaks=unique(dd$Chromosome), values=mycolors) +
  ylab("-log10(p)\n") +
  xlab("\nChromosome") +
  ggtitle("FWR vs MIG Comparison") +
  labs(subtitle = "Sex as Covariant") +
  theme(plot.title = element_text(hjust=0.5, face="bold") ) +
  theme(plot.subtitle = element_text(hjust=0.5))

ggsave("outputs/215/figure-2-redo.jpg", width=8.5, height=11/3)
```


## Basic sex GWAS

```{r}
data<-read_tsv("outputs/215/asso-50.lrt0.gz")
df <- data %>% mutate(log10p = -log10(dchisq(LRT, df = 1))) %>%
  filter(log10p>=0 & log10p != "Inf") %>%
  mutate(p = dchisq(LRT, df=1)) %>%
  mutate(bonfer = p.adjust(p, method = "bonferroni")) %>% 
  mutate(BHfdr = p.adjust(p, method = "BH")) %>%
  group_by(Chromosome) %>%
  mutate(adjpchrom = p.adjust(p, method = "bonferroni")) %>%
  mutate(BHfdrchrom = p.adjust(p, method = "BH")) %>% ungroup() %>%
  mutate(Index=1:n())

df %>% arrange(-LRT) %>% head(n=20)
```

```{r}
df %>% filter(Chromosome=="SUPER_10") %>% arrange(-LRT) %>% head(n=20)
```

```{r}
#c("SUPER_1","SUPER_10","SUPER_11","Scaffold_120","SUPER_3","Scaffold_129","SUPER_8")
ggplot(df %>% filter(Chromosome %in% c("SUPER_1","SUPER_10","SUPER_11","SUPER_4"))) +
  geom_point(aes(x=Position, y=log10p)) +
  facet_wrap(.~Chromosome, scales="free_x") +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ylab("-log10 (p-value)\n") +
  xlab("\n Position") 
ggsave("outputs/215/sex-gwas-curated-genome.jpeg")
```

## Local PCA

(base) ➜  215 git:(main) ✗ gunzip -c plink-05-filtered.vcf.gz | grep -v "#" | wc
    6371  828230 3600635
(base) ➜  215 git:(main) ✗ gunzip -c plink-81-05-filtered.vcf.gz | grep -v "#" | wc
    9447  850230 3812680
    
```{sh, eval=FALSE}
cat meta/super-chroms.txt  | while read line; do bcftools view -Ob -r $line outputs/215/plink-81-05-filtered.vcf.gz  > outputs/215/bcfs/$line.bcf; done;

for f in outputs/215/bcfs/*.bcf; do bcftools index $f; done;
```


```{r}
samples<-m81 %>% select(`Sequence File Name` ) %>% rename(ID=`Sequence File Name` )

population<-meta %>% select(SexChar) %>% rename(population=SexChar)

table<-cbind(samples, population)
write.table(table, "outputs/215/bcfs/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```

```{sh, eval=FALSE}
./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/215/bcfs -t snp -s 20 -m 5 -I /Users/mac/github/longfin-histories/outputs/215/bcfs/sample_info.tsv -j 215
./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/215/bcfs -t snp -s 15 -m 5 -I /Users/mac/github/longfin-histories/outputs/215/bcfs/sample_info.tsv -j 215

cp lostruct_results/type_snp_size_20_weights_none_jobid_215/mds_coords.csv ~/github/longfin-histories/outputs/215/
cp lostruct_results/type_snp_size_15_weights_none_jobid_215/mds_coords.csv ~/github/longfin-histories/outputs/215/mds_coords-15.csv

```



```{r}
mds<-read_csv("outputs/215/mds_coords.csv") # 20 snp windows
mds<-read_csv("outputs/215/mds_coords-15.csv") # 20 snp windows


mds$chrom<-factor(mds$chrom,levels=c("SUPER_1","SUPER_2","SUPER_3","SUPER_4","SUPER_5","SUPER_6","SUPER_7","SUPER_8","SUPER_9","SUPER_10","SUPER_11","SUPER_12","SUPER_13","SUPER_14","SUPER_15","SUPER_16","SUPER_17","SUPER_18","SUPER_19","SUPER_20","SUPER_21","SUPER_22","SUPER_23","SUPER_24","SUPER_25","SUPER_26","SUPER_27","SUPER_28"))
#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')


ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')


ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


### MDS1

```{r}
p1<-MDS1 %>% arrange(chrom,window) %>%mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```


### MDS2
```{r}
p2<-MDS2 %>% arrange(chrom,window) %>% mutate(Index=1:n())
out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```


```{r}
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```


### MDS3
```{r}
p3<-MDS3  %>% arrange(chrom,window) %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```
```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))


#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```


## What about potential diagnostic SNPs?

Let's look at regions of chroms and create heatmaps


bcftools view -Ov -r SUPER_10:16000000- outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_10-region.vcf
bcftools view -Ov -r SUPER_6:0-3500000 outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_6-region.vcf
bcftools view -Ov -r SUPER_23:0-2500000 outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_23-region.vcf
bcftools view -Ov -r SUPER_24:6000000- outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_24-region.vcf
bcftools view -Ov -r SUPER_1:0-10000000 outputs/214/plink-05-filtered.vcf.gz     > outputs/214/vcfs/SUPER_1-region.vcf

```{r}
ddf<-read_tsv("outputs/215/snps-all-81.geno.gz", col_names = FALSE) %>% select(-X84)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(X1=="SUPER_1")
tops<-read_csv("outputs/214/chrom1-loadings.csv")
ddf2<-ddf %>% filter(X2 %in% tops$Position)
#ddf2<-ddf %>% filter(X2 < 12e6) %>% filter(X2 > 10e6)
#ddf2 <- ddf2 %>% filter(X2 %in% c(20131490,20144334,20263784,20263652,29263756))

m<-ddf2 %>% select(-X1, -X2) %>% as.matrix()

annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Pheno<-m81$aggregated_new_classes

ann_colors = list(
  Pheno = c(FWR="red", MIG="blue"))

```

```{r}

as.ggplot(pheatmap(m, annotation_col = annot, annotation_colors =  ann_colors, color = viridis(3,option="magma", alpha=0.9),
                   labels_row = ddf2$X2, labels_col = m81$`Sequence File Name`, fontsize_row = 12, fontsize_col = 8, cluster_rows = TRUE, cluster_cols = TRUE))+
            ggtitle("Heatmap of Delta Smelt variants on SUPER_1\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/215/dsm-01-heatmap.pdf", width=14, height=8)

```

## Super 6
```{r}
ddf<-read_tsv("outputs/215/snps-all-81.geno.gz", col_names = FALSE) %>% select(-X84)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(X1=="SUPER_6")
tops<-read_csv("outputs/214/chrom6-loadings.csv")
ddf2<-ddf %>% filter(X2 %in% tops$Position)
#ddf2<-ddf %>% filter(X2 < 12e6) %>% filter(X2 > 10e6)
#ddf2 <- ddf2 %>% filter(X2 %in% c(20131490,20144334,20263784,20263652,29263756))

m<-ddf2 %>% select(-X1, -X2) %>% as.matrix()

annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Pheno<-m81$aggregated_new_classes

ann_colors = list(
  Pheno = c(FWR="red", MIG="blue"))

```

```{r}

as.ggplot(pheatmap(m, annotation_col = annot, annotation_colors =  ann_colors, color = viridis(3,option="magma", alpha=0.9),
                   labels_row = ddf2$X2, labels_col = m81$`Sequence File Name`, fontsize_row = 12, fontsize_col = 8, cluster_rows = TRUE, cluster_cols = TRUE))+
            ggtitle("Heatmap of Delta Smelt variants on SUPER_6\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/215/dsm-06-heatmap.pdf", width=14, height=8)

```

## Super 10
```{r}
ddf<-read_tsv("outputs/215/snps-all-81.geno.gz", col_names = FALSE) %>% select(-X84)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(X1=="SUPER_10")
tops<-read_csv("outputs/214/chrom10-loadings.csv")
ddf2<-ddf %>% filter(X2 %in% tops$Position)
#ddf2<-ddf %>% filter(X2 < 12e6) %>% filter(X2 > 10e6)
#ddf2 <- ddf2 %>% filter(X2 %in% c(20131490,20144334,20263784,20263652,29263756))

m<-ddf2 %>% select(-X1, -X2) %>% as.matrix()

annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Pheno<-m81$aggregated_new_classes

ann_colors = list(
  Pheno = c(FWR="red", MIG="blue"))

```

```{r}

as.ggplot(pheatmap(m, annotation_col = annot, annotation_colors =  ann_colors, color = viridis(3,option="magma", alpha=0.9),
                   labels_row = ddf2$X2, labels_col = m81$`Sequence File Name`, fontsize_row = 12, fontsize_col = 8, cluster_rows = TRUE, cluster_cols = TRUE))+
            ggtitle("Heatmap of Delta Smelt variants on SUPER_10\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/215/dsm-10-heatmap.pdf", width=14, height=8)

```


## Super 24
```{r}
ddf<-read_tsv("outputs/215/snps-all-81.geno.gz", col_names = FALSE) %>% select(-X84)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(X1=="SUPER_24")
tops<-read_csv("outputs/214/chrom24-loadings.csv")
ddf2<-ddf %>% filter(X2 %in% tops$Position)
#ddf2<-ddf %>% filter(X2 < 12e6) %>% filter(X2 > 10e6)
#ddf2 <- ddf2 %>% filter(X2 %in% c(20131490,20144334,20263784,20263652,29263756))

m<-ddf2 %>% select(-X1, -X2) %>% as.matrix()

annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Pheno<-m81$aggregated_new_classes

ann_colors = list(
  Pheno = c(FWR="red", MIG="blue"))

```

```{r}

as.ggplot(pheatmap(m, annotation_col = annot, annotation_colors =  ann_colors, color = viridis(3,option="magma", alpha=0.9),
                   labels_row = ddf2$X2, labels_col = m81$`Sequence File Name`, fontsize_row = 12, fontsize_col = 8, cluster_rows = TRUE, cluster_cols = TRUE))+
            ggtitle("Heatmap of Delta Smelt variants on SUPER_24\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/215/dsm-24-heatmap.pdf", width=14, height=8)

```