---
title: "300-polished-genome"
output: html_document
date: "2025-08-07"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ape)
library(viridis)
library(adegenet)
library(RColorBrewer)
library(tidyverse)
library(ggpubr)
library(grid)
library(gridExtra)
library(ggrepel)
library(ggplotify)
library(snpR)
library(vcfR)
library(ggtree)
library(tanggle)
library(pheatmap)
```

Reference genome: /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa

I have data from Saglam et al. aligned to this reference.

```{r}
sag<-read_csv("meta/saglam.csv", col_names=c("ID","Reads","Dedup","Cov")) %>% mutate(Path=paste0("data/polish-saglam/",ID,".sort.flt.bam"))
sag$Pop<-gsub("_\\d+","",sag$ID)
sag<-sag %>% relocate(Pop)
```

```{r}
sag %>% filter(Dedup > 2e5) %>% group_by(Pop) %>% summarize(Count=n(), Mean=mean(Dedup), Median=median(Dedup))
```

```{r}
ggplot(sag %>% filter(Dedup > 2e5)) +
  geom_histogram(aes(x=Dedup)) 
```


```{r}
s116<-sag %>% filter(Dedup > 5e5)
s72<-sag %>% filter(Dedup > 1e6)

s116 %>% group_by(Pop) %>% summarize(Count=n(), Mean=mean(Dedup), Median=median(Dedup))
ggplot(s116) +
  geom_histogram(aes(x=Dedup)) 
```

## Generate covariance matrix for sanity check

```{r}
s144<-sag %>% filter(Dedup > 2e5)
s144 %>% select(Path) %>% write_tsv("bamlists/s144.bamlist", col_names = FALSE)
s144 %>% select(ID) %>% write_tsv("bamlists/s144.names", col_names = FALSE)

s144$Pop<-factor(s144$Pop, levels=c("ALVS","CHPI","SFBY","PETA","SUIB","HUMB","COLR","LWSH","PTLC","HRLC","SKNA","YBAK"))

s72 %>% select(Path) %>% write_tsv("bamlists/s72.bamlist", col_names = FALSE)
s72 %>% select(ID) %>% write_tsv("bamlists/s72.names", col_names = FALSE)

s72$Pop<-factor(s72$Pop, levels=c("ALVS","CHPI","SFBY","PETA","SUIB","HUMB","COLR","LWSH","PTLC","HRLC","SKNA","YBAK"))

#26 fish
sfb<-s72 %>% filter(Pop %in% c("ALVS","CHPI","SFBY","PETA","SUIB"))

write_csv(sfb, file="meta/sfb.csv")

```

```{sh, eval=FALSE}
srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/s144.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-minInd 130 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/300/s144-ibs-90 >outputs/300/s144-ibs-90.out 2> outputs/300/s144-ibs-90.err &

srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/s144.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \ 
-minInd 108 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/300/s144-ibs-75 >outputs/300/s144-ibs-75.out 2> outputs/300/s144-ibs-75.err &
```

	-> Number of sites retained after filtering: 60378 @ 90%


```{r}
m <- as.matrix(read.table("outputs/300/s144-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(s144)

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Pop), pch=21, cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="magma") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("B") +
  theme(plot.title = element_text(hjust=0, size=16)) 

pc12

```
```{r}
pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Pop), pch=21, cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="magma") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("C") +
  theme(plot.title = element_text(hjust=0, size=16)) 

pc13

```

Scree plot
 1 -0.0326 -0.0561 -0.710  SFBY  SFBY_075 2764947  705849 0.0981 data/polish-saglam/SFBY_075.sort.flt.bam
 2 -0.0387 -0.0642 -0.656  SFBY  SFBY_083 2616986  683952 0.100  data/polish-saglam/SFBY_083.sort.flt.bam
 
 These are two SFBY ones, as well as the two weird YBAK ones
 
 These SFBY ones have a lot of missing data
 
```{r}
vardf<-var %>% as_tibble() %>% rename(`Percent Variance`=value) %>% head() %>% mutate(PC=1:n())

scree<-ggplot(vardf, aes(x=PC, y=`Percent Variance`)) +
  geom_line() +
  geom_point() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks = seq(1:6)) +
  theme(axis.text = element_text(size=10)) +
  theme(axis.title = element_text(size=14)) +
  theme(plot.title = element_text(hjust=0, size=16)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16)) 


scree
```

## Call SNPs

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 130 -bam bamlists/s144.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/300/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/300/snps.out 2> outputs/300/snps.err &

srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 65 -bam bamlists/s72.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/300/snps-wgs-1m  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/300/snps-1m.out 2> outputs/300/snps-1m.err &


#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/s144.names -o plink-renamed.vcf plink.vcf
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered.vcf > plink-pruned-max.vcf


## Here I'm looking at only higher coverage individuals
plink --tped snps-wgs-1m.tped --tfam snps-wgs-1m.tfam  --out plink-binary-1m --recode --allow-extra-chr --noweb
plink --ped plink-binary-1m.ped --map plink-binary-1m.map --recode vcf --allow-extra-chr -out plink-1m

module load bcftools
bcftools reheader -s ../../bamlists/s72.names -o plink-renamed-1m.vcf plink-1m.vcf
bcftools +fill-tags plink-renamed-1m.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered-1m.vcf

#Getting region of interest

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered-1m.vcf > plink-pruned-1m.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered-1m.vcf > plink-pruned-max-1m.vcf

```

	-> Number of sites retained after filtering: 25402 (s144)
	-> Number of sites retained after filtering: 69229 (s72)


Filter for missing individuals

```{sh, eval=FALSE}
paste \
<(bcftools query -f '[%SAMPLE\t]\n' filtered.vcf.gz | head -1 | tr '\t' '\n') \
<(bcftools query -f '[%GT\t]\n' filtered.vcf.gz | awk -v OFS="\t" '{for (i=1;i<=NF;i++) if ($i == "./.") sum[i]+=1 } END {for (i in sum) print i, sum[i] / NR }' | sort -k1,1n | cut -f 2) > missingness.stats
````

PTLC_006	0.194709
PTLC_010	0.204984
SFBY_134	0.209432
PTLC_014	0.222148
PTLC_028	0.241713
SFBY_083	0.268089
PTLC_013	0.287655
PTLC_019	0.297457
COLR_003	0.367806
SFBY_075	0.470711

Wow! These fish have over 20% missing calls.

```{r}
snpsRAND<-import.snpR.data(genotypes = "outputs/300/plink-pruned.vcf.gz", 
      sample.meta = s144 %>% select(Pop))

snps<-import.snpR.data(genotypes = "outputs/300/plink-pruned-max.vcf", 
      sample.meta = s144 %>% select(Pop))

snpsRAND<-import.snpR.data(genotypes = "outputs/300/plink-pruned-1m.vcf", 
      sample.meta = s72 %>% select(Pop))

snps<-import.snpR.data(genotypes = "outputs/300/plink-pruned-max-1m.vcf", 
      sample.meta = s72 %>% select(Pop))
```
```{r}
p <- plot_clusters(snps, facets = c("Pop"), viridis.option = "H")
p$plot$pca

ggsave("outputs/300/lfs-pca-1m.jpeg")
```

```{r}
p <- plot_clusters(snpsRAND, facets = c("Pop"), viridis.option = "H")
p$plot$pca

ggsave("outputs/300/lfs-pca-rand-1m.jpeg")
```



```{r}
ddf<-p$data$pca %>% as_tibble()
ddf$Pop<-factor(ddf$Pop, levels=c("ALVS","CHPI","SFBY","PETA","SUIB","HUMB","COLR","LWSH","PTLC","HRLC","SKNA","YBAK"))

labels<-ddf %>% group_by(Pop) %>% summarize(x=mean(PC1), y=mean(PC2), SampleSize=n())
labels13<-ddf %>% group_by(Pop) %>% summarize(x=mean(PC1), y=mean(PC3), SampleSize=n())

p12<-ggplot(ddf) +
  geom_point(aes(x=PC1, y=PC2, fill=Pop), pch=21, alpha=.75) +
 # geom_text_repel(data=labels, aes(x=x, y=y, label=paste0(Pop, " ", SampleSize)), max.overlaps = Inf) +
   geom_text_repel(data=labels, aes(x=x, y=y, label=Pop), max.overlaps = Inf) +
  scale_fill_viridis_d(option="H", name = "Sampling Group") +
  xlab(paste0("PC1 (", p$pca_loadings[1],"%)")) +
  ylab(paste0("PC2 (", p$pca_loadings[2],"%)")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  ggtitle("B") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) +
  theme(legend.position = "none")


p12
ggsave("outputs/300/pc1.jpeg")

p13<-ggplot(ddf) +
  geom_point(aes(x=PC1, y=PC3, fill=Pop), pch=21, alpha=.75) +
 # geom_text_repel(data=labels, aes(x=x, y=y, label=paste0(Pop, " ", SampleSize)), max.overlaps = Inf) +
   geom_text_repel(data=labels13, aes(x=x, y=y, label=Pop), max.overlaps = Inf) +
  scale_fill_viridis_d(option="H", name = "Sampling Group") +
  xlab(paste0("PC1 (", p$pca_loadings[1],"%)")) +
  ylab(paste0("PC3 (", p$pca_loadings[3],"%)")) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  ggtitle("C") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) 
 # theme(legend.position = "none")


p13
ggsave("outputs/300/pc13.jpeg")

vardf<-p$pca_loadings %>% as_tibble() %>% rename(`Percent Variance`=value) %>% head() %>% mutate(PC=1:n())

scree<-ggplot(vardf, aes(x=PC, y=`Percent Variance`)) +
  geom_line() +
  geom_point() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks = seq(1:6)) +
  theme(axis.text = element_text(size=10)) +
  theme(axis.title = element_text(size=14)) +
  theme(plot.title = element_text(hjust=0, size=16)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16)) 

scree

ggsave("outputs/300/scree.jpeg")

```

```{r}
blank <- grid.rect(gp=gpar(col="white"))
```

```{r}
ggarrange(ggplotGrob(ggarrange(scree, blank, ncol=1)), p12, p13, ncol=3, widths=c(0.5,1,1.3))

ggsave("outputs/300/supplemental-figure-s1.pdf", width=12, height=5)
```

## LD
```{sh, eval=FALSE}

cat outputs/300/1mbseqs.txt  | while read line; do bcftools view -Ov -r $line outputs/300/filtered.vcf.gz > outputs/300/vcfs/$line.vcf; done;

cat outputs/300/1mbseqs.txt  | while read line; do bcftools view -Ov -r $line outputs/300/filtered-1m.vcf.gz > outputs/300/vcfs/$line.vcf; done;
bcftools view  outputs/300/filtered-1m.vcf.gz  -r 'lcl|scaffold_26':8.25e6- > outputs/300/vcfs/26-region-1m.vcf
bcftools view  outputs/300/filtered-1m.vcf.gz  -r 'lcl|scaffold_11':17e6- > outputs/300/vcfs/11-region-1m.vcf
bcftools view  outputs/300/filtered-1m.vcf.gz  -r 'lcl|scaffold_11':19.5e6- > outputs/300/vcfs/11-region-19.5-1m.vcf
bcftools view  outputs/300/filtered-1m.vcf.gz  -r 'lcl|scaffold_11':18e6- > outputs/300/vcfs/11-region-18-1m.vcf

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;

```

```{r, eval=FALSE}
files<-list.files("outputs/300/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/300/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.1)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/300/chrom-ld/",chrom,".pdf"))
}

lapply(files, plotLd)
```


```{r}
lc<-read.delim("outputs/300/vcfs/lcl|scaffold_4.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.6)

pA<-  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("A")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("SNP Position A") +
  ylab("SNP Position B") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) 
pA
  
ggsave("outputs/300/scaffold04-ld-plot.pdf")
```


```{r}
lc<-read.delim("outputs/300/vcfs/lcl|scaffold_26.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.6)

pB<-  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("B")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("SNP Position A") +
  ylab("SNP Position B") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) 
pB
  
ggsave("outputs/300/scaffold26-ld-plot.pdf")
```

```{r}
lc2<-lc %>% filter(BP_A > 8.25e6)

ggplot(lc2) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())

```





```{r}
lc<-read.delim("outputs/300/vcfs/lcl|scaffold_11.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.6)

pC<-  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("C")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("SNP Position A") +
  ylab("SNP Position B") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) 
pC
  
ggsave("outputs/300/scaffold11-ld-plot.pdf")


lc2<-lc %>% filter(BP_A > 17e6)

pD<-ggplot(lc2) +
geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("D")+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  xlab("SNP Position A") +
  ylab("SNP Position B") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) +
  scale_x_continuous(labels = scales::label_scientific(digits = 2)) +
  scale_y_continuous(labels = scales::label_scientific(digits = 2))

  
pD
#20119207-20335395
```

```{r}
ggarrange(pA, pB, pC, pD)

ggsave("outputs/300/figure-2.pdf", width=10, height=8)
```

Getting region of interest
bcftools view lcl\|scaffold_26.vcf.gz -r 'lcl|scaffold_26':8.25e6- > linkage-26.vcf


# Check loadings
```{r}
vcf<-read.vcfR(file="outputs/300/vcfs/26-region-1m.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(s72$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(s72)

eig<-pca1$eig/sum(pca1$eig)*100

df$Pop<-factor(df$Pop, levels=c("ALVS","CHPI","SFBY","PETA","SUIB","HUMB","COLR","LWSH","PTLC","HRLC","SKNA","YBAK"))

```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H", name = "Sampling Group") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "bottom") 
scaff26<-pc12
pc12
ggsave("outputs/300/scaffold-26-region-pca.jpeg")
```
```{r}
annotcols<-df %>% select(Axis1) %>% mutate(Sex = ifelse(Axis1 > 0, "Female","Male")) %>% select(-Axis1)
```
```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .3) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```

## Let's make a heatmap  here

```{r}
library(pheatmap)
#ddf<-read_tsv("outputs/300/snps-wgs.geno.gz", col_names = c("FALSE")) %>% select(-X147)
ddf<-read_tsv("outputs/300/snps-wgs-1m.geno.gz",  col_names = c("FALSE")) %>% select(-X75)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(`FALSE`=="lcl|scaffold_26")
ddf2<-ddf %>% filter(X2 %in% tops$Position)
m<-ddf2 %>% select(-`FALSE`, -X2) %>% as.matrix()
annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Sex<-annotcols$Sex
annot
```

```{r}
as.ggplot(pheatmap(m, annotation_col = annot, cluster_rows = FALSE,
                   labels_col = s72$ID, labels_row = ddf2$X2, fontsize_row = 14, fontsize_col = 9)) +
  ggtitle("Heatmap of Longfin Smelt Variants on scaffold_26\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/300/lfs-sex-heatmap-1m.pdf", width=20, height=12)
```

Chrom 11

Scaffold 11


```{r}
lc<-read.delim("outputs/300/vcfs/lcl|scaffold_11.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.6)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_11 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())
  
ggsave("outputs/300/scaffold-11-ld-plot.pdf")
```

```{r}
lc2<-lc %>% filter(BP_A > 17e6)

ggplot(lc2) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_11 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())

```

```{r}
ddf<-read_tsv("outputs/300/snps-wgs-1m.geno.gz", col_names = c("FALSE")) %>% select(-X75)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(`FALSE`=="lcl|scaffold_11")
ddf2<-ddf %>% filter(X2 > 19.5e6)
m<-ddf2 %>% select(-`FALSE`, -X2) %>% as.matrix()

```

```{r}
genos<-read_csv("outputs/304/genotypes.csv")
genos<-s72 %>% left_join(genos, by=c("ID"="Ind")) %>% mutate(Genotype=ifelse(ID %in% c("HUMB_007","HUMB_008","HUMB_004"),"Het",Genotype))
genos<-genos %>% mutate(NewGeno=ifelse(Genotype=="RHom","RHom", ifelse(Genotype=="Het","Het","AHom")))
genos$NewGeno<-replace_na(genos$NewGeno,"AHom")
annot11<-as.data.frame(genos %>% select(NewGeno) )
annot11$Genotype<-annot11$NewGeno
annot11<-annot11 %>% select(-NewGeno)
rownames(annot11)<-colnames(m)

colors11<-viridis::viridis(3, option="cividis")
ann11_colors = list(
  Genotype = c( AHom=colors11[1],Het=colors11[2], RHom=colors11[3]))

```

```{r}
as.ggplot(pheatmap(m, labels_row = ddf2$X2, labels_col = s72$ID,fontsize_row = 14, fontsize_col = 11,
                   annotation_col = annot11, annotation_colors = ann11_colors, color = viridis(3,option="magma"))) +
  ggtitle("Heatmap of Longfin Smelt Variants on scaffold_11\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/300/lfs-11-heatmap-1m.pdf", width=20, height=8)
ggsave("outputs/300/figure-4.pdf", width=20, height=8)

```

```{r}
#vcf<-read.vcfR(file="outputs/300/vcfs/11-region-1m.vcf")
vcf<-read.vcfR(file="outputs/300/vcfs/11-region-19.5-1m.vcf")
#vcf<-read.vcfR(file="outputs/300/vcfs/11-region-18-1m.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(s72$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(s72)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=genos$NewGeno), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="cividis", name = "Candidate Chromosomal\nInversion Genotype") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "bottom") 
scaff11<-pc12
ggsave("outputs/300/scaffold-11-region-pca.jpeg")
```



#Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/300/vcfs",pattern = "*.vcf", full.names = TRUE)

plotPCA<-function(file) {
  meta<-s72
  chrom<-gsub("outputs/300/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$Pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=ID))
pc12

ggsave(paste0("outputs/300/chrom-pcs/",chrom,".pdf"))
}

lapply(files, plotPCA)
```

## Local PCA

```{sh, eval=FALSE}

 gunzip -c filtered-1m.vcf.gz | perl -pe 's/lcl\|//g' > filtered-1m-lcl.vcf
bgzip, tabix
cat 1mbseqs.txt | perl -pe 's/lcl\|//g' > 1mbseqs-lcl.txt

cat outputs/300/1mbseqs-lcl.txt  | while read line; do bcftools view -Ob -r $line outputs/300/filtered-1m-lcl.vcf.gz  > outputs/300/bcfs/$line.bcf; done;

for f in outputs/300/bcfs/*.bcf; do bcftools index $f; done;
```


```{r}
samples<-s72 %>% select(ID) %>% rename(ID=ID )

population<-s72 %>% select(Pop) %>% rename(population=Pop)

table<-cbind(samples, population)
write.table(table, "outputs/300/bcfs/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```

```{sh, eval=FALSE}
./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/300/bcfs -t snp -s 40 -m 5 -I /Users/mac/github/longfin-histories/outputs/300/bcfs/sample_info.tsv -j 300

./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/300/bcfs -t snp -s 10 -m 5 -I /Users/mac/github/longfin-histories/outputs/300/bcfs/sample_info.tsv -j 300

cp lostruct_results/type_snp_size_20_weights_none_jobid_300/mds_coords.csv ~/github/longfin-histories/outputs/300/
cp lostruct_results/type_snp_size_40_weights_none_jobid_300/mds_coords.csv ~/github/longfin-histories/outputs/300/mds_coords-40.csv
cp lostruct_results/type_snp_size_10_weights_none_jobid_300/mds_coords.csv ~/github/longfin-histories/outputs/300/mds_coords-10.csv

```



```{r}
mds<-read_csv("outputs/300//mds_coords.csv") # 20 snp windows
#mds<-read_csv("outputs/300//mds_coords-40.csv")
#mds<-read_csv("outputs/300//mds_coords-10.csv")

mds$chrom<-factor(mds$chrom,levels=c("scaffold_1","scaffold_2","scaffold_3","scaffold_4","scaffold_5","scaffold_6","scaffold_7","scaffold_8","scaffold_9","scaffold_10","scaffold_11","scaffold_12","scaffold_13","scaffold_14","scaffold_15","scaffold_16","scaffold_17","scaffold_18","scaffold_19","scaffold_20","scaffold_21","scaffold_22","scaffold_23","scaffold_24","scaffold_25","scaffold_26","scaffold_27","scaffold_28","scaffold_29"))
#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
ggsave("outputs/300/mds1.jpeg")

ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')


ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


### MDS1

```{r}
p1<-MDS1 %>% arrange(chrom,window) %>%mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```


### MDS2
```{r}
p2<-MDS2 %>% arrange(chrom,window) %>% mutate(Index=1:n())
out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```


```{r}
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```


### MDS3
```{r}
p3<-MDS3  %>% arrange(chrom,window) %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```
```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))


#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```

## Make phylo

outputs/300/plink-pruned.vcf.gz

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i plink-pruned.vcf.gz;
conda deactivate

removing SFBY_75 and 83 due to missingness
COLR_03 has a lot of missing data to

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s plink-pruned.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s plink-pruned.min4.phy.varsites.phy --seqtype DNA --redo

```{r}
dat<-read.dna(file="outputs/300/trees/plink-pruned.min4.phy")
write.nexus.data(dat, file="outputs/300/trees/plink-pruned.min4.nex")
```




Getting night smelt or others, have six night smelt in data/night

/home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa

bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa

(base) maccamp@farm:~/longfin-histories/data/night$ cat *stats
Night01,1419907,664540,0.142485
Night02,3506968,1655132,0.36273
Night03,5603663,2306524,0.50107
Night04,1826813,849640,0.182858
Night05,5981135,2311750,0.489617
Night06,4802268,1986320,0.429618

Rooted tree....

```{r}
night<-read_csv("meta/night.csv")
m148<-sag %>% bind_rows(night) %>% filter(Dedup > 2e5) %>% filter(!ID %in% c("SFBY_083","SFBY_075")  )
```

```{r}
m148 %>% select(Path) %>% write_tsv("bamlists/148.bamlist", col_names = FALSE)
m148 %>% select(ID) %>% write_tsv("bamlists/148.names", col_names = FALSE)

m148$Pop<-factor(m150$Pop, levels=c("ALVS","CHPI","SFBY","PETA","SUIB","HUMB","COLR","LWSH","PTLC","HRLC","SKNA","YBAK","Night"))
```


```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 133 -bam bamlists/148.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/300/snps-wgs-phylo  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/300/snps-phylo.out 2> outputs/300/snps-phylo.err &

#	-> Number of sites retained after filtering: 31185 


#create vcf (I should make a basic script to convert, prune and whatnot at a certain MAF)
plink --tped snps-wgs-phylo.tped --tfam snps-wgs-phylo.tfam  --out plink-binary-phylo --recode --allow-extra-chr --noweb
plink --ped plink-binary-phylo.ped --map plink-binary-phylo.map --recode vcf --allow-extra-chr -out plink-phylo

module load bcftools
bcftools reheader -s ../../bamlists/148.names -o plink-phylo-renamed.vcf plink-phylo.vcf
bcftools +fill-tags plink-phylo-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.10' > filtered-phylo.vcf

#Need to try a lower MAF I think
#this command hasn't been run
bcftools +fill-tags plink-phylo-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.05' > filtered-phylo-05.vcf

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered-phylo.vcf > plink-phylo-pruned.vcf
#placement of the night isn't great what about 

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered-phylo.vcf > plink-phylo-pruned-max.vcf

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered-phylo-05.vcf > plink-phylo-pruned-max05.vcf

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered-phylo-05.vcf > plink-phylo-pruned-rand.vcf

```
 in outputs/300/night
 
 
Checking missing data

```{sh, eval=FALSE}
paste \
<(bcftools query -f '[%SAMPLE\t]\n' plink-phylo-pruned.vcf | head -1 | tr '\t' '\n') \
<(bcftools query -f '[%GT\t]\n' plink-phylo-pruned.vcf | awk -v OFS="\t" '{for (i=1;i<=NF;i++) if ($i == "./.") sum[i]+=1 } END {for (i in sum) print i, sum[i] / NR }' | sort -k1,1n | cut -f 2) > missingness.stats
````

These have a lot of missing data

(base) ➜  night git:(main) ✗ cat missingness.stats | sort -k 2 -n | tail
SKNA_049	0.245657
LWSH_050	0.249697
PTLC_006	0.252121
SFBY_134	0.288889
PTLC_014	0.290909
PTLC_010	0.292929
PTLC_028	0.369697
PTLC_013	0.396768
PTLC_019	0.407273
COLR_003	0.471919

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i plink-phylo-pruned.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i plink-phylo-pruned-max.vcf;

conda deactivate



iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s plink-phylo-pruned.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s plink-phylo-pruned.min4.phy.varsites.phy --seqtype DNA --redo

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s plink-phylo-pruned-max.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s plink-phylo-pruned-max.min4.phy.varsites.phy --seqtype DNA --bcor 0.95 --redo

removed three SFBY sequences, perhaps hybrids or high missing data?

SFBY 142
SFBY 148
SFBY 134

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s down.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s down.phy.varsites.phy --seqtype DNA --redo

```{r}
dat<-read.dna(file="outputs/300/night/plink-phylo-pruned.min4.phy")
write.nexus.data(dat, file="outputs/300/night/plink-phylo-pruned.min4.nex")

dat<-read.dna(file="outputs/300/night/plink-phylo-pruned-max.min4.phy")
write.nexus.data(dat, file="outputs/300/night/max.nex")


dat<-read.dna(file="outputs/300/night/down-max.phy.varsites.phy")
write.nexus.data(dat, file="outputs/300/night/down-max.nex")


dat<-read.dna(file="outputs/300/allsnps/filtered-phylo-05.min4.phy")
write.nexus.data(dat, file="outputs/300/allsnps/allsnps05.nex")

dat<-read.dna(file="outputs/300/allsnps/filtered-phylo.min4.phy")
write.nexus.data(dat, file="outputs/300/allsnps/allsnps.nex")
```

I think we need to downsample the fish in SFBY...

Dowsampled southern group for 80 total fsih

iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s down-max.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s down-max.phy.varsites.phy --seqtype DNA --redo

Hmmm...

https://www.sciencedirect.com/science/article/pii/S1055790324001222 doesn't prune snps


in outputs/300/allsnps

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i filtered-phylo-05.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i filtered-phylo.vcf;

conda deactivate

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s filtered-phylo.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s filtered-phylo.min4.phy.varsites.phy --seqtype DNA --redo

# Region 26

```{r}
snps<-import.snpR.data(genotypes = "outputs/300/vcfs/26-region-1m.vcf", 
      sample.meta = s72 %>% select(Pop))
```

```{r}
p <- plot_clusters(snps, facets = c("Pop"), viridis.option = "H")
p$plot$pca
sexmeta<-p$data$pca %>% select(Pop, .sample.id, PC1) %>% mutate(Sex=ifelse(PC1 > 0, "Male","Female"))
```
```{r}
ddf<-p$data$pca %>% as_tibble()
ddf<-ddf %>% left_join(sexmeta)
ddf$Pop<-factor(ddf$Pop, levels=c("ALVS","CHPI","SFBY","PETA","SUIB","HUMB","COLR","LWSH","PTLC","HRLC","SKNA","YBAK"))
magma<-viridis(3,option="magma")
p12<-ggplot(ddf) +
  geom_point(aes(x=PC1, y=PC2, fill=Sex), pch=21, alpha=.75) +
  xlab(paste0("PC1 ", p$pca_loadings[1],"%")) +
  ylab(paste0("PC2 ", p$pca_loadings[2],"%")) +
  scale_fill_manual(values=c("purple","orange") )+
  theme_bw() +
  theme(panel.grid = element_blank()) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) 

panel3A<-p12
p12
```
```{r}
ddf %>% group_by(Sex) %>% summarize(Count=n())
```
```{r}
hets<-calc_hs(snps)
ho<-get.snpR.stats(hets, stats="hs")
ddf2<-left_join(ddf,ho$sample)
```

```{r}
panel3B<-ggplot(ddf2) +
  geom_boxplot(aes(x=Sex, y=hs, fill=Sex)) +
  ylab("Mean Heterozygosity") +
  xlab("Candidate Sex") +
  scale_fill_manual(values=c("purple","orange")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ggtitle("B") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) 


panel3B
```


Scaffold 11

# Region 11
vcf<-read.vcfR(file="outputs/300/vcfs/11-region-19.5-1m.vcf")

```{r}
panel3C<-scaff11 + theme(legend.position = "right") +
    scale_fill_viridis_d(option="cividis", name ="Candidate CI\n Genotype") +
  ggtitle("C") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) 

panel3C
```

```{r}
snps<-import.snpR.data(genotypes = "outputs/300/vcfs/11-region-19.5-1m.vcf", 
      sample.meta = genos %>% select(NewGeno))
```


```{r}
hets<-calc_hs(snps)
ho<-get.snpR.stats(hets, stats="hs")
sampleSizes<-ho$sample %>% group_by(NewGeno) %>% summarize(Count=n())
```

```{r}
panel3D<-ggplot(ho$sample) +
  geom_boxplot(aes(x=NewGeno, y=hs, fill=NewGeno), alpha=0.85) +
  geom_text(data=sampleSizes,aes(x=NewGeno, label=paste0("n = ",Count)),y=3.26)+
  ylab("Mean Heterozygosity") +
  xlab("Candidate Chromosomal Inversion Genotype") +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  scale_fill_viridis_d(option="cividis", name ="Candidate CI\n Genotype") +
  ggtitle("D") +
  theme(plot.title = element_text(hjust=0, size=16, face="bold")) 


panel3D
```
  
Putting it together

```{r}
ggarrange(panel3A, panel3B, panel3C, panel3D)

ggsave("outputs/300/figure-3.pdf", width=11, height=8.5)
ggsave("outputs/300/figure-3.jpeg", width=11, height=8.5)

```
  