---
title: "300-polished-genome"
output: html_document
date: "2025-08-07"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(ape)
library(tidyverse)
library(ggpubr)
library(ggplotify)
library(snpR)
library(ggtree)
library(tanggle)
library(pheatmap)
```

Reference genome: /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa

I have data from Saglam et al. aligned to this reference.

```{r}
sag<-read_csv("meta/saglam.csv", col_names=c("ID","Reads","Dedup","Cov")) %>% mutate(Path=paste0("data/polish-saglam/",ID,".sort.flt.bam"))
sag$Pop<-gsub("_\\d+","",sag$ID)
sag<-sag %>% relocate(Pop)
```

```{r}
sag %>% filter(Dedup > 2e5) %>% group_by(Pop) %>% summarize(Count=n(), Mean=mean(Dedup), Median=median(Dedup))
```

```{r}
ggplot(sag %>% filter(Dedup > 2e5)) +
  geom_histogram(aes(x=Dedup)) 
```


## Generate covariance matrix for sanity check

```{r}
s144<-sag %>% filter(Dedup > 2e5)
s144 %>% select(Path) %>% write_tsv("bamlists/s144.bamlist", col_names = FALSE)
s144 %>% select(ID) %>% write_tsv("bamlists/s144.names", col_names = FALSE)

s144$Pop<-factor(s144$Pop, levels=c("ALVS","CHPI","SFBY","PETA","SUIB","HUMB","COLR","LWSH","PTLC","HRLC","SKNA","YBAK"))
```

```{sh, eval=FALSE}
srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/s144.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-minInd 130 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/300/s144-ibs-90 >outputs/300/s144-ibs-90.out 2> outputs/300/s144-ibs-90.err &

srun -p high -t 10:00:00 --mem=64G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12  $HOME/angsd/angsd -P 12  \
-bam bamlists/s144.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \ 
-minInd 108 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/300/s144-ibs-75 >outputs/300/s144-ibs-75.out 2> outputs/300/s144-ibs-75.err &
```

	-> Number of sites retained after filtering: 60378 @ 90%


```{r}
m <- as.matrix(read.table("outputs/300/s144-ibs-90.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(s144)

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Pop), pch=21, cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="magma") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("B") +
  theme(plot.title = element_text(hjust=0, size=16)) 

pc12

```
```{r}
pc13<-ggplot(covs) +
  geom_point(aes(x=V1, y=V3, fill=Pop), pch=21, cex=3, alpha=0.75) +
  xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC3", " ", round((100*var[3]),2), "%", sep = "")) +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  #scale_fill_viridis_c() +
  scale_fill_viridis_d(option="magma") +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("C") +
  theme(plot.title = element_text(hjust=0, size=16)) 

pc13

```

Scree plot
 1 -0.0326 -0.0561 -0.710  SFBY  SFBY_075 2764947  705849 0.0981 data/polish-saglam/SFBY_075.sort.flt.bam
 2 -0.0387 -0.0642 -0.656  SFBY  SFBY_083 2616986  683952 0.100  data/polish-saglam/SFBY_083.sort.flt.bam
 
 These are two SFBY ones, as well as the two weird YBAK ones
 
 These SFBY ones have a lot of missing data
 
```{r}
vardf<-var %>% as_tibble() %>% rename(`Percent Variance`=value) %>% head() %>% mutate(PC=1:n())

scree<-ggplot(vardf, aes(x=PC, y=`Percent Variance`)) +
  geom_line() +
  geom_point() +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks = seq(1:6)) +
  theme(axis.text = element_text(size=10)) +
  theme(axis.title = element_text(size=14)) +
  theme(plot.title = element_text(hjust=0, size=16)) +
  theme(plot.title = element_text(hjust=0.5, face="bold")) +
  ggtitle("A") +
  theme(plot.title = element_text(hjust=0, size=16)) 


scree
```

## Call SNPs

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 130 -bam bamlists/s144.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/300/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/300/snps.out 2> outputs/300/snps.err &


#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/s144.names -o plink-renamed.vcf plink.vcf
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered.vcf > plink-pruned-max.vcf

```

	-> Number of sites retained after filtering: 25402 

Filter for missing individuals

```{sh, eval=FALSE}
paste \
<(bcftools query -f '[%SAMPLE\t]\n' filtered.vcf.gz | head -1 | tr '\t' '\n') \
<(bcftools query -f '[%GT\t]\n' filtered.vcf.gz | awk -v OFS="\t" '{for (i=1;i<=NF;i++) if ($i == "./.") sum[i]+=1 } END {for (i in sum) print i, sum[i] / NR }' | sort -k1,1n | cut -f 2) > missingness.stats
````

PTLC_006	0.194709
PTLC_010	0.204984
SFBY_134	0.209432
PTLC_014	0.222148
PTLC_028	0.241713
SFBY_083	0.268089
PTLC_013	0.287655
PTLC_019	0.297457
COLR_003	0.367806
SFBY_075	0.470711

Wow! These fish have over 20% missing calls.

```{r}
snpsRAND<-import.snpR.data(genotypes = "outputs/300/plink-pruned.vcf.gz", 
      sample.meta = s144 %>% select(Pop))

snps<-import.snpR.data(genotypes = "outputs/300/plink-pruned-max.vcf", 
      sample.meta = s144 %>% select(Pop))

```

```{r}
p <- plot_clusters(snpsRAND, facets = c("Pop"), viridis.option = "H")
p$plot$pca

ggsave("outputs/300/lfs-pca-rand.jpeg")
```

```{r}
p <- plot_clusters(snps, facets = c("Pop"), viridis.option = "H")
p$plot$pca

ggsave("outputs/300/lfs-pca.jpeg")
```



## LD

```{sh, eval=FALSE}

cat outputs/300/1mbseqs.txt  | while read line; do bcftools view -Ov -r $line outputs/300/filtered.vcf.gz > outputs/300/vcfs/$line.vcf; done;

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;

```

```{r, eval=FALSE}
files<-list.files("outputs/300/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/300/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.1)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/300/chrom-ld/",chrom,".pdf"))
}

lapply(files, plotLd)
```



```{r}
lc<-read.delim("outputs/300/vcfs/lcl|scaffold_26.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.2)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())
  
ggsave("outputs/300/scaffold26-ld-plot.pdf")
```

```{r}
lc2<-lc %>% filter(BP_A > 8.25e6)

ggplot(lc2) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())

```
Getting region of interest
bcftools view lcl\|scaffold_26.vcf.gz -r 'lcl|scaffold_26':8.25e6- > linkage-26.vcf


# Check loadings
```{r}
vcf<-read.vcfR(file="outputs/300/vcfs/linkage-26.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(s144$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(s144)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
pc12
```
```{r}
annotcols<-df %>% select(Axis1) %>% mutate(Sex = ifelse(Axis1 > 0, "Female","Male")) %>% select(-Axis1)
```
```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .45) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```

## Let's make a heatmap  here

```{r}
library(pheatmap)
ddf<-read_tsv("outputs/300/snps-wgs.geno.gz", col_names = c("FALSE")) %>% select(-X147)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(`FALSE`=="lcl|scaffold_26")
ddf2<-ddf %>% filter(X2 %in% tops$Position)
m<-ddf2 %>% select(-`FALSE`, -X2) %>% as.matrix()
annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Sex<-annotcols$Sex
annot
```

```{r}
as.ggplot(pheatmap(m, annotation_col = annot, cluster_rows = TRUE,
                   labels_col = s144$ID, labels_row = ddf2$X2, fontsize_row = 14, fontsize_col = 6)) +
  ggtitle("Heatmap of Longfin Smelt Variants on scaffold_26\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/300/lfs-sex-heatmap.pdf", width=20, height=12)
```

#Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/300/vcfs",pattern = "*.vcf", full.names = TRUE)

plotPCA<-function(file) {
  meta<-s144
  chrom<-gsub("outputs/300/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$Pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") +
  geom_text_repel(aes(x=Axis1, y=Axis2,label=ID))
pc12

ggsave(paste0("outputs/300/chrom-pcs/",chrom,".pdf"))
}

lapply(files, plotPCA)
```


Make phylo

outputs/300/plink-pruned.vcf.gz

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i plink-pruned.vcf.gz;
conda deactivate

removing SFBY_75 and 83 due to missingness
COLR_03 has a lot of missing data to

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s plink-pruned.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s plink-pruned.min4.phy.varsites.phy --seqtype DNA --redo

```{r}
dat<-read.dna(file="outputs/300/trees/plink-pruned.min4.phy")
write.nexus.data(dat, file="outputs/300/trees/plink-pruned.min4.nex")
```




Getting night smelt or others, have six night smelt in data/night

/home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa

bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa

(base) maccamp@farm:~/longfin-histories/data/night$ cat *stats
Night01,1419907,664540,0.142485
Night02,3506968,1655132,0.36273
Night03,5603663,2306524,0.50107
Night04,1826813,849640,0.182858
Night05,5981135,2311750,0.489617
Night06,4802268,1986320,0.429618

Rooted tree....

```{r}
night<-read_csv("meta/night.csv")
m148<-sag %>% bind_rows(night) %>% filter(Dedup > 2e5) %>% filter(!ID %in% c("SFBY_083","SFBY_075")  )
```

```{r}
m148 %>% select(Path) %>% write_tsv("bamlists/148.bamlist", col_names = FALSE)
m148 %>% select(ID) %>% write_tsv("bamlists/148.names", col_names = FALSE)

m148$Pop<-factor(m150$Pop, levels=c("ALVS","CHPI","SFBY","PETA","SUIB","HUMB","COLR","LWSH","PTLC","HRLC","SKNA","YBAK","Night"))
```


```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 133 -bam bamlists/148.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/300/snps-wgs-phylo  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/300/snps-phylo.out 2> outputs/300/snps-phylo.err &

#	-> Number of sites retained after filtering: 31185 


#create vcf (I should make a basic script to convert, prune and whatnot at a certain MAF)
plink --tped snps-wgs-phylo.tped --tfam snps-wgs-phylo.tfam  --out plink-binary-phylo --recode --allow-extra-chr --noweb
plink --ped plink-binary-phylo.ped --map plink-binary-phylo.map --recode vcf --allow-extra-chr -out plink-phylo

module load bcftools
bcftools reheader -s ../../bamlists/148.names -o plink-phylo-renamed.vcf plink-phylo.vcf
bcftools +fill-tags plink-phylo-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.10' > filtered-phylo.vcf

#Need to try a lower MAF I think
#this command hasn't been run
bcftools +fill-tags plink-phylo-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.05' > filtered-phylo-05.vcf

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered-phylo.vcf > plink-phylo-pruned.vcf
#placement of the night isn't great what about 

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered-phylo.vcf > plink-phylo-pruned-max.vcf

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered-phylo-05.vcf > plink-phylo-pruned-max05.vcf

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered-phylo-05.vcf > plink-phylo-pruned-rand.vcf

```
 in outputs/300/night
 
 
Checking missing data

```{sh, eval=FALSE}
paste \
<(bcftools query -f '[%SAMPLE\t]\n' plink-phylo-pruned.vcf | head -1 | tr '\t' '\n') \
<(bcftools query -f '[%GT\t]\n' plink-phylo-pruned.vcf | awk -v OFS="\t" '{for (i=1;i<=NF;i++) if ($i == "./.") sum[i]+=1 } END {for (i in sum) print i, sum[i] / NR }' | sort -k1,1n | cut -f 2) > missingness.stats
````

These have a lot of missing data

(base) ➜  night git:(main) ✗ cat missingness.stats | sort -k 2 -n | tail
SKNA_049	0.245657
LWSH_050	0.249697
PTLC_006	0.252121
SFBY_134	0.288889
PTLC_014	0.290909
PTLC_010	0.292929
PTLC_028	0.369697
PTLC_013	0.396768
PTLC_019	0.407273
COLR_003	0.471919

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i plink-phylo-pruned.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i plink-phylo-pruned-max.vcf;

conda deactivate



iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s plink-phylo-pruned.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s plink-phylo-pruned.min4.phy.varsites.phy --seqtype DNA --redo

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s plink-phylo-pruned-max.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s plink-phylo-pruned-max.min4.phy.varsites.phy --seqtype DNA --bcor 0.95 --redo

removed three SFBY sequences, perhaps hybrids or high missing data?

SFBY 142
SFBY 148
SFBY 134

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s down.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s down.phy.varsites.phy --seqtype DNA --redo

```{r}
dat<-read.dna(file="outputs/300/night/plink-phylo-pruned.min4.phy")
write.nexus.data(dat, file="outputs/300/night/plink-phylo-pruned.min4.nex")

dat<-read.dna(file="outputs/300/night/plink-phylo-pruned-max.min4.phy")
write.nexus.data(dat, file="outputs/300/night/max.nex")


dat<-read.dna(file="outputs/300/night/down-max.phy.varsites.phy")
write.nexus.data(dat, file="outputs/300/night/down-max.nex")


dat<-read.dna(file="outputs/300/allsnps/filtered-phylo-05.min4.phy")
write.nexus.data(dat, file="outputs/300/allsnps/allsnps05.nex")

dat<-read.dna(file="outputs/300/allsnps/filtered-phylo.min4.phy")
write.nexus.data(dat, file="outputs/300/allsnps/allsnps.nex")
```

I think we need to downsample the fish in SFBY...

Dowsampled southern group for 80 total fsih

iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s down-max.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s down-max.phy.varsites.phy --seqtype DNA --redo

Hmmm...

https://www.sciencedirect.com/science/article/pii/S1055790324001222 doesn't prune snps


in outputs/300/allsnps

conda activate py2;
~/github/mccloud-rrt/vcf2phylip.py -i filtered-phylo-05.vcf;
~/github/mccloud-rrt/vcf2phylip.py -i filtered-phylo.vcf;

conda deactivate

iqtree2  -nt AUTO -m MFP+ASC -bb 1000 -s filtered-phylo.min4.phy --seqtype DNA --redo
iqtree2  -nt AUTO -m MFP+ASC -bb 10000 -s filtered-phylo.min4.phy.varsites.phy --seqtype DNA --redo

