---
title: "303-life-history-fish"
output: html_document
date: "2025-11-19"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(ape)
library(tidyverse)
library(ggpubr)
library(ggrepel)
library(ggplotify)
library(snpR)
library(ggtree)
library(tanggle)
library(pheatmap)
```

Can I use the life history fish ?


-> Need to align to polished genome
`/home/maccamp/longfin-histories/data/alt-polish`

74 seqs

```{r}
fish<-read_csv("Longfin smelt meta and tissues/all_fish_clust_ids.csv") 
fish$SampleID<-gsub("_0","",fish$fishid)
fish$SampleID<-gsub("_","",fish$SampleID)
fish<-fish %>% relocate(SampleID) 
fish %>% group_by(clust_name_100) %>% summarize(Count=n())
```
`bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa` 

We also have known sex fish:
https://www.ncbi.nlm.nih.gov/sra/SRX30480321[accn]	Fam27 Dam      SRR35397523
https://www.ncbi.nlm.nih.gov/sra/SRX30480320[accn]	Fam27 Sire     SRR35397524    
https://www.ncbi.nlm.nih.gov/sra/SRX30480296[accn]	Fam11 Dam     SRR35397548
https://www.ncbi.nlm.nih.gov/sra/SRX30480295[accn]	Fam11 Sire     SRR35397549
https://www.ncbi.nlm.nih.gov/sra/SRX30480272[accn]	Fam06 Dam      SRR35397572
https://www.ncbi.nlm.nih.gov/sra/SRX30480271[accn]	Fam06 Sire     SRR35397573
https://www.ncbi.nlm.nih.gov/sra/SRX30480262[accn]	Fam45 Dam      SRR35397582
https://www.ncbi.nlm.nih.gov/sra/SRX30480261[accn]	Fam45 Sire     SRR35397583
https://www.ncbi.nlm.nih.gov/sra/SRX30480234[accn]	Fam01 Dam      SRR35397610
https://www.ncbi.nlm.nih.gov/sra/SRX30480217[accn]	Fam01 Sire     SRR35397627

putting in data/parents

need to dump
```{sh, eval=FALSE}
module load sratoolkit

for f in SRR*lite.1; do echo fastq-dump --outdir ./ --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip $f >> split-commands.sh; done;

module load parallel
srun -p high -t 02:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < split-commands.sh  

```

## Check metrics, compare to previous version.

(base) maccamp@farm:~/longfin-histories/data/alt-polish$ cat *stats > alt-polish-stat.csv

```{r}
gf<-read_csv("meta/alt-polish-stat.csv", col_names = c("SampleID","Aligned","Dedup","Cov"))
ggf<-left_join(gf,fish) %>% mutate(Path=paste0("data/alt-polish/",SampleID,".sort.flt.bam"))
ggf
```


```{r}
ggplot(ggf) +
  geom_histogram(aes(x=Dedup)) 
```
```{r}
ggplot(ggf %>% filter(Dedup > 5e5)) +
  geom_histogram(aes(x=Dedup)) 
```

Need to downsample

Let's downsample fish over 1.5e6 to 1.56

```{r}
ggf2<-ggf %>% mutate(Frac=1.5e6/Dedup) %>% mutate(NewPath=ifelse(Dedup > 1.5e6, paste0("data/downsample/",SampleID,".reduced.bam"),Path))
downsample<-ggf2 %>% filter(Dedup > 1.5e6 ) %>%
  mutate(ReductionCommand = paste0("samtools view -bs ",Frac, " ", Path, " > ", NewPath))

downsample %>% select(ReductionCommand) %>% write_csv("303.1-downsample.sh", col_names = FALSE)

```

```{sh, eval=FALSE}
module load parallel
srun -p high -t 02:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10  < 303.1-downsample.sh > outputs/303/downsample.stdout 2> outputs/303/downsample.stderr
#index .bams
for f in *.bam; do samtools index $f; done;
```


## Create bamlists
e.g.


```{r}
m33<-ggf2 %>% filter(Dedup > 5e5)
m27<-ggf2 %>% filter(Dedup > 1e6)

m33 %>% select(NewPath) %>% write_tsv("bamlists/m33.bamlist", col_names = FALSE)
m33 %>% select(SampleID) %>% write_tsv("bamlists/m33.names", col_names = FALSE)

m27 %>% select(NewPath) %>% write_tsv("bamlists/m27.bamlist", col_names = FALSE)
m27 %>% select(SampleID) %>% write_tsv("bamlists/m27.names", col_names = FALSE)

```


# Call SNPs

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 30 -bam bamlists/m33.bamlists -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/303/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/303/snps.out 2> outputs/303/snps.err &

srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 24 -bam bamlists/m27.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/303/snps-wgs-1m  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/303/snps-1m.out 2> outputs/303/snps-1m.err &


#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/s144.names -o plink-renamed.vcf plink.vcf
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered.vcf > plink-pruned-max.vcf


## Here I'm looking at only higher coverage individuals
plink --tped snps-wgs-1m.tped --tfam snps-wgs-1m.tfam  --out plink-binary-1m --recode --allow-extra-chr --noweb
plink --ped plink-binary-1m.ped --map plink-binary-1m.map --recode vcf --allow-extra-chr -out plink-1m

module load bcftools
bcftools reheader -s ../../bamlists/s72.names -o plink-renamed-1m.vcf plink-1m.vcf
bcftools +fill-tags plink-renamed-1m.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered-1m.vcf

#Getting region of interest

bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered-1m.vcf > plink-pruned-1m.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered-1m.vcf > plink-pruned-max-1m.vcf

```
