---
title: "303-life-history-fish"
output: html_document
date: "2025-11-19"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(ape)
library(tidyverse)
library(ggpubr)
library(ggrepel)
library(ggplotify)
library(snpR)
library(ggtree)
library(tanggle)
library(pheatmap)
```

Can I use the life history fish ?


-> Need to align to polished genome
`/home/maccamp/longfin-histories/data/alt-polish`

74 seqs

```{r}
fish<-read_csv("Longfin smelt meta and tissues/all_fish_clust_ids.csv") 
fish$SampleID<-gsub("_0","",fish$fishid)
fish$SampleID<-gsub("_","",fish$SampleID)
fish<-fish %>% relocate(SampleID) 
fish %>% group_by(clust_name_100) %>% summarize(Count=n())
```
`bash ../../101.2-do-align.sh to-align.txt /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa` 

We also have known sex fish:
https://www.ncbi.nlm.nih.gov/sra/SRX30480321[accn]	Fam27 Dam      SRR35397523
https://www.ncbi.nlm.nih.gov/sra/SRX30480320[accn]	Fam27 Sire     SRR35397524    
https://www.ncbi.nlm.nih.gov/sra/SRX30480296[accn]	Fam11 Dam     SRR35397548
https://www.ncbi.nlm.nih.gov/sra/SRX30480295[accn]	Fam11 Sire     SRR35397549
https://www.ncbi.nlm.nih.gov/sra/SRX30480272[accn]	Fam06 Dam      SRR35397572
https://www.ncbi.nlm.nih.gov/sra/SRX30480271[accn]	Fam06 Sire     SRR35397573
https://www.ncbi.nlm.nih.gov/sra/SRX30480262[accn]	Fam45 Dam      SRR35397582
https://www.ncbi.nlm.nih.gov/sra/SRX30480261[accn]	Fam45 Sire     SRR35397583
https://www.ncbi.nlm.nih.gov/sra/SRX30480234[accn]	Fam01 Dam      SRR35397610
https://www.ncbi.nlm.nih.gov/sra/SRX30480217[accn]	Fam01 Sire     SRR35397627

putting in data/parents

need to dump
```{sh, eval=FALSE}
module load sratoolkit

for f in SRR*lite.1; do echo fastq-dump --outdir ./ --skip-technical --read-filter pass --dumpbase --split-3 --gzip --clip $f >> split-commands.sh; done;

module load parallel
srun -p high -t 02:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10 < split-commands.sh  

```

## Check metrics, compare to previous version.

(base) maccamp@farm:~/longfin-histories/data/alt-polish$ cat *stats > alt-polish-stat.csv

```{r}
gf<-read_csv("meta/alt-polish-stat.csv", col_names = c("SampleID","Aligned","Dedup","Cov"))
ggf<-left_join(gf,fish) %>% mutate(Path=paste0("data/alt-polish/",SampleID,".sort.flt.bam"))
ggf
```


```{r}
ggplot(ggf) +
  geom_histogram(aes(x=Dedup)) 
```
```{r}
ggplot(ggf %>% filter(Dedup > 5e5)) +
  geom_histogram(aes(x=Dedup)) 
```

Need to downsample

Let's downsample fish over 2e6 to 2e6

```{r}
ggf2<-ggf %>% mutate(Frac=2e6/Dedup) %>% mutate(NewPath=ifelse(Dedup > 2e6, paste0("data/downsample/",SampleID,".reduced.bam"),Path))
downsample<-ggf2 %>% filter(Dedup > 2e6 ) %>%
  mutate(ReductionCommand = paste0("samtools view -bs ",Frac, " ", Path, " > ", NewPath))

downsample %>% select(ReductionCommand) %>% write_csv("303.1-downsample.sh", col_names = FALSE)

```

```{sh, eval=FALSE}
module load parallel
srun -p high -t 02:00:00 --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=10 parallel -j 10  < 303.1-downsample.sh > outputs/303/downsample.stdout 2> outputs/303/downsample.stderr
#index .bams
for f in *.bam; do samtools index $f; done;
```


## Create bamlists

```{r}
m33<-ggf2 %>% filter(Dedup > 5e5)
m27<-ggf2 %>% filter(Dedup > 1e6)

m33 %>% select(NewPath) %>% write_tsv("bamlists/m33.bamlist", col_names = FALSE)
m33 %>% select(SampleID) %>% write_tsv("bamlists/m33.names", col_names = FALSE)

m27 %>% select(NewPath) %>% write_tsv("bamlists/m27.bamlist", col_names = FALSE)
m27 %>% select(SampleID) %>% write_tsv("bamlists/m27.names", col_names = FALSE)

```


# Call SNPs

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 30 -bam bamlists/m33.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/303/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/303/snps.out 2> outputs/303/snps.err &

srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 24 -bam bamlists/m27.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/303/snps-wgs-1m  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/303/snps-1m.out 2> outputs/303/snps-1m.err &


#create vcf
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/m33.names -o plink-renamed.vcf plink.vcf
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered.vcf > plink-pruned-max.vcf


## Here I'm looking at only higher coverage individuals
plink --tped snps-wgs-1m.tped --tfam snps-wgs-1m.tfam  --out plink-binary-1m --recode --allow-extra-chr --noweb
plink --ped plink-binary-1m.ped --map plink-binary-1m.map --recode vcf --allow-extra-chr -out plink-1m

module load bcftools
bcftools reheader -s ../../bamlists/m27.names -o plink-renamed-1m.vcf plink-1m.vcf
bcftools +fill-tags plink-renamed-1m.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered-1m.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered-1m.vcf > plink-pruned-1m.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered-1m.vcf > plink-pruned-max-1m.vcf

#need to bgzip and tabix files

```


16195 variants and 33 people pass filters and QC.
31954 variants and 27 people pass filters and QC.


## LD

```{sh, eval=FALSE}

#cat outputs/300/1mbseqs.txt  | while read line; do bcftools view -Ov -r $line outputs/303/filtered.vcf.gz > outputs/303/vcfs/$line.vcf; done;

cat outputs/300/1mbseqs.txt  | while read line; do bcftools view -Ov -r $line outputs/303/filtered-1m.vcf.gz > outputs/303/vcfs/$line.vcf; done;
bcftools view  outputs/303/filtered-1m.vcf.gz  -r 'lcl|scaffold_26':8.25e6- > outputs/303/vcfs/26-region-1m.vcf

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;

```

```{r, eval=FALSE}
files<-list.files("outputs/303/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/303/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.6)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/303/chrom-ld/",chrom,".pdf"))
}

lapply(files, plotLd)
```



```{r}
lc<-read.delim("outputs/303/vcfs/lcl|scaffold_26.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.5)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())
  
ggsave("outputs/303/scaffold26-ld-plot.pdf")
```

```{r}
lc2<-lc %>% filter(BP_A > 8.25e6)

ggplot(lc2) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())

```
Getting region of interest
bcftools view lcl\|scaffold_26.vcf.gz -r 'lcl|scaffold_26':8.25e6- > linkage-26.vcf


# Check loadings
```{r}
vcf<-read.vcfR(file="outputs/303/vcfs/26-region-1m.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(s144$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(m27)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=clust_name_100), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
pc12
ggsave("outputs/303/scaffold-26-region-pca.jpeg")
```

```{r}
annotcols<-df %>% select(Axis1) %>% mutate(Sex = ifelse(Axis1 > 0, "Female","Male")) %>% select(-Axis1)
```

```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .3) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```

## Let's make a heatmap  here

```{r}
library(pheatmap)
#ddf<-read_tsv("outputs/303/snps-wgs.geno.gz", col_names = c("FALSE")) %>% select(-X147)
ddf<-read_tsv("outputs/303/snps-wgs-1m.geno.gz",  col_names = c("FALSE")) %>% select(-X30)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(`FALSE`=="lcl|scaffold_26")
ddf2<-ddf %>% filter(X2 %in% tops$Position)
m<-ddf2 %>% select(-`FALSE`, -X2) %>% as.matrix()
annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Sex<-annotcols$Sex
annot
```

```{r}
as.ggplot(pheatmap(m, annotation_col = annot, cluster_rows = FALSE,
                   labels_col = s144$ID, labels_row = ddf2$X2, fontsize_row = 14, fontsize_col = 9)) +
  ggtitle("Heatmap of Longfin Smelt Variants on scaffold_26\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/303/lfs-sex-heatmap-1m.pdf", width=20, height=12)
```

#Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/303/vcfs",pattern = "*.vcf", full.names = TRUE)
m27$Pop<-m27$clust_name_100
plotPCA<-function(file) {
  meta<-m27
  chrom<-gsub("outputs/303/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$Pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
  #geom_text_repel(aes(x=Axis1, y=Axis2,label=ID))
pc12

ggsave(paste0("outputs/303/chrom-pcs/",chrom,".pdf"))
}

lapply(files, plotPCA)
```

## Local PCA

```{sh, eval=FALSE}

 gunzip -c filtered-1m.vcf.gz | perl -pe 's/lcl\|//g' > filtered-1m-lcl.vcf
bgzip, tabix
cat 1mbseqs.txt | perl -pe 's/lcl\|//g' > 1mbseqs-lcl.txt

cat outputs/300/1mbseqs-lcl.txt  | while read line; do bcftools view -Ob -r $line outputs/303/filtered-1m-lcl.vcf.gz  > outputs/303/bcfs/$line.bcf; done;

for f in outputs/303/bcfs/*.bcf; do bcftools index $f; done;
```


```{r}
samples<-s72 %>% select(ID) %>% rename(ID=ID )

population<-s72 %>% select(Pop) %>% rename(population=Pop)

table<-cbind(samples, population)
write.table(table, "outputs/303/bcfs/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```

```{sh, eval=FALSE}

./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/303/bcfs -t snp -s 20 -m 5 -I /Users/mac/github/longfin-histories/outputs/303/bcfs/sample_info.tsv -j 303

cp lostruct_results/type_snp_size_20_weights_none_jobid_303/mds_coords.csv ~/github/longfin-histories/outputs/303/


```


cp lostruct_results/type_snp_size_40_weights_none_jobid_303/mds_coords.csv ~/github/longfin-histories/outputs/303/mds_coords-40.csv
cp lostruct_results/type_snp_size_10_weights_none_jobid_303/mds_coords.csv ~/github/longfin-histories/outputs/303/mds_coords-10.csv


./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/303/bcfs -t snp -s 40 -m 5 -I /Users/mac/github/longfin-histories/outputs/303/bcfs/sample_info.tsv -j 303





```{r}
mds<-read_csv("outputs/303//mds_coords.csv") # 20 snp windows
#mds<-read_csv("outputs/300//mds_coords-40.csv")
#mds<-read_csv("outputs/300//mds_coords-10.csv")

mds$chrom<-factor(mds$chrom,levels=c("scaffold_1","scaffold_2","scaffold_3","scaffold_4","scaffold_5","scaffold_6","scaffold_7","scaffold_8","scaffold_9","scaffold_10","scaffold_11","scaffold_12","scaffold_13","scaffold_14","scaffold_15","scaffold_16","scaffold_17","scaffold_18","scaffold_19","scaffold_20","scaffold_21","scaffold_22","scaffold_23","scaffold_24","scaffold_25","scaffold_26","scaffold_27","scaffold_28","scaffold_29"))
#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
ggsave("outputs/300/mds1.jpeg")

ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')


ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS4, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
```


### MDS1

```{r}
p1<-MDS1 %>% arrange(chrom,window) %>%mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```


### MDS2
```{r}
p2<-MDS2 %>% arrange(chrom,window) %>% mutate(Index=1:n())
out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```


```{r}
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```


### MDS3
```{r}
p3<-MDS3  %>% arrange(chrom,window) %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```
```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))


#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```
