---
title: "304-sfbay"
output: html_document
date: "2025-11-20"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```



```{r}
library(ape)
library(vcfR)
library(adegenet)
library(tidyverse)
library(ggpubr)
library(ggrepel)
library(ggplotify)
library(snpR)
library(ggtree)
library(tanggle)
library(RColorBrewer)
library(pheatmap)
```

Get San Francisco Bay samples, see 300

```{r}
sfb<-read_csv("meta/sfb.csv")
sfb %>% select(ID) %>% write_tsv("bamlists/26.names", col_names = FALSE)
sfb %>% select(Path) %>% write_tsv("bamlists/26.bamlist", col_names = FALSE)
```

26 samples

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 23 -bam bamlists/26.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/304/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/304/snps.out 2> outputs/304/snps.err &
```
	-> Number of sites retained after filtering: 67487 


Create VCF

```{sh, eval=FALSE}
plink --tped snps-wgs.tped --tfam snps-wgs.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/26.names -o plink-renamed.vcf plink.vcf
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered.vcf > plink-pruned-max.vcf
bgzip filtered.vcf
```

58569 in filtered

```{sh, eval=FALSE}
cat outputs/300/1mbseqs.txt  | while read line; do bcftools view -Ov -r $line outputs/304/filtered.vcf.gz > outputs/304/vcfs/$line.vcf; done;
bcftools view  outputs/304/filtered.vcf.gz  -r 'lcl|scaffold_26':8.25e6- > outputs/304/vcfs/26-region-1m.vcf
bcftools view  outputs/304/filtered.vcf.gz  -r 'lcl|scaffold_11':18e6- > outputs/304/vcfs/11-region-1m.vcf

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;
```


```{r, eval=FALSE}
files<-list.files("outputs/304/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/304/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.3)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/304/chrom-ld/",chrom,".pdf"))
}

lapply(files, plotLd)
```



```{r}
lc<-read.delim("outputs/304/vcfs/lcl|scaffold_26.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.1)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())
  
ggsave("outputs/304/scaffold26-ld-plot.pdf")
```

```{r}
lc2<-lc %>% filter(BP_A > 8.25e6)

ggplot(lc2) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())

```
```{r}
lc<-read.delim("outputs/304/vcfs/lcl|scaffold_11.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.5)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_11 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())
```


```{r}
lc2<-lc %>% filter(BP_A > 17e6)

ggplot(lc2) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_11 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())

```

# Check loadings
```{r}
vcf<-read.vcfR(file="outputs/304/vcfs/26-region-1m.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(sfb$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(sfb)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
pc12
ggsave("outputs/304/scaffold-26-region-pca.jpeg")
```

```{r}
annotcols<-df %>% select(Axis1) %>% mutate(Sex = ifelse(Axis1 > 0, "Female","Male")) %>% select(-Axis1)
```

```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .3) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```

## Let's make a heatmap  here

```{r}
ddf<-read_tsv("outputs/304/snps-wgs.geno.gz", col_names = c("FALSE")) %>% select(-X29)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(`FALSE`=="lcl|scaffold_26")
ddf2<-ddf %>% filter(X2 %in% tops$Position)
m<-ddf2 %>% select(-`FALSE`, -X2) %>% as.matrix()
annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Sex<-annotcols$Sex
annot
```

```{r}
as.ggplot(pheatmap(m, cluster_rows = TRUE, annotation_col = annot,
                   labels_col = sfb$ID, labels_row = ddf2$X2, fontsize_row = 14, fontsize_col = 9)) +
  ggtitle("Heatmap of Longfin Smelt Variants on scaffold_26\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/304/lfs-sex-heatmap-1m.pdf", width=20, height=12)
```


Scaffold 11

```{r}
ddf<-read_tsv("outputs/304/snps-wgs.geno.gz", col_names = c("FALSE")) %>% select(-X29)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(`FALSE`=="lcl|scaffold_11")
ddf2<-ddf %>% filter(X2 > 20e6)
m<-ddf2 %>% select(-`FALSE`, -X2) %>% as.matrix()
```

```{r}

as.ggplot(pheatmap(m, labels_row = ddf2$X2, labels_col = sfb$ID,fontsize_row = 14, fontsize_col = 9)) +
  ggtitle("Heatmap of Longfin Smelt Variants on scaffold_11\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/304/lfs-11-heatmap.pdf", width=12, height=8)

```


```{r}
vcf<-read.vcfR(file="outputs/304/vcfs/11-region-1m.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(sfb$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(sfb)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
df<-df %>% mutate(Genotype=ifelse(Axis1 >2, "RHom", ifelse(Axis1 > 0 & Axis1 < 2, "Het","AHom" )))
df %>% group_by(Genotype, Pop) %>% summarize(Count=n())
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Genotype), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
pc12
ggsave("outputs/304/scaffold-11-region-pca.jpeg")
```



```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .3) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```

Heterozygosity?

#Chrom PCS

```{r, eval=FALSE}
files<-list.files("outputs/304/vcfs",pattern = "*.vcf", full.names = TRUE)

plotPCA<-function(file) {
  meta<-sfb
  chrom<-gsub("outputs/304/vcfs/","",file)
  chrom<-gsub(".vcf","", chrom)
  
  vcf<-read.vcfR(file=file)
  genind<-vcfR2genind(vcf)
  genind@pop<-as.factor(meta$Pop)

gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(meta)
eig<-pca1$eig/sum(pca1$eig)*100


pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop),pch=21, alpha=0.75, cex=2) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
  #geom_text_repel(aes(x=Axis1, y=Axis2,label=ID))
pc12

ggsave(paste0("outputs/304/chrom-pcs/",chrom,".pdf"))
}

lapply(files, plotPCA)
```

# Local PCA

```{sh, eval=FALSE}

gunzip -c filtered.vcf.gz | perl -pe 's/lcl\|//g' > filtered-lcl.vcf
#bgzip, tabix
# only using up scaffold 29

cat outputs/304/1mbseqs-lcl.txt  | while read line; do bcftools view -Ob -r $line outputs/304/filtered-lcl.vcf.gz  > outputs/304/bcfs/$line.bcf; done;

for f in outputs/304/bcfs/*.bcf; do bcftools index $f; done;
```


```{r}
samples<-sfb %>% select(ID) %>% rename(ID=ID )

population<-sfb %>% select(Pop) %>% rename(population=Pop)

table<-cbind(samples, population)
write.table(table, "outputs/304/bcfs/sample_info.tsv", quote = TRUE, row.names = FALSE, sep="\t")
```


```{sh, eval=FALSE}
./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/304/bcfs -t snp -s 20 -m 5 -I /Users/mac/github/longfin-histories/outputs/304/bcfs/sample_info.tsv -j 304

cp lostruct_results/type_snp_size_20_weights_none_jobid_304/mds_coords.csv ~/github/longfin-histories/outputs/304/

./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/304/bcfs -t snp -s 15 -m 5 -I /Users/mac/github/longfin-histories/outputs/304/bcfs/sample_info.tsv -j 304
cp lostruct_results/type_snp_size_15_weights_none_jobid_304/mds_coords.csv ~/github/longfin-histories/outputs/304/mds_coords-15.csv

./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/304/bcfs -t snp -s 30 -m 5 -I /Users/mac/github/longfin-histories/outputs/304/bcfs/sample_info.tsv -j 304
cp lostruct_results/type_snp_size_30_weights_none_jobid_304/mds_coords.csv ~/github/longfin-histories/outputs/304/mds_coords-30.csv

#doesn't work
#./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/304/bcfs -t snp -s 10 -m 5 -I /Users/mac/github/longfin-histories/outputs/304/bcfs/sample_info.tsv -j 304
#cp lostruct_results/type_snp_size_10_weights_none_jobid_304/mds_coords.csv ~/github/longfin-histories/outputs/304/mds_coords-10.csv

#./run_lostruct.R -i /Users/mac/github/longfin-histories/outputs/304/bcfs -t snp -s 5 -m 5 -I /Users/mac/github/longfin-histories/outputs/304/bcfs/sample_info.tsv -j 304
#cp lostruct_results/type_snp_size_5_weights_none_jobid_304/mds_coords.csv ~/github/longfin-histories/outputs/304/mds_coords-5.csv
```


```{r}
mds<-read_csv("outputs/304/mds_coords.csv") # 20 snp windows
#mds<-read_csv("outputs/304/mds_coords-30.csv")
#mds<-read_csv("outputs/304/mds_coords-10.csv")

#mds<-read_csv("outputs/300//mds_coords-10.csv")

mds$chrom<-factor(mds$chrom,levels=c("scaffold_1","scaffold_2","scaffold_3","scaffold_4","scaffold_5","scaffold_6","scaffold_7","scaffold_8","scaffold_9","scaffold_10","scaffold_11","scaffold_12","scaffold_13","scaffold_14","scaffold_15","scaffold_16","scaffold_17","scaffold_18","scaffold_19","scaffold_20","scaffold_21","scaffold_22","scaffold_23","scaffold_24","scaffold_25","scaffold_26","scaffold_27","scaffold_28","scaffold_29"))
#make tidy
tidymds<-mds %>% gather(MDS, Value, 3:6)
MDS1<-filter(tidymds, MDS=="MDS1") %>% rename(MDS1=MDS) %>% rename(Value1=Value)
MDS2<-filter(tidymds, MDS=="MDS2") %>% rename(MDS2=MDS) %>% rename(Value2=Value)
MDS3<-filter(tidymds, MDS=="MDS3") %>% rename(MDS3=MDS) %>% rename(Value3=Value)
MDS4<-filter(tidymds, MDS=="MDS4") %>% rename(MDS4=MDS) %>% rename(Value4=Value)
ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS2, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')
ggsave("outputs/300/mds1.jpeg")

ggplot(mds)+
  geom_point(aes(x=MDS1, y=MDS3, fill=chrom), pch=21, alpha=0.75) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  theme(legend.position = 'none')


```


### MDS1

```{r}
p1<-MDS1 %>% arrange(chrom,window) %>%mutate(Index=1:n())

out <- boxplot.stats(p1$Value1)$out
out_ind <- which(p1$Value1 %in% c(out))
length(out_ind)
outliers<-p1[out_ind,]
outliers %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```

```{r}
#places to put labels based on index
chroms<-p1 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p1) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p1$Value1), ymax=max(p1$Value1)), fill=mycolors, alpha=0.25) +
  geom_point(data=p1, aes(x=Index, y=Value1, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers, aes(x=Index, y=Value1), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS1") +
  xlab("Chromosome")
```


### MDS2
```{r}
p2<-MDS2 %>% arrange(chrom,window) %>% mutate(Index=1:n())
out2 <- boxplot.stats(p2$Value2)$out
out_ind2 <- which(p2$Value2 %in% c(out2))
length(out_ind2)
outliers2<-p2[out_ind2,]
outliers2 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```


```{r}
chroms<-p2 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))

#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

ggplot(p2) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p2$Value2), ymax=max(p2$Value2)), fill=mycolors, alpha=0.25) +
  geom_point(data=p2, aes(x=Index, y=Value2, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers2, aes(x=Index, y=Value2), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS2") +
  xlab("Chromosome")
```


### MDS3
```{r}
p3<-MDS3  %>% arrange(chrom,window) %>% mutate(Index=1:n())

out3 <- boxplot.stats(p3$Value3)$out
out_ind3 <- which(p3$Value3 %in% c(out3))
out_ind3
outliers3<-p3[out_ind3,]
outliers3 %>% group_by(chrom) %>% summarize(Count=n()) %>% arrange(-Count)
```
```{r}
#places to put labels based on index
chroms<-p3 %>% group_by(chrom) %>% mutate(Start=min(Index), Stop=max(Index)) %>% select(chrom,Start,Stop) %>% unique() %>% 
  mutate(Position=round((Start+Stop)/2))


#Make enough colors
nb.cols <- nrow(chroms)
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)


ggplot(p3) +
  geom_rect(data=chroms, aes(xmin=Start, xmax=Stop, ymin=min(p3$Value3), ymax=max(p3$Value3)), fill=mycolors, alpha=0.25) +
  geom_point(data=p3, aes(x=Index, y=Value3, color=chrom), alpha=0.75, cex=0.5) +
  geom_point(data=outliers3, aes(x=Index, y=Value3), color="black", cex=0.5) +
  theme_bw() +
  scale_x_continuous(breaks=chroms$Position, labels=chroms$chrom) +
  theme(axis.text.x = element_text(angle=45, hjust = 1, face="bold", size=6)) +
  theme(legend.title = element_blank()) +
  theme(panel.grid = element_blank()) +
  theme(legend.position = "none") +
  scale_color_manual(values=mycolors) +
  ylab("MDS3") +
  xlab("Chromosome")
```
