---
title: "305-combined"
output: html_document
date: "2025-11-21"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```


```{r}
library(ape)
library(viridis)
library(tidyverse)
library(ggpubr)
library(ggrepel)
library(ggplotify)
library(snpR)
library(ggtree)
library(tanggle)
library(vcfR)
library(adegenet)
library(pheatmap)
```

Combine life history fish, SFB fish and parents. Let's combine all Saglam fish with 1e6 reads

```{r}
lfs<-read_csv("meta/life-history-fish.csv") %>% mutate(Category="Life History") %>% mutate(Pop="Life History") %>% mutate(Sex="UNK") %>% select(Pop, SampleID, Aligned, Dedup, Cov, clust_name_100, Category, Path, Sex)

sag<-read_csv("meta/saglam.csv", col_names=c("ID","Reads","Dedup","Cov")) %>% mutate(Path=paste0("data/polish-saglam/",ID,".sort.flt.bam"))
sag$Pop<-gsub("_\\d+","",sag$ID)
sag<-sag %>% rename(Aligned=Reads, SampleID=ID) %>% mutate(Category="Saglam") %>% mutate(clust_name_100="UNK") %>% mutate(Sex="UNK")
parents<-read_csv("meta/parents.csv")


bind_rows(sag,parents) %>% filter(Dedup > 1e6) %>% mutate(Total=n()) %>% group_by(Category, clust_name_100, Sex, Total) %>% summarize(Count=n()) %>% relocate(-Total)

m1<-bind_rows(sag,parents) %>% filter(Dedup > 1e6)
m1$clust_name_100<-factor(m1$clust_name_100, levels=c("FW","LS","MS","HS","UNK"))
m1 %>% select(Path) %>% write_tsv("bamlists/combined.bamlist", col_names = FALSE)
m1 %>% select(SampleID) %>% write_tsv("bamlists/combined.names", col_names = FALSE)


m2<-bind_rows(sag,parents) %>% filter(Dedup > 8e5)
m2 %>% select(Path) %>% write_tsv("bamlists/combined2.bamlist", col_names = FALSE)
m2 %>% select(SampleID) %>% write_tsv("bamlists/combined2.names", col_names = FALSE)

m3<-bind_rows(sag,parents,lfs) %>% filter(Dedup > 8e5)
m3 %>% select(Path) %>% write_tsv("bamlists/combined3.bamlist", col_names = FALSE)
m3 %>% select(SampleID) %>% write_tsv("bamlists/combined3.names", col_names = FALSE)

```
(base) maccamp@farm:~/longfin-histories/outputs/305$ grep 20131490 snps-wgs.tped 

lcl|scaffold_11 lcl|scaffold_11_20131490 0 20131490	A A	G A	A A	G G	G G	G G	G G	G G	G A	G G	G G	G A	G A	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G A	G G	G G	A A	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G A	G G	G G	A A	A A	G G	G A	G A	A A	G G	A A	G A	G G	G G	G A	0 0	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G G	G A	0 0	G G	G A	A A	A A	A A	G G

How to proceed?  One option is to call SNPs with sag fish and parents and look at LD and determine sex determination mechanism and scaffold 11. Then call snps from Life History fish (20131490?)
Otherwise, they can all be called together, then filtered later.


Trying 85% threshold first, want scaffold_11:20131490
Next 90% threshold

-rf /home/maccamp/genomes/longfin-polished/scaffold1126.txt \

```{sh, eval=FALSE}
srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 73 -bam bamlists/combined.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/305/snps-wgs  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/305/snps.out 2> outputs/305/snps.err &

srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 85 -bam bamlists/combined2.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/305/snps-wgs-2  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/305/snps-2.out 2> outputs/305/snps-2.err &

srun -t 12:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 110 -bam bamlists/combined3.bamlist -ref /home/maccamp/genomes/longfin-polished/lfs_run3_scaffolds_final.fa \
-rf /home/maccamp/genomes/longfin-polished/1mbseqs.txt \
-out outputs/305/snps-wgs-3  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.9 -doPlink 2  > outputs/305/snps-3.out 2> outputs/305/snps-3.err &


```

using two chroms
With MAF of 0.01 with 53 test fish, 10K+ snps 
With MAF of 0.05 with 53 test fish  2671 snps 85% missing. Two snps 'tag' inversion

MAF 0.05, 75% missing 3674 SNPs, and 5 'tag' inversion

Create VCF

```{sh, eval=FALSE}
plink --tped snps-wgs-2.tped --tfam snps-wgs-2.tfam  --out plink-binary --recode --allow-extra-chr --noweb
plink --ped plink-binary.ped --map plink-binary.map --recode vcf --allow-extra-chr -out plink

module load bcftools
bcftools reheader -s ../../bamlists/combined2.names -o plink-renamed.vcf plink.vcf
bcftools +fill-tags plink-renamed.vcf | bcftools view -i 'F_MISSING < 0.10' | bcftools view -i 'INFO/MAF > 0.01' > filtered.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand filtered.vcf > plink-pruned.vcf
bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode maxAF filtered.vcf > plink-pruned-max.vcf
bgzip filtered.vcf
```


```{sh, eval=FALSE}
paste \
<(bcftools query -f '[%SAMPLE\t]\n' filtered.vcf.gz | head -1 | tr '\t' '\n') \
<(bcftools query -f '[%GT\t]\n' filtered.vcf.gz | awk -v OFS="\t" '{for (i=1;i<=NF;i++) if ($i == "./.") sum[i]+=1 } END {for (i in sum) print i, sum[i] / NR }' | sort -k1,1n | cut -f 2) > missingness.stats
```

```{r}
miss<-read_tsv("outputs/305/missingness.stats", col_names=c("SampleID","Missingness"))
m22<-m2 %>% select(-Missingness) %>% left_join(miss, by=c("SampleID"="SampleID"))
m22<- m22 %>% dplyr::filter(Missingness<0.10)
m22 %>% select(SampleID) %>% write_tsv("outputs/305/m22.names", col_names = FALSE)
```


# PCA

```{r}
snps<-import.snpR.data(genotypes = "outputs/305/plink-pruned-max.vcf", 
      sample.meta = m2 %>% select(Pop))
```

```{r}
p <- plot_clusters(snps, facets = c("Pop"), viridis.option = "H")
p$plot$pca

ggsave("outputs/305/lfs-pca.jpeg")
```

# LD
```{sh, eval=FALSE}

bcftools view -S outputs/305/m22.names outputs/305/filtered.vcf.gz > outputs/305/filtered.22.vcf
bgzip outputs/305/filtered.22.vcf
tabix outputs/305/filtered.22.vcf.gz 
#cat outputs/300/1mbseqs.txt  | while read line; do bcftools view -Ov -r $line outputs/305/filtered.vcf.gz > outputs/305/vcfs/$line.vcf; done;
cat outputs/300/1mbseqs.txt  | while read line; do bcftools view -Ov -r $line outputs/305/filtered.22.vcf.gz > outputs/305/vcfs/$line.vcf; done;
bcftools view  outputs/305/filtered.22.vcf.gz  -r 'lcl|scaffold_26':8.25e6- > outputs/305/vcfs/26-region.vcf
bcftools view  outputs/305/filtered.22.vcf.gz  -r 'lcl|scaffold_11':17e6- > outputs/305/vcfs/11-region.vcf
bcftools view  outputs/305/filtered.22.vcf.gz  -r 'lcl|scaffold_11':19e6- > outputs/305/vcfs/11-region-2.vcf

for f in *.vcf; do plink --vcf $f --r2 inter-chr --ld-window-r2 0.1 --out `basename $f vcf`ldf --allow-extra-chr --double-id; done;

```

```{r, eval=FALSE}
files<-list.files("outputs/305/vcfs",pattern = "*.ldf.ld", full.names = TRUE)

plotLd<-function(file) {
  chrom<-gsub("outputs/305/vcfs/","",file)
  chrom<-gsub(".ldf.ld","", chrom)
  lc<-read.delim(file,sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.1)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle(paste0(chrom))+
  theme_bw() +
  theme(panel.grid = element_blank())
ggsave(paste0("outputs/305/chrom-ld/",chrom,".pdf"))
}

lapply(files, plotLd)
```



```{r}
lc<-read.delim("outputs/305/vcfs/lcl|scaffold_26.ldf.ld",sep="",stringsAsFactors=FALSE) %>% as_tibble() %>% arrange(R2) %>%  filter(R2 >0.5)

  ggplot(lc) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())
  
ggsave("outputs/305/scaffold26-ld-plot.pdf")
```

```{r}
lc2<-lc %>% filter(BP_A > 8.25e6)

ggplot(lc2) +
  geom_point(aes(x=BP_A, y=BP_B, color=R2), alpha=0.5) +
  scale_color_gradient(low="khaki1", high="red") + 
  ggtitle("lcl|scaffold_26 LD")+
  theme_bw() +
  theme(panel.grid = element_blank())

```
Getting region of interest
bcftools view lcl\|scaffold_26.vcf.gz -r 'lcl|scaffold_26':8.25e6- > linkage-26.vcf


# Check loadings
```{r}
vcf<-read.vcfR(file="outputs/305/vcfs/26-region.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(m22$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(m22 %>% filter())

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
pc12
ggsave("outputs/305/scaffold-26-region-pca.jpeg")
```


```{r}
loadings<-pca1$c1 %>% as_tibble()
loadings$Allele<-rownames(pca1$c1) 

loadings
loadings$Position<-gsub("lcl\\|.*_|\\.\\d$","",loadings$Allele)
loadings$Position<-as.numeric(loadings$Position)
loadings$Chrom<-gsub("_\\d+\\.[0|1]","",loadings$Allele)
tops<-loadings %>% arrange(-CS1) %>% slice_max(order_by = CS1,prop = .2) %>% select(-CS2, -CS3) %>% 
  mutate(MajorMinor=gsub("lcl\\|*\\.","",Allele)) 

tops %>% relocate(Chrom, Position, CS1)

```


## Scaffold 11


```{r}
vcf<-read.vcfR(file="outputs/305/vcfs/11-region.vcf")
genind<-vcfR2genind(vcf)
genind@pop<-as.factor(m22$Pop)
```


```{r}
gen<-genind
X <- tab(gen, NA.method="mean")

pca1 <- dudi.pca(X,scannf=FALSE,scale=FALSE,nf=3)

df<-pca1$li
df<-df %>% bind_cols(m22)

eig<-pca1$eig/sum(pca1$eig)*100
```

```{r}
df<-df %>% mutate(Genotype=ifelse(Axis1 < -5, "RHom", ifelse(Axis1 > -5 & Axis1 < 0, "Het","AHom" )))
#df %>% group_by(Genotype, Pop) %>% summarize(Count=n())
pc12<-ggplot(df) + 
  geom_point(aes(x=Axis1, y=Axis2, fill=Pop), alpha=0.75, cex=2, pch=21) +
  theme_bw() +
  xlab(paste0("PC1 ",round(eig[1],2), "%")) +
  ylab(paste0("PC2 ",round(eig[2],2), "%")) +
  theme(panel.grid = element_blank()) +
  scale_fill_viridis_d(option="H") +
  guides(fill = guide_legend(override.aes = list(pch = 22))) +
  theme(legend.position = "top") 
pc12
ggsave("outputs/305/scaffold-11-region-pca.jpeg")
```

# Heatmaps of key regions

```{r}
ddf<-read_tsv("outputs/305/snps-wgs-2.geno.gz", col_names = c("FALSE")) %>% select(-X103)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(`FALSE`=="lcl|scaffold_26")
ddf2<-ddf %>% filter(X2 %in% tops$Position)
m<-ddf2 %>% select(-`FALSE`, -X2) %>% as.matrix()
annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Sex<-m2$Sex
#annot$Pop<-m2$Pop
ann_colors = list(
  Sex = c( Female="purple",Male="orange",UNK="grey"))

```

```{r}
as.ggplot(pheatmap(m, cluster_rows = FALSE, annotation_col = annot, color = viridis(3,option="magma", alpha=0.9), annotation_colors =  ann_colors,
                   labels_col = m2$SampleID, labels_row = ddf2$X2, fontsize_row = 14, fontsize_col = 8)) +
  ggtitle("Heatmap of Longfin Smelt Variants on scaffold_26\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/305/lfs-sex-heatmap.pdf", width=20, height=12)
```


Scaffold 11

```{r}
ddf<-read_tsv("outputs/305/snps-wgs.geno.gz", col_names = FALSE) %>% select(-X83)
ddf[ddf==-1]<-NA
ddf<-ddf %>% filter(X1=="lcl|scaffold_11")
ddf2<-ddf %>% filter(X2 > 19e6)
#ddf2 <- ddf2 %>% filter(X2 %in% c(20131490,20144334,20263784,20263652,29263756))

m<-ddf2 %>% select(-X1, -X2) %>% as.matrix()

annot<-data.frame(sample=as.character(colnames(m))) %>% column_to_rownames("sample")

annot$Pheno<-m1$clust_name_100

ann_colors = list(
  Pheno = c(FW="orange", LS="red", MS="purple", HS="blue",UNK="grey"))

```

```{r}

as.ggplot(pheatmap(m, annotation_col = annot, annotation_colors =  ann_colors, color = viridis(3,option="magma", alpha=0.9),
                   labels_row = ddf2$X2, labels_col = m1$SampleID, fontsize_row = 12, fontsize_col = 8, cluster_rows = TRUE, cluster_cols = TRUE))+
            ggtitle("Heatmap of Longfin Smelt Variants on scaffold_11\n") + 
  theme(plot.title = element_text(hjust=0.5))
ggsave("outputs/305/lfs-11-heatmap.pdf", width=14, height=8)

```

Derived/inverted type appears to be linked to lower salinity environments.